{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "provenance": [],
      "authorship_tag": "ABX9TyMIKJmDgs/gYGTcflaRxom6",
      "include_colab_link": true
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "view-in-github",
        "colab_type": "text"
      },
      "source": [
        "<a href=\"https://colab.research.google.com/github/SilviaFrizzarini/comandos-linux/blob/main/C%C3%B3pia_de_Untitled1.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Pipeline Aula Prática - Parte2"
      ],
      "metadata": {
        "id": "FLHLrEl_Jbcj"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "Clonar o Git - renatopuga/lmabrasil-hg38"
      ],
      "metadata": {
        "id": "KBTypobzJngv"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "#Código na célula será interpretado como comandos Bash.\n",
        "%%bash\n",
        "#Remove o diretório chamado lmabrasil-hg38 e todo o seu conteúdo.\n",
        "rm -rf lmabrasil-hg38\n",
        "#Faz o download (clona) do repositório do GitHub localizado no URL.\n",
        "git clone https://github.com/renatopuga/lmabrasil-hg38.git"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "3qGWscydPJPn",
        "outputId": "d39f7564-3c81-4dbf-f3e6-be32be8ada51"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "Cloning into 'lmabrasil-hg38'...\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Instalar bcftools com plugin (+split-vep)"
      ],
      "metadata": {
        "id": "f2Ji1eKwLJkX"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "#Código na célula será interpretado como comandos Bash.\n",
        "%%bash\n",
        "#Clona o repositório htslib do GitHub + submódulos.\n",
        "git clone --recurse-submodules https://github.com/samtools/htslib.git\n",
        "#Clona o repositório bcftools do GitHub.\n",
        "git clone https://github.com/samtools/bcftools.git\n",
        "#Altera o diretório de trabalho para o diretório bcftools.\n",
        "cd bcftools\n",
        "#Compila o código-fonte do bcftools.\n",
        "make\n",
        "#Instala o programa compilado no sistema.\n",
        "make install"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "PxYln6oRJoj0",
        "outputId": "be4cd82f-6355-4cf2-aea6-12109d7d047b"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Submodule path 'htscodecs': checked out '51794289ac47455209c333182b6768f99a613948'\n",
            "cd ../htslib && make htslib.pc.tmp\n",
            "make[1]: Entering directory '/content/htslib'\n",
            "echo '# Default htscodecs.mk generated by Makefile' > htscodecs.mk\n",
            "echo 'include $(HTSPREFIX)htscodecs_bundled.mk' >> htscodecs.mk\n",
            "./hts_probe_cc.sh 'gcc' '-g -Wall -O2 -fvisibility=hidden ' '-fvisibility=hidden' >> htscodecs.mk\n",
            "sed -e '/^static_libs=/s/@static_LIBS@/-lz -lm -lbz2 -llzma -lcurl/;s#@[^-][^@]*@##g' htslib.pc.in > htslib.pc.tmp\n",
            "make[1]: Leaving directory '/content/htslib'\n",
            "cd ../htslib && make htslib_static.mk\n",
            "make[1]: Entering directory '/content/htslib'\n",
            "sed -n '/^static_libs=/s/[^=]*=/HTSLIB_static_LIBS = /p;/^static_ldflags=/s/[^=]*=/HTSLIB_static_LDFLAGS = /p' htslib.pc.tmp > htslib_static.mk\n",
            "make[1]: Leaving directory '/content/htslib'\n",
            "echo '/* Basic config.h generated by Makefile */' > config.h\n",
            "echo '#define ENABLE_BCF_PLUGINS 1' >> config.h\n",
            "echo '#define PLUGIN_EXT \".so\"' >> config.h\n",
            "echo '#define BCFTOOLS_VERSION \"1.21-60-g1535cc23\"' > version.h\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o main.o main.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o vcfindex.o vcfindex.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o tabix.o tabix.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o vcfstats.o vcfstats.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o vcfisec.o vcfisec.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o vcfmerge.o vcfmerge.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o vcfquery.o vcfquery.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o vcffilter.o vcffilter.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib    -c -o filter.o filter.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o vcfsom.o vcfsom.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o vcfnorm.o vcfnorm.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o vcfgtcheck.o vcfgtcheck.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o vcfview.o vcfview.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o vcfannotate.o vcfannotate.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o vcfroh.o vcfroh.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o vcfconcat.o vcfconcat.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o vcfcall.o vcfcall.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o mcall.o mcall.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o vcmp.o vcmp.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o gvcf.o gvcf.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o reheader.o reheader.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o convert.o convert.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o vcfconvert.o vcfconvert.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o tsv2vcf.o tsv2vcf.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o vcfcnv.o vcfcnv.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o vcfhead.o vcfhead.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o HMM.o HMM.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o consensus.o consensus.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o ploidy.o ploidy.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o bin.o bin.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o hclust.o hclust.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o version.o version.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o regidx.o regidx.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o smpl_ilist.o smpl_ilist.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o csq.o csq.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o vcfbuf.o vcfbuf.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o mpileup.o mpileup.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o bam2bcf.o bam2bcf.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o bam2bcf_indel.o bam2bcf_indel.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o bam2bcf_iaux.o bam2bcf_iaux.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o bam2bcf_edlib.o bam2bcf_edlib.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o read_consensus.o read_consensus.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o bam_sample.o bam_sample.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o vcfsort.o vcfsort.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o cols.o cols.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o extsort.o extsort.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o dist.o dist.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o abuf.o abuf.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o ccall.o ccall.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o em.o em.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o prob1.o prob1.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o kmin.o kmin.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o str_finder.o str_finder.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o gff.o gff.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o edlib.o edlib.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o mpileup2/mpileup.o mpileup2/mpileup.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib  -DPLUGINPATH='\"/usr/local/libexec/bcftools\"' -c -o vcfplugin.o vcfplugin.c\n",
            "cd ../htslib && make config.h\n",
            "make[1]: Entering directory '/content/htslib'\n",
            "echo '/* Default config.h generated by Makefile */' > config.h\n",
            "echo '#ifndef _XOPEN_SOURCE' >> config.h\n",
            "echo '#define _XOPEN_SOURCE 600' >> config.h\n",
            "echo '#endif' >> config.h\n",
            "echo '#define HAVE_LIBBZ2 1' >> config.h\n",
            "echo '#define HAVE_LIBLZMA 1' >> config.h\n",
            "echo '#ifndef __APPLE__' >> config.h\n",
            "echo '#define HAVE_LZMA_H 1' >> config.h\n",
            "echo '#endif' >> config.h\n",
            "echo '#define HAVE_DRAND48 1' >> config.h\n",
            "echo '#define HAVE_LIBCURL 1' >> config.h\n",
            "if [ \"x1\" != \"x\" ]; then \\\n",
            "    echo '#define HAVE_DECL___CPUID_COUNT 1' >> config.h ; \\\n",
            "    echo '#define HAVE_DECL___GET_CPUID_MAX 1' >> config.h ; \\\n",
            "fi\n",
            "if [ \"x1\" != \"x\" ]; then \\\n",
            "    echo '#define HAVE_POPCNT 1' >> config.h ; \\\n",
            "    echo '#define HAVE_SSE4_1 1' >> config.h ; \\\n",
            "    echo '#define HAVE_SSSE3 1' >> config.h ; \\\n",
            "    echo '#if defined(HTS_ALLOW_UNALIGNED) && HTS_ALLOW_UNALIGNED == 0' >> config.h ; \\\n",
            "    echo '#define UBSAN 1' >> config.h ; \\\n",
            "    echo '#endif' >> config.h ; \\\n",
            "fi\n",
            "if [ \"x1\" != \"x\" ] ; then \\\n",
            "    echo '#define HAVE_AVX2 1' >> config.h ; \\\n",
            "fi\n",
            "if [ \"x1\" != \"x\" ] ; then \\\n",
            "    echo '#define HAVE_AVX512 1' >> config.h ; \\\n",
            "fi\n",
            "echo '#if defined __x86_64__ || defined __arm__ || defined __aarch64__' >> config.h\n",
            "echo '#define HAVE_ATTRIBUTE_CONSTRUCTOR 1' >> config.h\n",
            "echo '#endif' >> config.h\n",
            "echo '#if (defined(__x86_64__) || defined(_M_X64))' >> config.h\n",
            "echo '#define HAVE_ATTRIBUTE_TARGET 1' >> config.h\n",
            "echo '#define HAVE_BUILTIN_CPU_SUPPORT_SSSE3 1' >> config.h\n",
            "echo '#endif' >> config.h\n",
            "echo '#if defined __linux__' >> config.h\n",
            "echo '#define HAVE_GETAUXVAL' >> config.h\n",
            "echo '#elif defined __FreeBSD__' >> config.h\n",
            "echo '#define HAVE_ELF_AUX_INFO' >> config.h\n",
            "echo '#elif defined __OpenBSD__' >> config.h\n",
            "echo '// Enable extra OpenBSD checks (see simd.c)' >> config.h\n",
            "echo '#define HAVE_OPENBSD' >> config.h\n",
            "echo '#endif' >> config.h\n",
            "make[1]: Leaving directory '/content/htslib'\n",
            "cd ../htslib && make hts-object-files\n",
            "make[1]: Entering directory '/content/htslib'\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o kfunc.o kfunc.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o kstring.o kstring.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o bcf_sr_sort.o bcf_sr_sort.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o bgzf.o bgzf.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o errmod.o errmod.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o faidx.o faidx.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o header.o header.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o hfile.o hfile.c\n",
            "echo '#define HTS_VERSION_TEXT \"1.21-20-gc705bec2\"' > version.h\n",
            "printf '#define HTS_CC \"%s\"\\n#define HTS_CPPFLAGS \"%s\"\\n#define HTS_CFLAGS \"%s\"\\n#define HTS_LDFLAGS \"%s\"\\n#define HTS_LIBS \"%s\"\\n' \\\n",
            "'gcc' \\\n",
            "'' \\\n",
            "'-g -Wall -O2 -fvisibility=hidden' \\\n",
            "'-fvisibility=hidden' \\\n",
            "'-lz -lm -lbz2 -llzma -lcurl' > config_vars.h\n",
            "Updating htscodecs/htscodecs/version.h : #define HTSCODECS_VERSION_TEXT \"1.6.1\"\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o hts.o hts.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o hts_expr.o hts_expr.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o hts_os.o hts_os.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o md5.o md5.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o multipart.o multipart.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o probaln.o probaln.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o realn.o realn.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o regidx.o regidx.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o region.o region.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o sam.o sam.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o sam_mods.o sam_mods.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o simd.o simd.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o synced_bcf_reader.o synced_bcf_reader.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o vcf_sweep.o vcf_sweep.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o tbx.o tbx.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o textutils.o textutils.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o thread_pool.o thread_pool.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o vcf.o vcf.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o vcfutils.o vcfutils.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o cram/cram_codecs.o cram/cram_codecs.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o cram/cram_decode.o cram/cram_decode.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o cram/cram_encode.o cram/cram_encode.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o cram/cram_external.o cram/cram_external.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o cram/cram_index.o cram/cram_index.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o cram/cram_io.o cram/cram_io.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o cram/cram_stats.o cram/cram_stats.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o cram/mFILE.o cram/mFILE.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o cram/open_trace_file.o cram/open_trace_file.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o cram/pooled_alloc.o cram/pooled_alloc.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o cram/string_alloc.o cram/string_alloc.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o htscodecs/htscodecs/arith_dynamic.o htscodecs/htscodecs/arith_dynamic.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o htscodecs/htscodecs/fqzcomp_qual.o htscodecs/htscodecs/fqzcomp_qual.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o htscodecs/htscodecs/htscodecs.o htscodecs/htscodecs/htscodecs.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o htscodecs/htscodecs/pack.o htscodecs/htscodecs/pack.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o htscodecs/htscodecs/rANS_static4x16pr.o htscodecs/htscodecs/rANS_static4x16pr.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden -mavx2 -mpopcnt -I.  -c -o htscodecs/htscodecs/rANS_static32x16pr_avx2.o htscodecs/htscodecs/rANS_static32x16pr_avx2.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden -mavx512f -mpopcnt -I.  -c -o htscodecs/htscodecs/rANS_static32x16pr_avx512.o htscodecs/htscodecs/rANS_static32x16pr_avx512.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden -msse4.1 -mpopcnt -mssse3 -I.  -c -o htscodecs/htscodecs/rANS_static32x16pr_sse4.o htscodecs/htscodecs/rANS_static32x16pr_sse4.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o htscodecs/htscodecs/rANS_static32x16pr_neon.o htscodecs/htscodecs/rANS_static32x16pr_neon.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o htscodecs/htscodecs/rANS_static32x16pr.o htscodecs/htscodecs/rANS_static32x16pr.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o htscodecs/htscodecs/rANS_static.o htscodecs/htscodecs/rANS_static.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o htscodecs/htscodecs/rle.o htscodecs/htscodecs/rle.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o htscodecs/htscodecs/tokenise_name3.o htscodecs/htscodecs/tokenise_name3.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o htscodecs/htscodecs/utils.o htscodecs/htscodecs/utils.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o hfile_libcurl.o hfile_libcurl.c\n",
            "touch hts-object-files\n",
            "make[1]: Leaving directory '/content/htslib'\n",
            "cd ../htslib && make lib-static\n",
            "make[1]: Entering directory '/content/htslib'\n",
            "ar -rc libhts.a kfunc.o kstring.o bcf_sr_sort.o bgzf.o errmod.o faidx.o header.o hfile.o hts.o hts_expr.o hts_os.o md5.o multipart.o probaln.o realn.o regidx.o region.o sam.o sam_mods.o simd.o synced_bcf_reader.o vcf_sweep.o tbx.o textutils.o thread_pool.o vcf.o vcfutils.o cram/cram_codecs.o cram/cram_decode.o cram/cram_encode.o cram/cram_external.o cram/cram_index.o cram/cram_io.o cram/cram_stats.o cram/mFILE.o cram/open_trace_file.o cram/pooled_alloc.o cram/string_alloc.o htscodecs/htscodecs/arith_dynamic.o htscodecs/htscodecs/fqzcomp_qual.o htscodecs/htscodecs/htscodecs.o htscodecs/htscodecs/pack.o htscodecs/htscodecs/rANS_static4x16pr.o htscodecs/htscodecs/rANS_static32x16pr_avx2.o htscodecs/htscodecs/rANS_static32x16pr_avx512.o htscodecs/htscodecs/rANS_static32x16pr_sse4.o htscodecs/htscodecs/rANS_static32x16pr_neon.o htscodecs/htscodecs/rANS_static32x16pr.o htscodecs/htscodecs/rANS_static.o htscodecs/htscodecs/rle.o htscodecs/htscodecs/tokenise_name3.o htscodecs/htscodecs/utils.o hfile_libcurl.o\n",
            "ranlib libhts.a\n",
            "make[1]: Leaving directory '/content/htslib'\n",
            "gcc -rdynamic   -o bcftools main.o vcfindex.o tabix.o vcfstats.o vcfisec.o vcfmerge.o vcfquery.o vcffilter.o filter.o vcfsom.o vcfnorm.o vcfgtcheck.o vcfview.o vcfannotate.o vcfroh.o vcfconcat.o vcfcall.o mcall.o vcmp.o gvcf.o reheader.o convert.o vcfconvert.o tsv2vcf.o vcfcnv.o vcfhead.o HMM.o consensus.o ploidy.o bin.o hclust.o version.o regidx.o smpl_ilist.o csq.o vcfbuf.o mpileup.o bam2bcf.o bam2bcf_indel.o bam2bcf_iaux.o bam2bcf_edlib.o read_consensus.o bam_sample.o vcfsort.o cols.o extsort.o dist.o abuf.o ccall.o em.o prob1.o kmin.o str_finder.o gff.o edlib.o mpileup2/mpileup.o vcfplugin.o ../htslib/libhts.a -lz -lm -lbz2 -llzma -lcurl -lm -lz -ldl    -lpthread\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o test/test-rbuf.o test/test-rbuf.c\n",
            "gcc  -o test/test-rbuf test/test-rbuf.o -lz -ldl \n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o test/test-regidx.o test/test-regidx.c\n",
            "gcc   -o test/test-regidx test/test-regidx.o regidx.o ../htslib/libhts.a -lz -lm -lbz2 -llzma -lcurl -lpthread -lz -ldl \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/ad-bias.so convert.c version.c plugins/ad-bias.c  \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/add-variantkey.so version.c plugins/add-variantkey.c \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/af-dist.so bin.c version.c plugins/af-dist.c  \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/afs.so version.c plugins/afs.c  \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/allele-length.so version.c plugins/allele-length.c \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/check-ploidy.so version.c plugins/check-ploidy.c \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/check-sparsity.so version.c plugins/check-sparsity.c \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/color-chrs.so HMM.c version.c plugins/color-chrs.c  \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/contrast.so version.c plugins/contrast.c \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/counts.so version.c plugins/counts.c \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/dosage.so version.c plugins/dosage.c \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/fill-AN-AC.so version.c plugins/fill-AN-AC.c \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib     -o plugins/fill-from-fasta.so filter.c version.c plugins/fill-from-fasta.c  \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/fill-tags.so version.c plugins/fill-tags.c \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/fixploidy.so ploidy.c regidx.c version.c plugins/fixploidy.c  \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/fixref.so version.c plugins/fixref.c \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/frameshifts.so version.c plugins/frameshifts.c \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/GTisec.so version.c plugins/GTisec.c  \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/GTsubset.so version.c plugins/GTsubset.c  \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/guess-ploidy.so version.c plugins/guess-ploidy.c \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/gvcfz.so version.c plugins/gvcfz.c \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/impute-info.so version.c plugins/impute-info.c \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/indel-stats.so version.c plugins/indel-stats.c \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/isecGT.so smpl_ilist.c version.c plugins/isecGT.c  \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/mendelian2.so version.c plugins/mendelian2.c \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/missing2ref.so version.c plugins/missing2ref.c \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/parental-origin.so version.c plugins/parental-origin.c \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/prune.so version.c plugins/prune.c \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/remove-overlaps.so version.c plugins/remove-overlaps.c \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/scatter.so version.c plugins/scatter.c \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/setGT.so version.c plugins/setGT.c \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/smpl-stats.so version.c plugins/smpl-stats.c \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/split.so version.c plugins/split.c \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/split-vep.so version.c plugins/split-vep.c \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/tag2tag.so version.c plugins/tag2tag.c \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/trio-dnm2.so version.c plugins/trio-dnm2.c \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/trio-stats.so version.c plugins/trio-stats.c \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/trio-switch-rate.so version.c plugins/trio-switch-rate.c \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/variant-distance.so version.c plugins/variant-distance.c \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/variantkey-hex.so version.c plugins/variantkey-hex.c \n",
            "mkdir -p -m 755 /usr/local/bin /usr/local/share/man/man1 /usr/local/libexec/bcftools\n",
            "install -p bcftools /usr/local/bin\n",
            "install -p misc/color-chrs.pl misc/gff2gff.py misc/guess-ploidy.py misc/plot-vcfstats misc/plot-roh.py misc/run-roh.pl misc/vcfutils.pl /usr/local/bin\n",
            "install -p -m 644 doc/bcftools.1 /usr/local/share/man/man1\n",
            "install -p plugins/*.so /usr/local/libexec/bcftools\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "Cloning into 'htslib'...\n",
            "Submodule 'htscodecs' (https://github.com/samtools/htscodecs.git) registered for path 'htscodecs'\n",
            "Cloning into '/content/htslib/htscodecs'...\n",
            "Cloning into 'bcftools'...\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Instalar udocker"
      ],
      "metadata": {
        "id": "_ylQgZ2hLPyW"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "#Código na célula será interpretado como comandos Bash.\n",
        "%%bash\n",
        "# Fonte: https://gist.github.com/mwufi/6718b30761cd109f9aff04c5144eb885\n",
        "#Instala o pacote Python.\n",
        "pip install udocker\n",
        "#Configura o ambiente do udocker para uso.\n",
        "udocker --allow-root install"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "0gWRUUpDLkKc",
        "outputId": "2ebcba32-1738-4d93-a34f-d363eba6c7b3"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting udocker\n",
            "  Downloading udocker-1.3.17-py2.py3-none-any.whl.metadata (37 kB)\n",
            "Downloading udocker-1.3.17-py2.py3-none-any.whl (119 kB)\n",
            "   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 119.6/119.6 kB 4.0 MB/s eta 0:00:00\n",
            "Installing collected packages: udocker\n",
            "Successfully installed udocker-1.3.17\n",
            "Info: creating repo: /root/.udocker\n",
            "Info: udocker command line interface 1.3.17\n",
            "Info: searching for udockertools >= 1.2.11\n",
            "Info: installing udockertools 1.2.11\n",
            "Info: installation of udockertools successful\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Download imagem do ensembl-vep"
      ],
      "metadata": {
        "id": "dVtcDKATLrO8"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!udocker --allow-root pull ensemblorg/ensembl-vep"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "x4Vn2HdkLruT",
        "outputId": "2591f316-771f-473e-95e3-63d63f44ff5d"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Info: downloading layer sha256:1d2b74402bf762db911b537c16df1307fd7fc70dcb5935090e54f582b6e65ef7\n",
            "Info: downloading layer sha256:bae88c57509dd07df870fb9d68b1d89169d86c0fd55872392ab7bb27afc3a68e\n",
            "Info: downloading layer sha256:e1a8a0853e7f6c89b83f4edc8893fa7ee92ae88befbec9393ddf814aeb8c0cb5\n",
            "Info: downloading layer sha256:cbe9c1736971e3a2b2d9b0e1c9215e2c7608cbfe201c8846fade02eeedd708c0\n",
            "Info: downloading layer sha256:4f4fb700ef54461cfa02571ae0db9a0dc1e0cdb5577484a6d75e68dc38e8acc1\n",
            "Info: downloading layer sha256:86919589c85a218116694e1d7d0301095ce49549d50428006fbdce5d36eeafb7\n",
            "Info: downloading layer sha256:a7164eead3c3d5c971068476d0f312893f1f9a64caac1e7119fcdef82f1ac3ef\n",
            "Info: downloading layer sha256:83270939064276cb55982870b14c481c7a430dc147e8b3b4a37d59ae5fc2ecb6\n",
            "Info: downloading layer sha256:8961add2e3310524992af3a74c7edb423d5ac08b57a43b975997a1fdf711280e\n",
            "Info: downloading layer sha256:b8fd7f0b70cdd1f385c4a9c3265887b1c001e9006379b74234c1231ceb148e2e\n",
            "Info: downloading layer sha256:08281b80bde300c78e91e16945fd608c67a78c49ac76d5bc77cb9b0a0e2af264\n",
            "Info: downloading layer sha256:4f4fb700ef54461cfa02571ae0db9a0dc1e0cdb5577484a6d75e68dc38e8acc1\n",
            "Info: downloading layer sha256:9dfa485aac3b90ce1eb35521f3030981a694efd2af7e0cee75d294c13d91872d\n",
            "Info: downloading layer sha256:c5958e5f73a43b44ec13dd3ef05fdfa46bbe7e7078cae71b6a88ed1ef0c5b7b1\n",
            "Info: downloading layer sha256:4f4fb700ef54461cfa02571ae0db9a0dc1e0cdb5577484a6d75e68dc38e8acc1\n",
            "Info: downloading layer sha256:b64f100902e8c89fad94f897028be49672a099da496c16805e07ca74ef7c11cf\n",
            "Info: downloading layer sha256:4f4fb700ef54461cfa02571ae0db9a0dc1e0cdb5577484a6d75e68dc38e8acc1\n",
            "Info: downloading layer sha256:ee01cd33bc20b41c75078039eefaaa041e9f2a4bca4299b161507f6e4d230bcf\n",
            "Info: downloading layer sha256:4f4fb700ef54461cfa02571ae0db9a0dc1e0cdb5577484a6d75e68dc38e8acc1\n",
            "Info: downloading layer sha256:3661ff44f8bb1dc1410aee3d3c2bc212aa97af85a8757298949a416457f26233\n",
            "Info: downloading layer sha256:e1610fc68bf92b07a6151af3109e953945751cd371dd25e1789d1ce6be39b564\n",
            "Info: downloading layer sha256:d0cfc1185bf49fd1daabdf82b48f53d781737125bf926cf5e4c13a6d148ff541\n",
            "Info: downloading layer sha256:6414378b647780fee8fd903ddb9541d134a1947ce092d08bdeb23a54cb3684ac\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#Executa um script chamado vep-gc.sh, localizado no diretório lmabrasil-hg38, com dois argumentos: WP017 (identificador de amostra) e Clonal_Hematopoiesis_driver_genes.txt (lista de genes).\n",
        "!sh lmabrasil-hg38/vep-gc.sh WP017 Clonal_Hematopoiesis_driver_genes.txt"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "9_3PvnbUM5tc",
        "outputId": "f5d86478-329d-4339-95bb-9c6dacc75c6c"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            " \n",
            " ****************************************************************************** \n",
            " *                                                                            * \n",
            " *               STARTING 0030e2d7-2db2-356a-9885-8d910bb56706                * \n",
            " *                                                                            * \n",
            " ****************************************************************************** \n",
            " executing: filter_vep\n",
            "\n",
            "-------------------- WARNING ----------------------\n",
            "MSG: The script 'filter_vep' couldn't find the VEP annotations INFO field CSQ in 181 lines of the input file\n",
            "FILE: src/ensembl-vep/filter_vep LINE: 345\n",
            "CALLED BY: src/ensembl-vep/filter_vep  LINE: 55\n",
            "Date (localtime)    = Sat Dec 21 13:03:28 2024\n",
            "Ensembl API version = 113\n",
            "---------------------------------------------------\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#Importa a biblioteca Pandas (dados tabulares).\n",
        "import pandas as pd\n",
        "#Configuração de exibição do Pandas para exibir todas as colunas, sem limites.\n",
        "pd.set_option('display.max_columns', None)\n",
        "#Configuração de exibição do Pandas para exibir todo o conteúdo de uma célula.\n",
        "pd.set_option('display.max_colwidth', None)\n",
        "#Lê um arquivo tabular e o carrega em um DataFrame.\n",
        "df_WP017 = pd.read_csv('/content/lmabrasil-hg38/vep_output/liftOver_WP017_hg19ToHg38.vep.filter.tsv',sep='\\t',index_col=False)\n",
        "#Exibe o DataFrame df_WP017 carregado do arquivo TSV.\n",
        "df_WP017"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 372
        },
        "id": "5EKf9y6rRhMZ",
        "outputId": "366bcfcb-7bcd-48e7-be42-ff0c7289f772"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "   CHROM        POS                                                    REF  \\\n",
              "0   chr1  114716127                                                      C   \n",
              "1  chr16   67616834                                                      G   \n",
              "2  chr17    7669662                                                      T   \n",
              "3  chr19   12943750  AGCAGAGGCTTAAGGAGGAGGAAGAAGACAAGAAACGCAAAGAGGAGGAGGAG   \n",
              "4  chr20   32434638                                                      A   \n",
              "5  chr21   43094667                                                      T   \n",
              "6   chr8  100712766                                                      C   \n",
              "7   chr8  100715522                                                      T   \n",
              "\n",
              "  ALT                  Location  SYMBOL         Consequence      Feature  \\\n",
              "0   T            chr1:114716127    NRAS    missense_variant  NM_002524.5   \n",
              "1   A            chr16:67616834    CTCF    missense_variant  NM_006565.4   \n",
              "2   G             chr17:7669662    TP53    missense_variant  NM_000546.6   \n",
              "3   A   chr19:12943751-12943802    CALR  frameshift_variant  NM_004343.4   \n",
              "4  AG   chr20:32434638-32434639   ASXL1  frameshift_variant  NM_015338.6   \n",
              "5   G            chr21:43094667   U2AF1    missense_variant  NM_006758.3   \n",
              "6  CA  chr8:100712766-100712767  PABPC1  frameshift_variant  NM_002568.4   \n",
              "7   C            chr8:100715522  PABPC1    missense_variant  NM_002568.4   \n",
              "\n",
              "          MANE_SELECT         BIOTYPE                       HGVSc  \\\n",
              "0   ENST00000369535.5  protein_coding         NM_002524.5:c.34G>A   \n",
              "1  ENST00000264010.10  protein_coding       NM_006565.4:c.1042G>A   \n",
              "2   ENST00000269305.9  protein_coding       NM_000546.6:c.1129A>C   \n",
              "3  ENST00000316448.10  protein_coding  NM_004343.4:c.1099_1150del   \n",
              "4  ENST00000375687.10  protein_coding       NM_015338.6:c.1934dup   \n",
              "5   ENST00000291552.9  protein_coding        NM_006758.3:c.470A>C   \n",
              "6  ENST00000318607.10  protein_coding   NM_002568.4:c.761_762insT   \n",
              "7  ENST00000318607.10  protein_coding        NM_002568.4:c.583A>G   \n",
              "\n",
              "                            HGVSp   EXON INTRON VARIANT_CLASS  \\\n",
              "0          NP_002515.1:p.Gly12Ser    2/7      .           SNV   \n",
              "1         NP_006556.1:p.Glu348Lys   5/12      .           SNV   \n",
              "2         NP_000537.3:p.Thr377Pro  11/11      .           SNV   \n",
              "3  NP_004334.1:p.Leu367ThrfsTer46    9/9      .      deletion   \n",
              "4  NP_056153.2:p.Gly646TrpfsTer12  13/13      .     insertion   \n",
              "5         NP_006749.1:p.Gln157Pro    6/8      .           SNV   \n",
              "6  NP_002559.2:p.Lys254AsnfsTer24   6/15      .     insertion   \n",
              "7         NP_002559.2:p.Ile195Val   4/15      .           SNV   \n",
              "\n",
              "                               SIFT                  PolyPhen gnomADg_AF  \\\n",
              "0  deleterious_low_confidence(0.02)                         .          .   \n",
              "1                    deleterious(0)  possibly_damaging(0.631)          .   \n",
              "2                   tolerated(0.42)                         .  5.302e-05   \n",
              "3                                 .                         .  1.971e-05   \n",
              "4                                 .                         .  0.0006444   \n",
              "5     deleterious_low_confidence(0)  possibly_damaging(0.541)  7.313e-05   \n",
              "6                                 .                         .          0   \n",
              "7  deleterious_low_confidence(0.04)                         .  1.973e-05   \n",
              "\n",
              "      MAX_AF    IMPACT                        CLIN_SIG      SOMATIC  \\\n",
              "0          .  MODERATE    pathogenic&likely_pathogenic  0&0&0&1&1&1   \n",
              "1          .  MODERATE                               .            1   \n",
              "2  0.0004959  MODERATE          uncertain_significance          0&1   \n",
              "3   6.55e-05      HIGH                      pathogenic            .   \n",
              "4   0.001739      HIGH    likely_pathogenic&pathogenic          0&1   \n",
              "5  0.0001146  MODERATE  not_provided&likely_pathogenic        0&1&1   \n",
              "6    0.00301      HIGH                               .          0&1   \n",
              "7   0.000346  MODERATE                               .          0&1   \n",
              "\n",
              "                                                     Existing_variation  \\\n",
              "0  rs121913250&CM136798&CM177295&COSV54736487&COSV54736621&COSV54736940   \n",
              "1                                                          COSV50461493   \n",
              "2                                              rs774269719&COSV52716766   \n",
              "3                                                          rs1555760738   \n",
              "4                                              rs750318549&COSV60102510   \n",
              "5                                 rs371246226&COSV52341120&COSV52341147   \n",
              "6                                              rs140822921&COSV59402392   \n",
              "7                                              rs200479111&COSV59397682   \n",
              "\n",
              "                                            FILTER TumorID   GT   DP       AD  \\\n",
              "0                                             PASS   WP017  0/1  232  117,115   \n",
              "1                                             PASS   WP017  0/1  224  101,123   \n",
              "2  base_qual;haplotype;normal_artifact;strand_bias   WP017  0|1  119   101,18   \n",
              "3                                             PASS   WP017  0/1  102    62,40   \n",
              "4                                             PASS   WP017  0/1  118    59,59   \n",
              "5                                             PASS   WP017  0/1  163    89,74   \n",
              "6       clustered_events;haplotype;normal_artifact   WP017  0|1  112    99,13   \n",
              "7                 clustered_events;normal_artifact   WP017  0/1   79    68,11   \n",
              "\n",
              "      AF NormalID  NGT  NDP    NAD    NAF  \n",
              "0  0.504    WP018  0/0  155  152,3  0.028  \n",
              "1  0.538    WP018  0/0   68   66,2  0.044  \n",
              "2  0.112    WP018  0|0   60   55,5  0.049  \n",
              "3  0.416    WP018  0/0   50   50,0  0.022  \n",
              "4  0.492    WP018  0/0   62   60,2  0.045  \n",
              "5  0.457    WP018  0/0  107  105,2  0.029  \n",
              "6  0.123    WP018  0|0   53   50,3  0.070  \n",
              "7  0.154    WP018  0/0   50   49,1  0.041  "
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-78fc4a13-7599-49f4-a992-07a867227932\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>CHROM</th>\n",
              "      <th>POS</th>\n",
              "      <th>REF</th>\n",
              "      <th>ALT</th>\n",
              "      <th>Location</th>\n",
              "      <th>SYMBOL</th>\n",
              "      <th>Consequence</th>\n",
              "      <th>Feature</th>\n",
              "      <th>MANE_SELECT</th>\n",
              "      <th>BIOTYPE</th>\n",
              "      <th>HGVSc</th>\n",
              "      <th>HGVSp</th>\n",
              "      <th>EXON</th>\n",
              "      <th>INTRON</th>\n",
              "      <th>VARIANT_CLASS</th>\n",
              "      <th>SIFT</th>\n",
              "      <th>PolyPhen</th>\n",
              "      <th>gnomADg_AF</th>\n",
              "      <th>MAX_AF</th>\n",
              "      <th>IMPACT</th>\n",
              "      <th>CLIN_SIG</th>\n",
              "      <th>SOMATIC</th>\n",
              "      <th>Existing_variation</th>\n",
              "      <th>FILTER</th>\n",
              "      <th>TumorID</th>\n",
              "      <th>GT</th>\n",
              "      <th>DP</th>\n",
              "      <th>AD</th>\n",
              "      <th>AF</th>\n",
              "      <th>NormalID</th>\n",
              "      <th>NGT</th>\n",
              "      <th>NDP</th>\n",
              "      <th>NAD</th>\n",
              "      <th>NAF</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>chr1</td>\n",
              "      <td>114716127</td>\n",
              "      <td>C</td>\n",
              "      <td>T</td>\n",
              "      <td>chr1:114716127</td>\n",
              "      <td>NRAS</td>\n",
              "      <td>missense_variant</td>\n",
              "      <td>NM_002524.5</td>\n",
              "      <td>ENST00000369535.5</td>\n",
              "      <td>protein_coding</td>\n",
              "      <td>NM_002524.5:c.34G&gt;A</td>\n",
              "      <td>NP_002515.1:p.Gly12Ser</td>\n",
              "      <td>2/7</td>\n",
              "      <td>.</td>\n",
              "      <td>SNV</td>\n",
              "      <td>deleterious_low_confidence(0.02)</td>\n",
              "      <td>.</td>\n",
              "      <td>.</td>\n",
              "      <td>.</td>\n",
              "      <td>MODERATE</td>\n",
              "      <td>pathogenic&amp;likely_pathogenic</td>\n",
              "      <td>0&amp;0&amp;0&amp;1&amp;1&amp;1</td>\n",
              "      <td>rs121913250&amp;CM136798&amp;CM177295&amp;COSV54736487&amp;COSV54736621&amp;COSV54736940</td>\n",
              "      <td>PASS</td>\n",
              "      <td>WP017</td>\n",
              "      <td>0/1</td>\n",
              "      <td>232</td>\n",
              "      <td>117,115</td>\n",
              "      <td>0.504</td>\n",
              "      <td>WP018</td>\n",
              "      <td>0/0</td>\n",
              "      <td>155</td>\n",
              "      <td>152,3</td>\n",
              "      <td>0.028</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>chr16</td>\n",
              "      <td>67616834</td>\n",
              "      <td>G</td>\n",
              "      <td>A</td>\n",
              "      <td>chr16:67616834</td>\n",
              "      <td>CTCF</td>\n",
              "      <td>missense_variant</td>\n",
              "      <td>NM_006565.4</td>\n",
              "      <td>ENST00000264010.10</td>\n",
              "      <td>protein_coding</td>\n",
              "      <td>NM_006565.4:c.1042G&gt;A</td>\n",
              "      <td>NP_006556.1:p.Glu348Lys</td>\n",
              "      <td>5/12</td>\n",
              "      <td>.</td>\n",
              "      <td>SNV</td>\n",
              "      <td>deleterious(0)</td>\n",
              "      <td>possibly_damaging(0.631)</td>\n",
              "      <td>.</td>\n",
              "      <td>.</td>\n",
              "      <td>MODERATE</td>\n",
              "      <td>.</td>\n",
              "      <td>1</td>\n",
              "      <td>COSV50461493</td>\n",
              "      <td>PASS</td>\n",
              "      <td>WP017</td>\n",
              "      <td>0/1</td>\n",
              "      <td>224</td>\n",
              "      <td>101,123</td>\n",
              "      <td>0.538</td>\n",
              "      <td>WP018</td>\n",
              "      <td>0/0</td>\n",
              "      <td>68</td>\n",
              "      <td>66,2</td>\n",
              "      <td>0.044</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>chr17</td>\n",
              "      <td>7669662</td>\n",
              "      <td>T</td>\n",
              "      <td>G</td>\n",
              "      <td>chr17:7669662</td>\n",
              "      <td>TP53</td>\n",
              "      <td>missense_variant</td>\n",
              "      <td>NM_000546.6</td>\n",
              "      <td>ENST00000269305.9</td>\n",
              "      <td>protein_coding</td>\n",
              "      <td>NM_000546.6:c.1129A&gt;C</td>\n",
              "      <td>NP_000537.3:p.Thr377Pro</td>\n",
              "      <td>11/11</td>\n",
              "      <td>.</td>\n",
              "      <td>SNV</td>\n",
              "      <td>tolerated(0.42)</td>\n",
              "      <td>.</td>\n",
              "      <td>5.302e-05</td>\n",
              "      <td>0.0004959</td>\n",
              "      <td>MODERATE</td>\n",
              "      <td>uncertain_significance</td>\n",
              "      <td>0&amp;1</td>\n",
              "      <td>rs774269719&amp;COSV52716766</td>\n",
              "      <td>base_qual;haplotype;normal_artifact;strand_bias</td>\n",
              "      <td>WP017</td>\n",
              "      <td>0|1</td>\n",
              "      <td>119</td>\n",
              "      <td>101,18</td>\n",
              "      <td>0.112</td>\n",
              "      <td>WP018</td>\n",
              "      <td>0|0</td>\n",
              "      <td>60</td>\n",
              "      <td>55,5</td>\n",
              "      <td>0.049</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>chr19</td>\n",
              "      <td>12943750</td>\n",
              "      <td>AGCAGAGGCTTAAGGAGGAGGAAGAAGACAAGAAACGCAAAGAGGAGGAGGAG</td>\n",
              "      <td>A</td>\n",
              "      <td>chr19:12943751-12943802</td>\n",
              "      <td>CALR</td>\n",
              "      <td>frameshift_variant</td>\n",
              "      <td>NM_004343.4</td>\n",
              "      <td>ENST00000316448.10</td>\n",
              "      <td>protein_coding</td>\n",
              "      <td>NM_004343.4:c.1099_1150del</td>\n",
              "      <td>NP_004334.1:p.Leu367ThrfsTer46</td>\n",
              "      <td>9/9</td>\n",
              "      <td>.</td>\n",
              "      <td>deletion</td>\n",
              "      <td>.</td>\n",
              "      <td>.</td>\n",
              "      <td>1.971e-05</td>\n",
              "      <td>6.55e-05</td>\n",
              "      <td>HIGH</td>\n",
              "      <td>pathogenic</td>\n",
              "      <td>.</td>\n",
              "      <td>rs1555760738</td>\n",
              "      <td>PASS</td>\n",
              "      <td>WP017</td>\n",
              "      <td>0/1</td>\n",
              "      <td>102</td>\n",
              "      <td>62,40</td>\n",
              "      <td>0.416</td>\n",
              "      <td>WP018</td>\n",
              "      <td>0/0</td>\n",
              "      <td>50</td>\n",
              "      <td>50,0</td>\n",
              "      <td>0.022</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>chr20</td>\n",
              "      <td>32434638</td>\n",
              "      <td>A</td>\n",
              "      <td>AG</td>\n",
              "      <td>chr20:32434638-32434639</td>\n",
              "      <td>ASXL1</td>\n",
              "      <td>frameshift_variant</td>\n",
              "      <td>NM_015338.6</td>\n",
              "      <td>ENST00000375687.10</td>\n",
              "      <td>protein_coding</td>\n",
              "      <td>NM_015338.6:c.1934dup</td>\n",
              "      <td>NP_056153.2:p.Gly646TrpfsTer12</td>\n",
              "      <td>13/13</td>\n",
              "      <td>.</td>\n",
              "      <td>insertion</td>\n",
              "      <td>.</td>\n",
              "      <td>.</td>\n",
              "      <td>0.0006444</td>\n",
              "      <td>0.001739</td>\n",
              "      <td>HIGH</td>\n",
              "      <td>likely_pathogenic&amp;pathogenic</td>\n",
              "      <td>0&amp;1</td>\n",
              "      <td>rs750318549&amp;COSV60102510</td>\n",
              "      <td>PASS</td>\n",
              "      <td>WP017</td>\n",
              "      <td>0/1</td>\n",
              "      <td>118</td>\n",
              "      <td>59,59</td>\n",
              "      <td>0.492</td>\n",
              "      <td>WP018</td>\n",
              "      <td>0/0</td>\n",
              "      <td>62</td>\n",
              "      <td>60,2</td>\n",
              "      <td>0.045</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5</th>\n",
              "      <td>chr21</td>\n",
              "      <td>43094667</td>\n",
              "      <td>T</td>\n",
              "      <td>G</td>\n",
              "      <td>chr21:43094667</td>\n",
              "      <td>U2AF1</td>\n",
              "      <td>missense_variant</td>\n",
              "      <td>NM_006758.3</td>\n",
              "      <td>ENST00000291552.9</td>\n",
              "      <td>protein_coding</td>\n",
              "      <td>NM_006758.3:c.470A&gt;C</td>\n",
              "      <td>NP_006749.1:p.Gln157Pro</td>\n",
              "      <td>6/8</td>\n",
              "      <td>.</td>\n",
              "      <td>SNV</td>\n",
              "      <td>deleterious_low_confidence(0)</td>\n",
              "      <td>possibly_damaging(0.541)</td>\n",
              "      <td>7.313e-05</td>\n",
              "      <td>0.0001146</td>\n",
              "      <td>MODERATE</td>\n",
              "      <td>not_provided&amp;likely_pathogenic</td>\n",
              "      <td>0&amp;1&amp;1</td>\n",
              "      <td>rs371246226&amp;COSV52341120&amp;COSV52341147</td>\n",
              "      <td>PASS</td>\n",
              "      <td>WP017</td>\n",
              "      <td>0/1</td>\n",
              "      <td>163</td>\n",
              "      <td>89,74</td>\n",
              "      <td>0.457</td>\n",
              "      <td>WP018</td>\n",
              "      <td>0/0</td>\n",
              "      <td>107</td>\n",
              "      <td>105,2</td>\n",
              "      <td>0.029</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>6</th>\n",
              "      <td>chr8</td>\n",
              "      <td>100712766</td>\n",
              "      <td>C</td>\n",
              "      <td>CA</td>\n",
              "      <td>chr8:100712766-100712767</td>\n",
              "      <td>PABPC1</td>\n",
              "      <td>frameshift_variant</td>\n",
              "      <td>NM_002568.4</td>\n",
              "      <td>ENST00000318607.10</td>\n",
              "      <td>protein_coding</td>\n",
              "      <td>NM_002568.4:c.761_762insT</td>\n",
              "      <td>NP_002559.2:p.Lys254AsnfsTer24</td>\n",
              "      <td>6/15</td>\n",
              "      <td>.</td>\n",
              "      <td>insertion</td>\n",
              "      <td>.</td>\n",
              "      <td>.</td>\n",
              "      <td>0</td>\n",
              "      <td>0.00301</td>\n",
              "      <td>HIGH</td>\n",
              "      <td>.</td>\n",
              "      <td>0&amp;1</td>\n",
              "      <td>rs140822921&amp;COSV59402392</td>\n",
              "      <td>clustered_events;haplotype;normal_artifact</td>\n",
              "      <td>WP017</td>\n",
              "      <td>0|1</td>\n",
              "      <td>112</td>\n",
              "      <td>99,13</td>\n",
              "      <td>0.123</td>\n",
              "      <td>WP018</td>\n",
              "      <td>0|0</td>\n",
              "      <td>53</td>\n",
              "      <td>50,3</td>\n",
              "      <td>0.070</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>7</th>\n",
              "      <td>chr8</td>\n",
              "      <td>100715522</td>\n",
              "      <td>T</td>\n",
              "      <td>C</td>\n",
              "      <td>chr8:100715522</td>\n",
              "      <td>PABPC1</td>\n",
              "      <td>missense_variant</td>\n",
              "      <td>NM_002568.4</td>\n",
              "      <td>ENST00000318607.10</td>\n",
              "      <td>protein_coding</td>\n",
              "      <td>NM_002568.4:c.583A&gt;G</td>\n",
              "      <td>NP_002559.2:p.Ile195Val</td>\n",
              "      <td>4/15</td>\n",
              "      <td>.</td>\n",
              "      <td>SNV</td>\n",
              "      <td>deleterious_low_confidence(0.04)</td>\n",
              "      <td>.</td>\n",
              "      <td>1.973e-05</td>\n",
              "      <td>0.000346</td>\n",
              "      <td>MODERATE</td>\n",
              "      <td>.</td>\n",
              "      <td>0&amp;1</td>\n",
              "      <td>rs200479111&amp;COSV59397682</td>\n",
              "      <td>clustered_events;normal_artifact</td>\n",
              "      <td>WP017</td>\n",
              "      <td>0/1</td>\n",
              "      <td>79</td>\n",
              "      <td>68,11</td>\n",
              "      <td>0.154</td>\n",
              "      <td>WP018</td>\n",
              "      <td>0/0</td>\n",
              "      <td>50</td>\n",
              "      <td>49,1</td>\n",
              "      <td>0.041</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-78fc4a13-7599-49f4-a992-07a867227932')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-78fc4a13-7599-49f4-a992-07a867227932 button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-78fc4a13-7599-49f4-a992-07a867227932');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "<div id=\"df-61679a4e-2a3e-46ac-86c6-a9d168845e0f\">\n",
              "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-61679a4e-2a3e-46ac-86c6-a9d168845e0f')\"\n",
              "            title=\"Suggest charts\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "  </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "  <script>\n",
              "    async function quickchart(key) {\n",
              "      const quickchartButtonEl =\n",
              "        document.querySelector('#' + key + ' button');\n",
              "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "      try {\n",
              "        const charts = await google.colab.kernel.invokeFunction(\n",
              "            'suggestCharts', [key], {});\n",
              "      } catch (error) {\n",
              "        console.error('Error during call to suggestCharts:', error);\n",
              "      }\n",
              "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "    }\n",
              "    (() => {\n",
              "      let quickchartButtonEl =\n",
              "        document.querySelector('#df-61679a4e-2a3e-46ac-86c6-a9d168845e0f button');\n",
              "      quickchartButtonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "    })();\n",
              "  </script>\n",
              "</div>\n",
              "\n",
              "  <div id=\"id_4b58a76a-6e0e-4e59-a7da-ded8c2d88962\">\n",
              "    <style>\n",
              "      .colab-df-generate {\n",
              "        background-color: #E8F0FE;\n",
              "        border: none;\n",
              "        border-radius: 50%;\n",
              "        cursor: pointer;\n",
              "        display: none;\n",
              "        fill: #1967D2;\n",
              "        height: 32px;\n",
              "        padding: 0 0 0 0;\n",
              "        width: 32px;\n",
              "      }\n",
              "\n",
              "      .colab-df-generate:hover {\n",
              "        background-color: #E2EBFA;\n",
              "        box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "        fill: #174EA6;\n",
              "      }\n",
              "\n",
              "      [theme=dark] .colab-df-generate {\n",
              "        background-color: #3B4455;\n",
              "        fill: #D2E3FC;\n",
              "      }\n",
              "\n",
              "      [theme=dark] .colab-df-generate:hover {\n",
              "        background-color: #434B5C;\n",
              "        box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "        filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "        fill: #FFFFFF;\n",
              "      }\n",
              "    </style>\n",
              "    <button class=\"colab-df-generate\" onclick=\"generateWithVariable('df_WP017')\"\n",
              "            title=\"Generate code using this dataframe.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "       width=\"24px\">\n",
              "    <path d=\"M7,19H8.4L18.45,9,17,7.55,7,17.6ZM5,21V16.75L18.45,3.32a2,2,0,0,1,2.83,0l1.4,1.43a1.91,1.91,0,0,1,.58,1.4,1.91,1.91,0,0,1-.58,1.4L9.25,21ZM18.45,9,17,7.55Zm-12,3A5.31,5.31,0,0,0,4.9,8.1,5.31,5.31,0,0,0,1,6.5,5.31,5.31,0,0,0,4.9,4.9,5.31,5.31,0,0,0,6.5,1,5.31,5.31,0,0,0,8.1,4.9,5.31,5.31,0,0,0,12,6.5,5.46,5.46,0,0,0,6.5,12Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "    <script>\n",
              "      (() => {\n",
              "      const buttonEl =\n",
              "        document.querySelector('#id_4b58a76a-6e0e-4e59-a7da-ded8c2d88962 button.colab-df-generate');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      buttonEl.onclick = () => {\n",
              "        google.colab.notebook.generateWithVariable('df_WP017');\n",
              "      }\n",
              "      })();\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe",
              "variable_name": "df_WP017"
            }
          },
          "metadata": {},
          "execution_count": 8
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#Executa um script chamado vep-gc.sh, localizado no diretório lmabrasil-hg38, com dois argumentos: WP058 (identificador de amostra) e Clonal_Hematopoiesis_driver_genes.txt (lista de genes).\n",
        "!sh lmabrasil-hg38/vep-gc.sh WP058 Clonal_Hematopoiesis_driver_genes.txt"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "gC-_KY2NTPKe",
        "outputId": "b121864e-cdce-4815-a64b-0b3f0d09705f"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            " \n",
            " ****************************************************************************** \n",
            " *                                                                            * \n",
            " *               STARTING bf1fce2d-fb8a-31a0-8ca8-6950a5554a0b                * \n",
            " *                                                                            * \n",
            " ****************************************************************************** \n",
            " executing: filter_vep\n",
            "\n",
            "-------------------- WARNING ----------------------\n",
            "MSG: The script 'filter_vep' couldn't find the VEP annotations INFO field CSQ in 116 lines of the input file\n",
            "FILE: src/ensembl-vep/filter_vep LINE: 345\n",
            "CALLED BY: src/ensembl-vep/filter_vep  LINE: 55\n",
            "Date (localtime)    = Sat Dec 21 13:05:09 2024\n",
            "Ensembl API version = 113\n",
            "---------------------------------------------------\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#Carrega para um DataFrame do Pandas os dados genômicos anotados da amostra WP058.\n",
        "df_WP058 = pd.read_csv('/content/lmabrasil-hg38/vep_output/liftOver_WP058_hg19ToHg38.vep.filter.tsv',sep='\\t',index_col=False, engine= 'python')\n",
        "df_WP058"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 166
        },
        "id": "wrNbx1PITax-",
        "outputId": "a2e99b00-d5fd-4233-971b-06d64d1d08d2"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "   CHROM       POS                                                    REF ALT  \\\n",
              "0  chr19  12943750  AGCAGAGGCTTAAGGAGGAGGAAGAAGACAAGAAACGCAAAGAGGAGGAGGAG   A   \n",
              "1  chr20  32434638                                                      A  AG   \n",
              "\n",
              "                  Location SYMBOL         Consequence      Feature  \\\n",
              "0  chr19:12943751-12943802   CALR  frameshift_variant  NM_004343.4   \n",
              "1  chr20:32434638-32434639  ASXL1  frameshift_variant  NM_015338.6   \n",
              "\n",
              "          MANE_SELECT         BIOTYPE                       HGVSc  \\\n",
              "0  ENST00000316448.10  protein_coding  NM_004343.4:c.1099_1150del   \n",
              "1  ENST00000375687.10  protein_coding       NM_015338.6:c.1934dup   \n",
              "\n",
              "                            HGVSp   EXON INTRON VARIANT_CLASS SIFT PolyPhen  \\\n",
              "0  NP_004334.1:p.Leu367ThrfsTer46    9/9      .      deletion    .        .   \n",
              "1  NP_056153.2:p.Gly646TrpfsTer12  13/13      .     insertion    .        .   \n",
              "\n",
              "   gnomADg_AF    MAX_AF IMPACT                      CLIN_SIG SOMATIC  \\\n",
              "0    0.000020  0.000066   HIGH                    pathogenic       .   \n",
              "1    0.000644  0.001739   HIGH  likely_pathogenic&pathogenic     0&1   \n",
              "\n",
              "         Existing_variation                    FILTER TumorID   GT  DP     AD  \\\n",
              "0              rs1555760738                  map_qual   WP058  0/1  39  28,11   \n",
              "1  rs750318549&COSV60102510  germline;normal_artifact   WP058  0/1  49  25,24   \n",
              "\n",
              "      AF NormalID  NGT  NDP   NAD    NAF  \n",
              "0  0.337    WP059  0/0   31  31,0  0.037  \n",
              "1  0.440    WP059  0/0   38  35,3  0.078  "
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-8c57fce7-6f54-41e9-930e-073db03ca0d6\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>CHROM</th>\n",
              "      <th>POS</th>\n",
              "      <th>REF</th>\n",
              "      <th>ALT</th>\n",
              "      <th>Location</th>\n",
              "      <th>SYMBOL</th>\n",
              "      <th>Consequence</th>\n",
              "      <th>Feature</th>\n",
              "      <th>MANE_SELECT</th>\n",
              "      <th>BIOTYPE</th>\n",
              "      <th>HGVSc</th>\n",
              "      <th>HGVSp</th>\n",
              "      <th>EXON</th>\n",
              "      <th>INTRON</th>\n",
              "      <th>VARIANT_CLASS</th>\n",
              "      <th>SIFT</th>\n",
              "      <th>PolyPhen</th>\n",
              "      <th>gnomADg_AF</th>\n",
              "      <th>MAX_AF</th>\n",
              "      <th>IMPACT</th>\n",
              "      <th>CLIN_SIG</th>\n",
              "      <th>SOMATIC</th>\n",
              "      <th>Existing_variation</th>\n",
              "      <th>FILTER</th>\n",
              "      <th>TumorID</th>\n",
              "      <th>GT</th>\n",
              "      <th>DP</th>\n",
              "      <th>AD</th>\n",
              "      <th>AF</th>\n",
              "      <th>NormalID</th>\n",
              "      <th>NGT</th>\n",
              "      <th>NDP</th>\n",
              "      <th>NAD</th>\n",
              "      <th>NAF</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>chr19</td>\n",
              "      <td>12943750</td>\n",
              "      <td>AGCAGAGGCTTAAGGAGGAGGAAGAAGACAAGAAACGCAAAGAGGAGGAGGAG</td>\n",
              "      <td>A</td>\n",
              "      <td>chr19:12943751-12943802</td>\n",
              "      <td>CALR</td>\n",
              "      <td>frameshift_variant</td>\n",
              "      <td>NM_004343.4</td>\n",
              "      <td>ENST00000316448.10</td>\n",
              "      <td>protein_coding</td>\n",
              "      <td>NM_004343.4:c.1099_1150del</td>\n",
              "      <td>NP_004334.1:p.Leu367ThrfsTer46</td>\n",
              "      <td>9/9</td>\n",
              "      <td>.</td>\n",
              "      <td>deletion</td>\n",
              "      <td>.</td>\n",
              "      <td>.</td>\n",
              "      <td>0.000020</td>\n",
              "      <td>0.000066</td>\n",
              "      <td>HIGH</td>\n",
              "      <td>pathogenic</td>\n",
              "      <td>.</td>\n",
              "      <td>rs1555760738</td>\n",
              "      <td>map_qual</td>\n",
              "      <td>WP058</td>\n",
              "      <td>0/1</td>\n",
              "      <td>39</td>\n",
              "      <td>28,11</td>\n",
              "      <td>0.337</td>\n",
              "      <td>WP059</td>\n",
              "      <td>0/0</td>\n",
              "      <td>31</td>\n",
              "      <td>31,0</td>\n",
              "      <td>0.037</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>chr20</td>\n",
              "      <td>32434638</td>\n",
              "      <td>A</td>\n",
              "      <td>AG</td>\n",
              "      <td>chr20:32434638-32434639</td>\n",
              "      <td>ASXL1</td>\n",
              "      <td>frameshift_variant</td>\n",
              "      <td>NM_015338.6</td>\n",
              "      <td>ENST00000375687.10</td>\n",
              "      <td>protein_coding</td>\n",
              "      <td>NM_015338.6:c.1934dup</td>\n",
              "      <td>NP_056153.2:p.Gly646TrpfsTer12</td>\n",
              "      <td>13/13</td>\n",
              "      <td>.</td>\n",
              "      <td>insertion</td>\n",
              "      <td>.</td>\n",
              "      <td>.</td>\n",
              "      <td>0.000644</td>\n",
              "      <td>0.001739</td>\n",
              "      <td>HIGH</td>\n",
              "      <td>likely_pathogenic&amp;pathogenic</td>\n",
              "      <td>0&amp;1</td>\n",
              "      <td>rs750318549&amp;COSV60102510</td>\n",
              "      <td>germline;normal_artifact</td>\n",
              "      <td>WP058</td>\n",
              "      <td>0/1</td>\n",
              "      <td>49</td>\n",
              "      <td>25,24</td>\n",
              "      <td>0.440</td>\n",
              "      <td>WP059</td>\n",
              "      <td>0/0</td>\n",
              "      <td>38</td>\n",
              "      <td>35,3</td>\n",
              "      <td>0.078</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-8c57fce7-6f54-41e9-930e-073db03ca0d6')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-8c57fce7-6f54-41e9-930e-073db03ca0d6 button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-8c57fce7-6f54-41e9-930e-073db03ca0d6');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "<div id=\"df-44ffef34-0e1a-46e4-a305-7ebca3bdc44e\">\n",
              "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-44ffef34-0e1a-46e4-a305-7ebca3bdc44e')\"\n",
              "            title=\"Suggest charts\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "  </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "  <script>\n",
              "    async function quickchart(key) {\n",
              "      const quickchartButtonEl =\n",
              "        document.querySelector('#' + key + ' button');\n",
              "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "      try {\n",
              "        const charts = await google.colab.kernel.invokeFunction(\n",
              "            'suggestCharts', [key], {});\n",
              "      } catch (error) {\n",
              "        console.error('Error during call to suggestCharts:', error);\n",
              "      }\n",
              "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "    }\n",
              "    (() => {\n",
              "      let quickchartButtonEl =\n",
              "        document.querySelector('#df-44ffef34-0e1a-46e4-a305-7ebca3bdc44e button');\n",
              "      quickchartButtonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "    })();\n",
              "  </script>\n",
              "</div>\n",
              "\n",
              "  <div id=\"id_2fdbce8e-bfee-42db-92b3-204ce2c153bc\">\n",
              "    <style>\n",
              "      .colab-df-generate {\n",
              "        background-color: #E8F0FE;\n",
              "        border: none;\n",
              "        border-radius: 50%;\n",
              "        cursor: pointer;\n",
              "        display: none;\n",
              "        fill: #1967D2;\n",
              "        height: 32px;\n",
              "        padding: 0 0 0 0;\n",
              "        width: 32px;\n",
              "      }\n",
              "\n",
              "      .colab-df-generate:hover {\n",
              "        background-color: #E2EBFA;\n",
              "        box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "        fill: #174EA6;\n",
              "      }\n",
              "\n",
              "      [theme=dark] .colab-df-generate {\n",
              "        background-color: #3B4455;\n",
              "        fill: #D2E3FC;\n",
              "      }\n",
              "\n",
              "      [theme=dark] .colab-df-generate:hover {\n",
              "        background-color: #434B5C;\n",
              "        box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "        filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "        fill: #FFFFFF;\n",
              "      }\n",
              "    </style>\n",
              "    <button class=\"colab-df-generate\" onclick=\"generateWithVariable('df_WP058')\"\n",
              "            title=\"Generate code using this dataframe.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "       width=\"24px\">\n",
              "    <path d=\"M7,19H8.4L18.45,9,17,7.55,7,17.6ZM5,21V16.75L18.45,3.32a2,2,0,0,1,2.83,0l1.4,1.43a1.91,1.91,0,0,1,.58,1.4,1.91,1.91,0,0,1-.58,1.4L9.25,21ZM18.45,9,17,7.55Zm-12,3A5.31,5.31,0,0,0,4.9,8.1,5.31,5.31,0,0,0,1,6.5,5.31,5.31,0,0,0,4.9,4.9,5.31,5.31,0,0,0,6.5,1,5.31,5.31,0,0,0,8.1,4.9,5.31,5.31,0,0,0,12,6.5,5.46,5.46,0,0,0,6.5,12Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "    <script>\n",
              "      (() => {\n",
              "      const buttonEl =\n",
              "        document.querySelector('#id_2fdbce8e-bfee-42db-92b3-204ce2c153bc button.colab-df-generate');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      buttonEl.onclick = () => {\n",
              "        google.colab.notebook.generateWithVariable('df_WP058');\n",
              "      }\n",
              "      })();\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe",
              "variable_name": "df_WP058"
            }
          },
          "metadata": {},
          "execution_count": 10
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#Executa um script chamado vep-gc.sh, localizado no diretório lmabrasil-hg38, com dois argumentos: WP019 (identificador de amostra) e Clonal_Hematopoiesis_driver_genes.txt (lista de genes).\n",
        "!sh lmabrasil-hg38/vep-gc.sh WP019 Clonal_Hematopoiesis_driver_genes.txt"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "SgTV8-EiTena",
        "outputId": "1445eda3-3ba7-478c-98df-7dcdf555bc34"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            " \n",
            " ****************************************************************************** \n",
            " *                                                                            * \n",
            " *               STARTING 946d0977-2e78-3650-a7fd-de11ce51fae2                * \n",
            " *                                                                            * \n",
            " ****************************************************************************** \n",
            " executing: filter_vep\n",
            "\n",
            "-------------------- WARNING ----------------------\n",
            "MSG: The script 'filter_vep' couldn't find the VEP annotations INFO field CSQ in 185 lines of the input file\n",
            "FILE: src/ensembl-vep/filter_vep LINE: 345\n",
            "CALLED BY: src/ensembl-vep/filter_vep  LINE: 55\n",
            "Date (localtime)    = Sat Dec 21 13:08:11 2024\n",
            "Ensembl API version = 113\n",
            "---------------------------------------------------\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#carrega para um DataFrame do Pandas os dados genômicos anotados da amostra WP019.\n",
        "df_WP019 = pd.read_csv('/content/lmabrasil-hg38/vep_output/liftOver_WP019_hg19ToHg38.vep.filter.tsv',sep='\\t',index_col=False, engine= 'python')\n",
        "df_WP019"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 163
        },
        "id": "Ac2C-5D_Tgmd",
        "outputId": "6d07b96d-790f-4c95-9ae7-e1fc522f898a"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "   CHROM        POS REF ALT        Location SYMBOL       Consequence  \\\n",
              "0   chr1  179120104  GA   G  chr1:179120105   ABL2    intron_variant   \n",
              "1  chr17   76736877   G   T  chr17:76736877  SRSF2  missense_variant   \n",
              "2   chr9    5073770   G   T    chr9:5073770   JAK2  missense_variant   \n",
              "\n",
              "          Feature         MANE_SELECT         BIOTYPE  \\\n",
              "0     NM_007314.4   ENST00000502732.6  protein_coding   \n",
              "1  NM_001195427.2  ENST00000359995.10  protein_coding   \n",
              "2     NM_004972.4   ENST00000381652.4  protein_coding   \n",
              "\n",
              "                      HGVSc                      HGVSp   EXON INTRON  \\\n",
              "0  NM_007314.4:c.1045+85del                          .      .   6/11   \n",
              "1   NM_001195427.2:c.284C>A  NP_001182356.1:p.Pro95His    1/3      .   \n",
              "2     NM_004972.4:c.1849G>T    NP_004963.1:p.Val617Phe  14/25      .   \n",
              "\n",
              "  VARIANT_CLASS                              SIFT                  PolyPhen  \\\n",
              "0      deletion                                 .                         .   \n",
              "1           SNV                    deleterious(0)                         .   \n",
              "2           SNV  deleterious_low_confidence(0.01)  probably_damaging(0.937)   \n",
              "\n",
              "   gnomADg_AF    MAX_AF    IMPACT  \\\n",
              "0    0.000618  0.002508  MODIFIER   \n",
              "1    0.000118  0.000236  MODERATE   \n",
              "2    0.000362  0.000866  MODERATE   \n",
              "\n",
              "                                                                                                                      CLIN_SIG  \\\n",
              "0                                                                                                                            .   \n",
              "1                                                                                                                            .   \n",
              "2  pathogenic&not_provided&conflicting_interpretations_of_pathogenicity&pathogenic/likely_pathogenic&likely_pathogenic&affects   \n",
              "\n",
              "     SOMATIC                                            Existing_variation  \\\n",
              "0        0&1                                      rs944373851&COSV61016729   \n",
              "1  0&0&1&1&1  rs751713049&CM1511359&COSV57969809&COSV57969816&COSV57969830   \n",
              "2    0&0&1&1                 rs77375493&CM123094&COSV67569051&COSV67571909   \n",
              "\n",
              "                     FILTER TumorID   GT   DP       AD     AF NormalID  NGT  \\\n",
              "0  normal_artifact;slippage   WP019  0/1   78     70,8  0.105    WP020  0/0   \n",
              "1                      PASS   WP019  0/1  284  161,123  0.434    WP020  0/0   \n",
              "2                      PASS   WP019  0/1  153    66,87  0.577    WP020  0/0   \n",
              "\n",
              "   NDP    NAD       NAF  \n",
              "0   43   42,1  0.045000  \n",
              "1  127  127,0  0.008097  \n",
              "2   87   86,1  0.024000  "
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-437ab7a4-34cb-4680-bdcf-9342c7c93017\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>CHROM</th>\n",
              "      <th>POS</th>\n",
              "      <th>REF</th>\n",
              "      <th>ALT</th>\n",
              "      <th>Location</th>\n",
              "      <th>SYMBOL</th>\n",
              "      <th>Consequence</th>\n",
              "      <th>Feature</th>\n",
              "      <th>MANE_SELECT</th>\n",
              "      <th>BIOTYPE</th>\n",
              "      <th>HGVSc</th>\n",
              "      <th>HGVSp</th>\n",
              "      <th>EXON</th>\n",
              "      <th>INTRON</th>\n",
              "      <th>VARIANT_CLASS</th>\n",
              "      <th>SIFT</th>\n",
              "      <th>PolyPhen</th>\n",
              "      <th>gnomADg_AF</th>\n",
              "      <th>MAX_AF</th>\n",
              "      <th>IMPACT</th>\n",
              "      <th>CLIN_SIG</th>\n",
              "      <th>SOMATIC</th>\n",
              "      <th>Existing_variation</th>\n",
              "      <th>FILTER</th>\n",
              "      <th>TumorID</th>\n",
              "      <th>GT</th>\n",
              "      <th>DP</th>\n",
              "      <th>AD</th>\n",
              "      <th>AF</th>\n",
              "      <th>NormalID</th>\n",
              "      <th>NGT</th>\n",
              "      <th>NDP</th>\n",
              "      <th>NAD</th>\n",
              "      <th>NAF</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>chr1</td>\n",
              "      <td>179120104</td>\n",
              "      <td>GA</td>\n",
              "      <td>G</td>\n",
              "      <td>chr1:179120105</td>\n",
              "      <td>ABL2</td>\n",
              "      <td>intron_variant</td>\n",
              "      <td>NM_007314.4</td>\n",
              "      <td>ENST00000502732.6</td>\n",
              "      <td>protein_coding</td>\n",
              "      <td>NM_007314.4:c.1045+85del</td>\n",
              "      <td>.</td>\n",
              "      <td>.</td>\n",
              "      <td>6/11</td>\n",
              "      <td>deletion</td>\n",
              "      <td>.</td>\n",
              "      <td>.</td>\n",
              "      <td>0.000618</td>\n",
              "      <td>0.002508</td>\n",
              "      <td>MODIFIER</td>\n",
              "      <td>.</td>\n",
              "      <td>0&amp;1</td>\n",
              "      <td>rs944373851&amp;COSV61016729</td>\n",
              "      <td>normal_artifact;slippage</td>\n",
              "      <td>WP019</td>\n",
              "      <td>0/1</td>\n",
              "      <td>78</td>\n",
              "      <td>70,8</td>\n",
              "      <td>0.105</td>\n",
              "      <td>WP020</td>\n",
              "      <td>0/0</td>\n",
              "      <td>43</td>\n",
              "      <td>42,1</td>\n",
              "      <td>0.045000</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>chr17</td>\n",
              "      <td>76736877</td>\n",
              "      <td>G</td>\n",
              "      <td>T</td>\n",
              "      <td>chr17:76736877</td>\n",
              "      <td>SRSF2</td>\n",
              "      <td>missense_variant</td>\n",
              "      <td>NM_001195427.2</td>\n",
              "      <td>ENST00000359995.10</td>\n",
              "      <td>protein_coding</td>\n",
              "      <td>NM_001195427.2:c.284C&gt;A</td>\n",
              "      <td>NP_001182356.1:p.Pro95His</td>\n",
              "      <td>1/3</td>\n",
              "      <td>.</td>\n",
              "      <td>SNV</td>\n",
              "      <td>deleterious(0)</td>\n",
              "      <td>.</td>\n",
              "      <td>0.000118</td>\n",
              "      <td>0.000236</td>\n",
              "      <td>MODERATE</td>\n",
              "      <td>.</td>\n",
              "      <td>0&amp;0&amp;1&amp;1&amp;1</td>\n",
              "      <td>rs751713049&amp;CM1511359&amp;COSV57969809&amp;COSV57969816&amp;COSV57969830</td>\n",
              "      <td>PASS</td>\n",
              "      <td>WP019</td>\n",
              "      <td>0/1</td>\n",
              "      <td>284</td>\n",
              "      <td>161,123</td>\n",
              "      <td>0.434</td>\n",
              "      <td>WP020</td>\n",
              "      <td>0/0</td>\n",
              "      <td>127</td>\n",
              "      <td>127,0</td>\n",
              "      <td>0.008097</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>chr9</td>\n",
              "      <td>5073770</td>\n",
              "      <td>G</td>\n",
              "      <td>T</td>\n",
              "      <td>chr9:5073770</td>\n",
              "      <td>JAK2</td>\n",
              "      <td>missense_variant</td>\n",
              "      <td>NM_004972.4</td>\n",
              "      <td>ENST00000381652.4</td>\n",
              "      <td>protein_coding</td>\n",
              "      <td>NM_004972.4:c.1849G&gt;T</td>\n",
              "      <td>NP_004963.1:p.Val617Phe</td>\n",
              "      <td>14/25</td>\n",
              "      <td>.</td>\n",
              "      <td>SNV</td>\n",
              "      <td>deleterious_low_confidence(0.01)</td>\n",
              "      <td>probably_damaging(0.937)</td>\n",
              "      <td>0.000362</td>\n",
              "      <td>0.000866</td>\n",
              "      <td>MODERATE</td>\n",
              "      <td>pathogenic&amp;not_provided&amp;conflicting_interpretations_of_pathogenicity&amp;pathogenic/likely_pathogenic&amp;likely_pathogenic&amp;affects</td>\n",
              "      <td>0&amp;0&amp;1&amp;1</td>\n",
              "      <td>rs77375493&amp;CM123094&amp;COSV67569051&amp;COSV67571909</td>\n",
              "      <td>PASS</td>\n",
              "      <td>WP019</td>\n",
              "      <td>0/1</td>\n",
              "      <td>153</td>\n",
              "      <td>66,87</td>\n",
              "      <td>0.577</td>\n",
              "      <td>WP020</td>\n",
              "      <td>0/0</td>\n",
              "      <td>87</td>\n",
              "      <td>86,1</td>\n",
              "      <td>0.024000</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-437ab7a4-34cb-4680-bdcf-9342c7c93017')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-437ab7a4-34cb-4680-bdcf-9342c7c93017 button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-437ab7a4-34cb-4680-bdcf-9342c7c93017');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "<div id=\"df-52597219-b7f0-4b72-8cf4-b414fcf220e9\">\n",
              "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-52597219-b7f0-4b72-8cf4-b414fcf220e9')\"\n",
              "            title=\"Suggest charts\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "  </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "  <script>\n",
              "    async function quickchart(key) {\n",
              "      const quickchartButtonEl =\n",
              "        document.querySelector('#' + key + ' button');\n",
              "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "      try {\n",
              "        const charts = await google.colab.kernel.invokeFunction(\n",
              "            'suggestCharts', [key], {});\n",
              "      } catch (error) {\n",
              "        console.error('Error during call to suggestCharts:', error);\n",
              "      }\n",
              "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "    }\n",
              "    (() => {\n",
              "      let quickchartButtonEl =\n",
              "        document.querySelector('#df-52597219-b7f0-4b72-8cf4-b414fcf220e9 button');\n",
              "      quickchartButtonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "    })();\n",
              "  </script>\n",
              "</div>\n",
              "\n",
              "  <div id=\"id_fe3ac92f-b0c3-41b8-8b04-622df4fb4eb7\">\n",
              "    <style>\n",
              "      .colab-df-generate {\n",
              "        background-color: #E8F0FE;\n",
              "        border: none;\n",
              "        border-radius: 50%;\n",
              "        cursor: pointer;\n",
              "        display: none;\n",
              "        fill: #1967D2;\n",
              "        height: 32px;\n",
              "        padding: 0 0 0 0;\n",
              "        width: 32px;\n",
              "      }\n",
              "\n",
              "      .colab-df-generate:hover {\n",
              "        background-color: #E2EBFA;\n",
              "        box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "        fill: #174EA6;\n",
              "      }\n",
              "\n",
              "      [theme=dark] .colab-df-generate {\n",
              "        background-color: #3B4455;\n",
              "        fill: #D2E3FC;\n",
              "      }\n",
              "\n",
              "      [theme=dark] .colab-df-generate:hover {\n",
              "        background-color: #434B5C;\n",
              "        box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "        filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "        fill: #FFFFFF;\n",
              "      }\n",
              "    </style>\n",
              "    <button class=\"colab-df-generate\" onclick=\"generateWithVariable('df_WP019')\"\n",
              "            title=\"Generate code using this dataframe.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "       width=\"24px\">\n",
              "    <path d=\"M7,19H8.4L18.45,9,17,7.55,7,17.6ZM5,21V16.75L18.45,3.32a2,2,0,0,1,2.83,0l1.4,1.43a1.91,1.91,0,0,1,.58,1.4,1.91,1.91,0,0,1-.58,1.4L9.25,21ZM18.45,9,17,7.55Zm-12,3A5.31,5.31,0,0,0,4.9,8.1,5.31,5.31,0,0,0,1,6.5,5.31,5.31,0,0,0,4.9,4.9,5.31,5.31,0,0,0,6.5,1,5.31,5.31,0,0,0,8.1,4.9,5.31,5.31,0,0,0,12,6.5,5.46,5.46,0,0,0,6.5,12Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "    <script>\n",
              "      (() => {\n",
              "      const buttonEl =\n",
              "        document.querySelector('#id_fe3ac92f-b0c3-41b8-8b04-622df4fb4eb7 button.colab-df-generate');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      buttonEl.onclick = () => {\n",
              "        google.colab.notebook.generateWithVariable('df_WP019');\n",
              "      }\n",
              "      })();\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe",
              "variable_name": "df_WP019"
            }
          },
          "metadata": {},
          "execution_count": 12
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#Analisa variantes genéticas da amostra WP068\n",
        "!sh lmabrasil-hg38/vep-gc.sh WP068 Abnormal_mast_cell_morphology.txt"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "nHeueiK1Ti1B",
        "outputId": "21bb9882-550b-46ca-9664-64c9325c145f"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            " \n",
            " ****************************************************************************** \n",
            " *                                                                            * \n",
            " *               STARTING a5f99a02-71e1-396f-b0b8-24b5907bb9dc                * \n",
            " *                                                                            * \n",
            " ****************************************************************************** \n",
            " executing: filter_vep\n",
            "\n",
            "-------------------- WARNING ----------------------\n",
            "MSG: The script 'filter_vep' couldn't find the VEP annotations INFO field CSQ in 146 lines of the input file\n",
            "FILE: src/ensembl-vep/filter_vep LINE: 345\n",
            "CALLED BY: src/ensembl-vep/filter_vep  LINE: 55\n",
            "Date (localtime)    = Sat Dec 21 13:10:39 2024\n",
            "Ensembl API version = 113\n",
            "---------------------------------------------------\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#Exibe os dados genômicos da amostra WP068 que foram anotados pelo VEP após a conversão do genoma hg19 para hg38.\n",
        "df_WP068 = pd.read_csv('/content/lmabrasil-hg38/vep_output/liftOver_WP068_hg19ToHg38.vep.filter.tsv',sep='\\t',index_col=False, engine= 'python')\n",
        "df_WP068"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 145
        },
        "id": "bQ2U7KIBTlHT",
        "outputId": "123a24ba-3fda-41d0-e057-58aa42c030fb"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "  CHROM        POS REF ALT        Location SYMBOL       Consequence  \\\n",
              "0  chr4   54733155   A   T   chr4:54733155    KIT  missense_variant   \n",
              "1  chr4  105276128   T   C  chr4:105276128   TET2  missense_variant   \n",
              "\n",
              "          Feature        MANE_SELECT         BIOTYPE  \\\n",
              "0     NM_000222.3  ENST00000288135.6  protein_coding   \n",
              "1  NM_001127208.3  ENST00000380013.9  protein_coding   \n",
              "\n",
              "                      HGVSc                        HGVSp   EXON INTRON  \\\n",
              "0     NM_000222.3:c.2447A>T      NP_000213.1:p.Asp816Val  17/21      .   \n",
              "1  NM_001127208.3:c.5618T>C  NP_001120680.1:p.Ile1873Thr  11/11      .   \n",
              "\n",
              "  VARIANT_CLASS            SIFT                  PolyPhen gnomADg_AF  MAX_AF  \\\n",
              "0           SNV  deleterious(0)  probably_damaging(0.936)          .  0.0000   \n",
              "1           SNV  deleterious(0)  probably_damaging(0.999)  4.604e-05  0.0008   \n",
              "\n",
              "     IMPACT                                                   CLIN_SIG  \\\n",
              "0  MODERATE  uncertain_significance&pathogenic&likely_pathogenic&other   \n",
              "1  MODERATE                                                          .   \n",
              "\n",
              "     SOMATIC                                           Existing_variation  \\\n",
              "0  0&0&1&1&1  rs121913507&CM952169&COSV55386424&COSV55388612&COSV55408930   \n",
              "1      0&1&1                        rs116519313&COSV54396330&COSV99482974   \n",
              "\n",
              "  FILTER TumorID   GT   DP       AD     AF NormalID  NGT  NDP    NAD    NAF  \n",
              "0   PASS   WP068  0/1  228   150,78  0.353    WP069  0/0   86   86,0  0.014  \n",
              "1   PASS   WP068  0/1  345  226,119  0.358    WP069  0/0  181  175,6  0.039  "
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-577fa299-8270-4bd2-b994-4a00dbfb0f2d\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>CHROM</th>\n",
              "      <th>POS</th>\n",
              "      <th>REF</th>\n",
              "      <th>ALT</th>\n",
              "      <th>Location</th>\n",
              "      <th>SYMBOL</th>\n",
              "      <th>Consequence</th>\n",
              "      <th>Feature</th>\n",
              "      <th>MANE_SELECT</th>\n",
              "      <th>BIOTYPE</th>\n",
              "      <th>HGVSc</th>\n",
              "      <th>HGVSp</th>\n",
              "      <th>EXON</th>\n",
              "      <th>INTRON</th>\n",
              "      <th>VARIANT_CLASS</th>\n",
              "      <th>SIFT</th>\n",
              "      <th>PolyPhen</th>\n",
              "      <th>gnomADg_AF</th>\n",
              "      <th>MAX_AF</th>\n",
              "      <th>IMPACT</th>\n",
              "      <th>CLIN_SIG</th>\n",
              "      <th>SOMATIC</th>\n",
              "      <th>Existing_variation</th>\n",
              "      <th>FILTER</th>\n",
              "      <th>TumorID</th>\n",
              "      <th>GT</th>\n",
              "      <th>DP</th>\n",
              "      <th>AD</th>\n",
              "      <th>AF</th>\n",
              "      <th>NormalID</th>\n",
              "      <th>NGT</th>\n",
              "      <th>NDP</th>\n",
              "      <th>NAD</th>\n",
              "      <th>NAF</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>chr4</td>\n",
              "      <td>54733155</td>\n",
              "      <td>A</td>\n",
              "      <td>T</td>\n",
              "      <td>chr4:54733155</td>\n",
              "      <td>KIT</td>\n",
              "      <td>missense_variant</td>\n",
              "      <td>NM_000222.3</td>\n",
              "      <td>ENST00000288135.6</td>\n",
              "      <td>protein_coding</td>\n",
              "      <td>NM_000222.3:c.2447A&gt;T</td>\n",
              "      <td>NP_000213.1:p.Asp816Val</td>\n",
              "      <td>17/21</td>\n",
              "      <td>.</td>\n",
              "      <td>SNV</td>\n",
              "      <td>deleterious(0)</td>\n",
              "      <td>probably_damaging(0.936)</td>\n",
              "      <td>.</td>\n",
              "      <td>0.0000</td>\n",
              "      <td>MODERATE</td>\n",
              "      <td>uncertain_significance&amp;pathogenic&amp;likely_pathogenic&amp;other</td>\n",
              "      <td>0&amp;0&amp;1&amp;1&amp;1</td>\n",
              "      <td>rs121913507&amp;CM952169&amp;COSV55386424&amp;COSV55388612&amp;COSV55408930</td>\n",
              "      <td>PASS</td>\n",
              "      <td>WP068</td>\n",
              "      <td>0/1</td>\n",
              "      <td>228</td>\n",
              "      <td>150,78</td>\n",
              "      <td>0.353</td>\n",
              "      <td>WP069</td>\n",
              "      <td>0/0</td>\n",
              "      <td>86</td>\n",
              "      <td>86,0</td>\n",
              "      <td>0.014</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>chr4</td>\n",
              "      <td>105276128</td>\n",
              "      <td>T</td>\n",
              "      <td>C</td>\n",
              "      <td>chr4:105276128</td>\n",
              "      <td>TET2</td>\n",
              "      <td>missense_variant</td>\n",
              "      <td>NM_001127208.3</td>\n",
              "      <td>ENST00000380013.9</td>\n",
              "      <td>protein_coding</td>\n",
              "      <td>NM_001127208.3:c.5618T&gt;C</td>\n",
              "      <td>NP_001120680.1:p.Ile1873Thr</td>\n",
              "      <td>11/11</td>\n",
              "      <td>.</td>\n",
              "      <td>SNV</td>\n",
              "      <td>deleterious(0)</td>\n",
              "      <td>probably_damaging(0.999)</td>\n",
              "      <td>4.604e-05</td>\n",
              "      <td>0.0008</td>\n",
              "      <td>MODERATE</td>\n",
              "      <td>.</td>\n",
              "      <td>0&amp;1&amp;1</td>\n",
              "      <td>rs116519313&amp;COSV54396330&amp;COSV99482974</td>\n",
              "      <td>PASS</td>\n",
              "      <td>WP068</td>\n",
              "      <td>0/1</td>\n",
              "      <td>345</td>\n",
              "      <td>226,119</td>\n",
              "      <td>0.358</td>\n",
              "      <td>WP069</td>\n",
              "      <td>0/0</td>\n",
              "      <td>181</td>\n",
              "      <td>175,6</td>\n",
              "      <td>0.039</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-577fa299-8270-4bd2-b994-4a00dbfb0f2d')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-577fa299-8270-4bd2-b994-4a00dbfb0f2d button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-577fa299-8270-4bd2-b994-4a00dbfb0f2d');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "<div id=\"df-605ba75c-c057-4310-a4fb-4194d11b8a92\">\n",
              "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-605ba75c-c057-4310-a4fb-4194d11b8a92')\"\n",
              "            title=\"Suggest charts\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "  </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "  <script>\n",
              "    async function quickchart(key) {\n",
              "      const quickchartButtonEl =\n",
              "        document.querySelector('#' + key + ' button');\n",
              "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "      try {\n",
              "        const charts = await google.colab.kernel.invokeFunction(\n",
              "            'suggestCharts', [key], {});\n",
              "      } catch (error) {\n",
              "        console.error('Error during call to suggestCharts:', error);\n",
              "      }\n",
              "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "    }\n",
              "    (() => {\n",
              "      let quickchartButtonEl =\n",
              "        document.querySelector('#df-605ba75c-c057-4310-a4fb-4194d11b8a92 button');\n",
              "      quickchartButtonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "    })();\n",
              "  </script>\n",
              "</div>\n",
              "\n",
              "  <div id=\"id_81ed2dce-dcda-4c15-819e-94dff2dd9fd0\">\n",
              "    <style>\n",
              "      .colab-df-generate {\n",
              "        background-color: #E8F0FE;\n",
              "        border: none;\n",
              "        border-radius: 50%;\n",
              "        cursor: pointer;\n",
              "        display: none;\n",
              "        fill: #1967D2;\n",
              "        height: 32px;\n",
              "        padding: 0 0 0 0;\n",
              "        width: 32px;\n",
              "      }\n",
              "\n",
              "      .colab-df-generate:hover {\n",
              "        background-color: #E2EBFA;\n",
              "        box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "        fill: #174EA6;\n",
              "      }\n",
              "\n",
              "      [theme=dark] .colab-df-generate {\n",
              "        background-color: #3B4455;\n",
              "        fill: #D2E3FC;\n",
              "      }\n",
              "\n",
              "      [theme=dark] .colab-df-generate:hover {\n",
              "        background-color: #434B5C;\n",
              "        box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "        filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "        fill: #FFFFFF;\n",
              "      }\n",
              "    </style>\n",
              "    <button class=\"colab-df-generate\" onclick=\"generateWithVariable('df_WP068')\"\n",
              "            title=\"Generate code using this dataframe.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "       width=\"24px\">\n",
              "    <path d=\"M7,19H8.4L18.45,9,17,7.55,7,17.6ZM5,21V16.75L18.45,3.32a2,2,0,0,1,2.83,0l1.4,1.43a1.91,1.91,0,0,1,.58,1.4,1.91,1.91,0,0,1-.58,1.4L9.25,21ZM18.45,9,17,7.55Zm-12,3A5.31,5.31,0,0,0,4.9,8.1,5.31,5.31,0,0,0,1,6.5,5.31,5.31,0,0,0,4.9,4.9,5.31,5.31,0,0,0,6.5,1,5.31,5.31,0,0,0,8.1,4.9,5.31,5.31,0,0,0,12,6.5,5.46,5.46,0,0,0,6.5,12Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "    <script>\n",
              "      (() => {\n",
              "      const buttonEl =\n",
              "        document.querySelector('#id_81ed2dce-dcda-4c15-819e-94dff2dd9fd0 button.colab-df-generate');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      buttonEl.onclick = () => {\n",
              "        google.colab.notebook.generateWithVariable('df_WP068');\n",
              "      }\n",
              "      })();\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe",
              "variable_name": "df_WP068"
            }
          },
          "metadata": {},
          "execution_count": 14
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#Código na célula será interpretado como comandos Bash.\n",
        "%%bash\n",
        "#Executa o script vep-gc.sh para analisar variantes genéticas da amostra WP048, utilizando a lista de genes associados à hematopoiese clonal.\n",
        "sh lmabrasil-hg38/vep-gc.sh WP048 Clonal_Hematopoiesis_driver_genes.txt"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "6OJo4sahTmx1",
        "outputId": "72758a8f-7638-40c7-afda-d2415ab61442"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            " \n",
            " ****************************************************************************** \n",
            " *                                                                            * \n",
            " *               STARTING abef4d0b-02ac-34bd-a2f5-83493b558b4e                * \n",
            " *                                                                            * \n",
            " ****************************************************************************** \n",
            " executing: filter_vep\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "\n",
            "-------------------- EXCEPTION --------------------\n",
            "MSG: ERROR: Could not find input file vep_output/liftOver_WP048_hg19ToHg38.vep.vcf\n",
            "\n",
            "STACK main::main /opt/vep/src/ensembl-vep/filter_vep:150\n",
            "STACK toplevel /opt/vep/src/ensembl-vep/filter_vep:55\n",
            "Date (localtime)    = Sat Dec 21 13:14:08 2024\n",
            "Ensembl API version = 113\n",
            "---------------------------------------------------\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#Exibe os dados da amostra WP048, que foram anotados pelo VEP após a conversão do genoma de hg19 para hg38.\n",
        "df_WP048 = pd.read_csv('/content/lmabrasil-hg38/vep_output/liftOver_WP048_hg19ToHg38.vep.filter.tsv',sep='\\t',index_col=False, engine= 'python')\n",
        "df_WP048"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 145
        },
        "id": "0lrgj2F2ToTh",
        "outputId": "e2f0e4f9-5fad-4b87-e3c6-675a0ee25157"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "  CHROM        POS REF ALT        Location SYMBOL       Consequence  \\\n",
              "0  chr1  114716123   C   T  chr1:114716123   NRAS  missense_variant   \n",
              "1  chr9    5073770   G   T    chr9:5073770   JAK2  missense_variant   \n",
              "\n",
              "       Feature        MANE_SELECT         BIOTYPE                  HGVSc  \\\n",
              "0  NM_002524.5  ENST00000369535.5  protein_coding    NM_002524.5:c.38G>A   \n",
              "1  NM_004972.4  ENST00000381652.4  protein_coding  NM_004972.4:c.1849G>T   \n",
              "\n",
              "                     HGVSp   EXON INTRON VARIANT_CLASS  \\\n",
              "0   NP_002515.1:p.Gly13Asp    2/7      .           SNV   \n",
              "1  NP_004963.1:p.Val617Phe  14/25      .           SNV   \n",
              "\n",
              "                               SIFT                  PolyPhen gnomADg_AF  \\\n",
              "0    tolerated_low_confidence(0.06)                         .          .   \n",
              "1  deleterious_low_confidence(0.01)  probably_damaging(0.937)  0.0003621   \n",
              "\n",
              "     MAX_AF    IMPACT  \\\n",
              "0  0.000099  MODERATE   \n",
              "1  0.000866  MODERATE   \n",
              "\n",
              "                                                                                                                      CLIN_SIG  \\\n",
              "0                                                                                                 pathogenic&likely_pathogenic   \n",
              "1  pathogenic&not_provided&conflicting_interpretations_of_pathogenicity&pathogenic/likely_pathogenic&likely_pathogenic&affects   \n",
              "\n",
              "     SOMATIC                                           Existing_variation  \\\n",
              "0  0&0&1&1&1  rs121434596&CM071907&COSV54736416&COSV54736480&COSV54736793   \n",
              "1    0&0&1&1                rs77375493&CM123094&COSV67569051&COSV67571909   \n",
              "\n",
              "  FILTER TumorID   GT   DP      AD     AF NormalID  NGT  NDP    NAD    NAF  \n",
              "0   PASS   WP048  0/1  232  177,55  0.258    WP049  0/0  105  105,0  0.011  \n",
              "1   PASS   WP048  0/1  165   84,81  0.480    WP049  0/0   93   93,0  0.014  "
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-db506587-7d1a-4d18-889d-89d0a8eb213d\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>CHROM</th>\n",
              "      <th>POS</th>\n",
              "      <th>REF</th>\n",
              "      <th>ALT</th>\n",
              "      <th>Location</th>\n",
              "      <th>SYMBOL</th>\n",
              "      <th>Consequence</th>\n",
              "      <th>Feature</th>\n",
              "      <th>MANE_SELECT</th>\n",
              "      <th>BIOTYPE</th>\n",
              "      <th>HGVSc</th>\n",
              "      <th>HGVSp</th>\n",
              "      <th>EXON</th>\n",
              "      <th>INTRON</th>\n",
              "      <th>VARIANT_CLASS</th>\n",
              "      <th>SIFT</th>\n",
              "      <th>PolyPhen</th>\n",
              "      <th>gnomADg_AF</th>\n",
              "      <th>MAX_AF</th>\n",
              "      <th>IMPACT</th>\n",
              "      <th>CLIN_SIG</th>\n",
              "      <th>SOMATIC</th>\n",
              "      <th>Existing_variation</th>\n",
              "      <th>FILTER</th>\n",
              "      <th>TumorID</th>\n",
              "      <th>GT</th>\n",
              "      <th>DP</th>\n",
              "      <th>AD</th>\n",
              "      <th>AF</th>\n",
              "      <th>NormalID</th>\n",
              "      <th>NGT</th>\n",
              "      <th>NDP</th>\n",
              "      <th>NAD</th>\n",
              "      <th>NAF</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>chr1</td>\n",
              "      <td>114716123</td>\n",
              "      <td>C</td>\n",
              "      <td>T</td>\n",
              "      <td>chr1:114716123</td>\n",
              "      <td>NRAS</td>\n",
              "      <td>missense_variant</td>\n",
              "      <td>NM_002524.5</td>\n",
              "      <td>ENST00000369535.5</td>\n",
              "      <td>protein_coding</td>\n",
              "      <td>NM_002524.5:c.38G&gt;A</td>\n",
              "      <td>NP_002515.1:p.Gly13Asp</td>\n",
              "      <td>2/7</td>\n",
              "      <td>.</td>\n",
              "      <td>SNV</td>\n",
              "      <td>tolerated_low_confidence(0.06)</td>\n",
              "      <td>.</td>\n",
              "      <td>.</td>\n",
              "      <td>0.000099</td>\n",
              "      <td>MODERATE</td>\n",
              "      <td>pathogenic&amp;likely_pathogenic</td>\n",
              "      <td>0&amp;0&amp;1&amp;1&amp;1</td>\n",
              "      <td>rs121434596&amp;CM071907&amp;COSV54736416&amp;COSV54736480&amp;COSV54736793</td>\n",
              "      <td>PASS</td>\n",
              "      <td>WP048</td>\n",
              "      <td>0/1</td>\n",
              "      <td>232</td>\n",
              "      <td>177,55</td>\n",
              "      <td>0.258</td>\n",
              "      <td>WP049</td>\n",
              "      <td>0/0</td>\n",
              "      <td>105</td>\n",
              "      <td>105,0</td>\n",
              "      <td>0.011</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>chr9</td>\n",
              "      <td>5073770</td>\n",
              "      <td>G</td>\n",
              "      <td>T</td>\n",
              "      <td>chr9:5073770</td>\n",
              "      <td>JAK2</td>\n",
              "      <td>missense_variant</td>\n",
              "      <td>NM_004972.4</td>\n",
              "      <td>ENST00000381652.4</td>\n",
              "      <td>protein_coding</td>\n",
              "      <td>NM_004972.4:c.1849G&gt;T</td>\n",
              "      <td>NP_004963.1:p.Val617Phe</td>\n",
              "      <td>14/25</td>\n",
              "      <td>.</td>\n",
              "      <td>SNV</td>\n",
              "      <td>deleterious_low_confidence(0.01)</td>\n",
              "      <td>probably_damaging(0.937)</td>\n",
              "      <td>0.0003621</td>\n",
              "      <td>0.000866</td>\n",
              "      <td>MODERATE</td>\n",
              "      <td>pathogenic&amp;not_provided&amp;conflicting_interpretations_of_pathogenicity&amp;pathogenic/likely_pathogenic&amp;likely_pathogenic&amp;affects</td>\n",
              "      <td>0&amp;0&amp;1&amp;1</td>\n",
              "      <td>rs77375493&amp;CM123094&amp;COSV67569051&amp;COSV67571909</td>\n",
              "      <td>PASS</td>\n",
              "      <td>WP048</td>\n",
              "      <td>0/1</td>\n",
              "      <td>165</td>\n",
              "      <td>84,81</td>\n",
              "      <td>0.480</td>\n",
              "      <td>WP049</td>\n",
              "      <td>0/0</td>\n",
              "      <td>93</td>\n",
              "      <td>93,0</td>\n",
              "      <td>0.014</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-db506587-7d1a-4d18-889d-89d0a8eb213d')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-db506587-7d1a-4d18-889d-89d0a8eb213d button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-db506587-7d1a-4d18-889d-89d0a8eb213d');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "<div id=\"df-5c47f749-5261-4440-aaeb-730cdae56a4f\">\n",
              "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-5c47f749-5261-4440-aaeb-730cdae56a4f')\"\n",
              "            title=\"Suggest charts\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "  </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "  <script>\n",
              "    async function quickchart(key) {\n",
              "      const quickchartButtonEl =\n",
              "        document.querySelector('#' + key + ' button');\n",
              "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "      try {\n",
              "        const charts = await google.colab.kernel.invokeFunction(\n",
              "            'suggestCharts', [key], {});\n",
              "      } catch (error) {\n",
              "        console.error('Error during call to suggestCharts:', error);\n",
              "      }\n",
              "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "    }\n",
              "    (() => {\n",
              "      let quickchartButtonEl =\n",
              "        document.querySelector('#df-5c47f749-5261-4440-aaeb-730cdae56a4f button');\n",
              "      quickchartButtonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "    })();\n",
              "  </script>\n",
              "</div>\n",
              "\n",
              "  <div id=\"id_d1ed4b08-f5ce-48d4-b93b-a0d5c251b0cb\">\n",
              "    <style>\n",
              "      .colab-df-generate {\n",
              "        background-color: #E8F0FE;\n",
              "        border: none;\n",
              "        border-radius: 50%;\n",
              "        cursor: pointer;\n",
              "        display: none;\n",
              "        fill: #1967D2;\n",
              "        height: 32px;\n",
              "        padding: 0 0 0 0;\n",
              "        width: 32px;\n",
              "      }\n",
              "\n",
              "      .colab-df-generate:hover {\n",
              "        background-color: #E2EBFA;\n",
              "        box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "        fill: #174EA6;\n",
              "      }\n",
              "\n",
              "      [theme=dark] .colab-df-generate {\n",
              "        background-color: #3B4455;\n",
              "        fill: #D2E3FC;\n",
              "      }\n",
              "\n",
              "      [theme=dark] .colab-df-generate:hover {\n",
              "        background-color: #434B5C;\n",
              "        box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "        filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "        fill: #FFFFFF;\n",
              "      }\n",
              "    </style>\n",
              "    <button class=\"colab-df-generate\" onclick=\"generateWithVariable('df_WP048')\"\n",
              "            title=\"Generate code using this dataframe.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "       width=\"24px\">\n",
              "    <path d=\"M7,19H8.4L18.45,9,17,7.55,7,17.6ZM5,21V16.75L18.45,3.32a2,2,0,0,1,2.83,0l1.4,1.43a1.91,1.91,0,0,1,.58,1.4,1.91,1.91,0,0,1-.58,1.4L9.25,21ZM18.45,9,17,7.55Zm-12,3A5.31,5.31,0,0,0,4.9,8.1,5.31,5.31,0,0,0,1,6.5,5.31,5.31,0,0,0,4.9,4.9,5.31,5.31,0,0,0,6.5,1,5.31,5.31,0,0,0,8.1,4.9,5.31,5.31,0,0,0,12,6.5,5.46,5.46,0,0,0,6.5,12Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "    <script>\n",
              "      (() => {\n",
              "      const buttonEl =\n",
              "        document.querySelector('#id_d1ed4b08-f5ce-48d4-b93b-a0d5c251b0cb button.colab-df-generate');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      buttonEl.onclick = () => {\n",
              "        google.colab.notebook.generateWithVariable('df_WP048');\n",
              "      }\n",
              "      })();\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe",
              "variable_name": "df_WP048"
            }
          },
          "metadata": {},
          "execution_count": 16
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Cancer Genome Interpreter (CGI) - API"
      ],
      "metadata": {
        "id": "oI43C71FWbHw"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "Gerando Token no seu usuário CGI"
      ],
      "metadata": {
        "id": "r3pDVk9kWemw"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "# **WP017**"
      ],
      "metadata": {
        "id": "VcPoiZRxfbok"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "#WP017\n",
        "#Código na célula será interpretado como comandos Bash.\n",
        "%%bash\n",
        "#Extrai as quatro primeiras colunas do arquivo, substitui o texto CHROM por CHR nas colunas extraídas, salva o resultado no arquivo e exisbe as 10 primeiras linhas do arquivo.\n",
        "cut -f1-4 /content/lmabrasil-hg38/vep_output/liftOver_WP017_hg19ToHg38.vep.filter.tsv | sed -e \"s/CHROM/CHR/g\"  > df_WP017-cgi.txt\n",
        "head df_WP017-cgi.txt"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "qyg7moagWg5P",
        "outputId": "ffa3433b-d074-4480-d18b-ae3341db6bac"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "CHR\tPOS\tREF\tALT\n",
            "chr1\t114716127\tC\tT\n",
            "chr16\t67616834\tG\tA\n",
            "chr17\t7669662\tT\tG\n",
            "chr19\t12943750\tAGCAGAGGCTTAAGGAGGAGGAAGAAGACAAGAAACGCAAAGAGGAGGAGGAG\tA\n",
            "chr20\t32434638\tA\tAG\n",
            "chr21\t43094667\tT\tG\n",
            "chr8\t100712766\tC\tCA\n",
            "chr8\t100715522\tT\tC\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Enviando um Job - job_id."
      ],
      "metadata": {
        "id": "TWFl978fXKsg"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "import requests\n",
        "#Contém informações de autenticação.\n",
        "headers = {'Authorization': 'silfrizzarini@gmail.com 7f1b6ddf77bc5186c964'}\n",
        "#Dicionário que contém os dados a serem enviados no corpo da requisição.\n",
        "payload = {'cancer_type': 'HEMATO', 'title': 'Somatic MF', 'reference': 'hg38'}\n",
        "#Envia a requisição.\n",
        "r = requests.post('https://www.cancergenomeinterpreter.org/api/v1',\n",
        "                headers=headers,\n",
        "                files={\n",
        "                        'mutations': open('/content/df_WP017-cgi.txt', 'rb')\n",
        "                        },\n",
        "                data=payload)\n",
        "#A resposta da requisição é armazenada na variável r. Converte a resposta em formato JSON.\n",
        "r.json()"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 35
        },
        "id": "3Yj3af73XMRe",
        "outputId": "21661682-c176-4ad5-88fb-1bd94f7836d6"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "'0600109548f9620e87e4'"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "string"
            }
          },
          "metadata": {},
          "execution_count": 18
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Visualizando os identificadores."
      ],
      "metadata": {
        "id": "YIMXoDwVX3Q1"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "#Envia uma requisição para o Cancer Genome Interpreter para obter os resultados de uma análise, converte a resposta para o formato JSON e retorna informações sobre as variantes genéticas analisadas.\n",
        "import requests\n",
        "job_id =\"0600109548f9620e87e4\"\n",
        "\n",
        "headers = {'Authorization': 'silfrizzarini@gmail.com 7f1b6ddf77bc5186c964'}\n",
        "r = requests.get('https://www.cancergenomeinterpreter.org/api/v1/%s' % job_id, headers=headers)\n",
        "r.json()"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "C0fVIU6-X4La",
        "outputId": "2b822310-4e07-43a1-cc47-b591795f53cf"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "{'status': 'Done',\n",
              " 'metadata': {'id': '0600109548f9620e87e4',\n",
              "  'user': 'silfrizzarini@gmail.com',\n",
              "  'title': 'Somatic MF',\n",
              "  'cancertype': 'HEMATO',\n",
              "  'reference': 'hg38',\n",
              "  'dataset': 'input.tsv',\n",
              "  'date': '2024-12-21 13:17:04'}}"
            ]
          },
          "metadata": {},
          "execution_count": 19
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Acessando informações do Job."
      ],
      "metadata": {
        "id": "cSgM47s4YBV_"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "#Envia uma requisição para o Cancer Genome Interpreter para obter os logs da análise, retorna em formato JSON e mostra detalhes sobre o andamento ou erros da análise.\n",
        "import requests\n",
        "job_id =\"0600109548f9620e87e4\"\n",
        "\n",
        "headers = {'Authorization': 'silfrizzarini@gmail.com 7f1b6ddf77bc5186c964'}\n",
        "payload={'action':'logs'}\n",
        "r = requests.get('https://www.cancergenomeinterpreter.org/api/v1/%s' % job_id, headers=headers, params=payload)\n",
        "r.json()"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "0mulkTtcYB_S",
        "outputId": "0d06356d-4f57-45be-f36c-70ab3e280988"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "{'status': 'Done',\n",
              " 'logs': ['# cgi analyze input.tsv -c HEMATO -g hg38',\n",
              "  '2024-12-21 14:17:09,155 INFO     Parsing input01.tsv\\n',\n",
              "  '2024-12-21 14:17:12,744 INFO     Running VEP\\n',\n",
              "  '2024-12-21 14:17:14,542 INFO     Check cancer genes and consensus roles\\n',\n",
              "  '2024-12-21 14:17:14,628 INFO     Annotate BoostDM mutations\\n',\n",
              "  '2024-12-21 14:17:15,856 INFO     Annotate OncodriveMUT mutations\\n',\n",
              "  '2024-12-21 14:17:22,751 INFO     Annotate validated oncogenic mutations\\n',\n",
              "  '2024-12-21 14:17:22,898 INFO     Check oncogenic classification\\n',\n",
              "  '2024-12-21 14:17:22,959 INFO     Matching biomarkers\\n',\n",
              "  '2024-12-21 14:17:23,065 INFO     Prescription finished\\n',\n",
              "  '2024-12-21 14:17:23,077 INFO     Aggregate metrics\\n',\n",
              "  '2024-12-21 14:17:48,494 INFO     Compress output files\\n',\n",
              "  '2024-12-21 14:17:48,570 INFO     Analysis done\\n']}"
            ]
          },
          "metadata": {},
          "execution_count": 20
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Download dos Resultados (file.zip)."
      ],
      "metadata": {
        "id": "OhWhINTaYItQ"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "#Baixa os resultados da análise em file.zip, fica disponível no diretório de trabalho.\n",
        "import requests\n",
        "job_id =\"0600109548f9620e87e4\"\n",
        "\n",
        "headers = {'Authorization': 'silfrizzarini@gmail.com 7f1b6ddf77bc5186c964'}\n",
        "payload={'action':'download'}\n",
        "r = requests.get('https://www.cancergenomeinterpreter.org/api/v1/%s' % job_id, headers=headers, params=payload)\n",
        "with open('file.zip', 'wb') as fd:\n",
        "    fd.write(r._content)"
      ],
      "metadata": {
        "id": "1jE0WJX3YJi2"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "Descompactando o arquivo file.zip"
      ],
      "metadata": {
        "id": "9SMB35kGYRqP"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "#Permite confirmar o tipo do arquivo e garantir que ele tenha sido baixado corretamente.\n",
        "!file file.zip"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "UL9TT7gKYSOz",
        "outputId": "5a4b7fa0-fea2-4da7-910a-56009758599d"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "file.zip: Zip archive data, at least v2.0 to extract, compression method=deflate\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#Irá descompactar o conteúdo do arquivo file.zip e colocá-lo dentro de um diretório chamado WP017.\n",
        "!unzip -d WP017 file.zip"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "1MGdoPzpYXyj",
        "outputId": "239290a5-39f2-4431-a02a-32dc18e25301"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Archive:  file.zip\n",
            "  inflating: WP017/alterations.tsv   \n",
            "  inflating: WP017/biomarkers.tsv    \n",
            "  inflating: WP017/input01.tsv       \n",
            "  inflating: WP017/summary.txt       \n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Resultado: alterations.tsv"
      ],
      "metadata": {
        "id": "fM9zkdgWYa52"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "#Lê o arquivo alterations.tsv que está no diretório /content/WP017/, usando tabulação. Retorna um DataFrame contendo os dados do arquivo.\n",
        "pd.read_csv('/content/WP017/alterations.tsv',sep='\\t',index_col=False, engine= 'python')"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 563
        },
        "id": "AytorrGXYbeV",
        "outputId": "b634f5a7-e62f-4181-b3c3-2c6e2cf2499a"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "    Input ID  CHROMOSOME   POSITION  \\\n",
              "0  input01_1           1  114716127   \n",
              "1  input01_2          16   67616834   \n",
              "2  input01_3          17    7669662   \n",
              "3  input01_4          19   12943751   \n",
              "4  input01_5          20   32434638   \n",
              "5  input01_6          21   43094667   \n",
              "6  input01_7           8  100712766   \n",
              "7  input01_8           8  100715522   \n",
              "\n",
              "                                                    REF ALT    CHR        POS  \\\n",
              "0                                                     C   T   chr1  114716127   \n",
              "1                                                     G   A  chr16   67616834   \n",
              "2                                                     T   G  chr17    7669662   \n",
              "3  GCAGAGGCTTAAGGAGGAGGAAGAAGACAAGAAACGCAAAGAGGAGGAGGAG   -  chr19   12943750   \n",
              "4                                                     -   G  chr20   32434638   \n",
              "5                                                     T   G  chr21   43094667   \n",
              "6                                                     -   A   chr8  100712766   \n",
              "7                                                     T   C   chr8  100715522   \n",
              "\n",
              "  ALT_TYPE STRAND CGI-Sample ID CGI-Gene          CGI-Protein Change  \\\n",
              "0      snp      +       input01     NRAS                        G12S   \n",
              "1      snp      +       input01     CTCF                       E348K   \n",
              "2      snp      +       input01     TP53                       T377P   \n",
              "3    indel      +       input01     CALR  EQRLKEEEEDKKRKEEEE364-381X   \n",
              "4    indel      +       input01    ASXL1                   -642-643X   \n",
              "5      snp      +       input01    U2AF1                       Q157P   \n",
              "6    indel      +       input01   PABPC1                      K254NX   \n",
              "7      snp      +       input01   PABPC1                       I195V   \n",
              "\n",
              "                 CGI-Oncogenic Summary  \\\n",
              "0  oncogenic (predicted and annotated)   \n",
              "1                oncogenic (predicted)   \n",
              "2                        non-oncogenic   \n",
              "3                oncogenic (predicted)   \n",
              "4                oncogenic (predicted)   \n",
              "5  oncogenic (predicted and annotated)   \n",
              "6                oncogenic (predicted)   \n",
              "7                        non-oncogenic   \n",
              "\n",
              "                      CGI-Oncogenic Prediction  \\\n",
              "0  driver (boostDM: non-tissue-specific model)   \n",
              "1                        driver (oncodriveMUT)   \n",
              "2                     passenger (oncodriveMUT)   \n",
              "3                        driver (oncodriveMUT)   \n",
              "4                        driver (oncodriveMUT)   \n",
              "5  driver (boostDM: non-tissue-specific model)   \n",
              "6                        driver (oncodriveMUT)   \n",
              "7                     passenger (oncodriveMUT)   \n",
              "\n",
              "  CGI-External oncogenic annotation  \\\n",
              "0                cgi,clinvar:177778   \n",
              "1                               NaN   \n",
              "2                               NaN   \n",
              "3                               NaN   \n",
              "4                               NaN   \n",
              "5         cgi,oncokb,clinvar:376024   \n",
              "6                               NaN   \n",
              "7                               NaN   \n",
              "\n",
              "                                                                     CGI-Mutation  \\\n",
              "0                                                              chr1:114716127 C>T   \n",
              "1                                                              chr16:67616834 G>A   \n",
              "2                                                               chr17:7669662 T>G   \n",
              "3  chr19:12943751-12943751 GCAGAGGCTTAAGGAGGAGGAAGAAGACAAGAAACGCAAAGAGGAGGAGGAG>-   \n",
              "4                                                     chr20:32434638-32434639 ->G   \n",
              "5                                                              chr21:43094667 T>G   \n",
              "6                                                    chr8:100712766-100712767 ->A   \n",
              "7                                                              chr8:100715522 T>C   \n",
              "\n",
              "      CGI-Consequence   CGI-Transcript CGI-STRAND CGI-Type  \\\n",
              "0    missense_variant  ENST00000369535          +      SNV   \n",
              "1    missense_variant  ENST00000646076          +      SNV   \n",
              "2    missense_variant  ENST00000269305          +      SNV   \n",
              "3  frameshift_variant  ENST00000316448          +      DEL   \n",
              "4  frameshift_variant  ENST00000375687          +      INS   \n",
              "5    missense_variant  ENST00000291552          +      SNV   \n",
              "6  frameshift_variant  ENST00000523555          +      INS   \n",
              "7    missense_variant  ENST00000523555          +      SNV   \n",
              "\n",
              "                                                             CGI-HGVS  \\\n",
              "0                       ENST00000369535:c.34G>A;p.(Gly12Ser);p.(G12S)   \n",
              "1                   ENST00000646076:c.1042G>A;p.(Glu348Lys);p.(E348K)   \n",
              "2                   ENST00000269305:c.1129A>C;p.(Thr377Pro);p.(T377P)   \n",
              "3  ENST00000316448:c.1099_1150del;p.(Leu367ThrfsTer46);p.(L367Tfs*46)   \n",
              "4       ENST00000375687:c.1934dup;p.(Gly646TrpfsTer12);p.(G646Wfs*12)   \n",
              "5                    ENST00000291552:c.470A>C;p.(Gln157Pro);p.(Q157P)   \n",
              "6   ENST00000523555:c.761_762insT;p.(Lys254AsnfsTer24);p.(K254Nfs*24)   \n",
              "7                    ENST00000523555:c.583A>G;p.(Ile195Val);p.(I195V)   \n",
              "\n",
              "                           CGI-HGVSc                             CGI-HGVSp  \n",
              "0          ENST00000369535.5:c.34G>A          ENSP00000358548.4:p.Gly12Ser  \n",
              "1        ENST00000646076.1:c.1042G>A         ENSP00000494538.1:p.Glu348Lys  \n",
              "2        ENST00000269305.9:c.1129A>C         ENSP00000269305.4:p.Thr377Pro  \n",
              "3  ENST00000316448.10:c.1099_1150del  ENSP00000320866.4:p.Leu367ThrfsTer46  \n",
              "4       ENST00000375687.10:c.1934dup  ENSP00000364839.4:p.Gly646TrpfsTer12  \n",
              "5         ENST00000291552.9:c.470A>C         ENSP00000291552.4:p.Gln157Pro  \n",
              "6    ENST00000523555.6:c.761_762insT  ENSP00000429892.2:p.Lys254AsnfsTer24  \n",
              "7         ENST00000523555.6:c.583A>G         ENSP00000429892.2:p.Ile195Val  "
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-fef10141-b74a-408f-8e47-4b4727b9a850\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>Input ID</th>\n",
              "      <th>CHROMOSOME</th>\n",
              "      <th>POSITION</th>\n",
              "      <th>REF</th>\n",
              "      <th>ALT</th>\n",
              "      <th>CHR</th>\n",
              "      <th>POS</th>\n",
              "      <th>ALT_TYPE</th>\n",
              "      <th>STRAND</th>\n",
              "      <th>CGI-Sample ID</th>\n",
              "      <th>CGI-Gene</th>\n",
              "      <th>CGI-Protein Change</th>\n",
              "      <th>CGI-Oncogenic Summary</th>\n",
              "      <th>CGI-Oncogenic Prediction</th>\n",
              "      <th>CGI-External oncogenic annotation</th>\n",
              "      <th>CGI-Mutation</th>\n",
              "      <th>CGI-Consequence</th>\n",
              "      <th>CGI-Transcript</th>\n",
              "      <th>CGI-STRAND</th>\n",
              "      <th>CGI-Type</th>\n",
              "      <th>CGI-HGVS</th>\n",
              "      <th>CGI-HGVSc</th>\n",
              "      <th>CGI-HGVSp</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>input01_1</td>\n",
              "      <td>1</td>\n",
              "      <td>114716127</td>\n",
              "      <td>C</td>\n",
              "      <td>T</td>\n",
              "      <td>chr1</td>\n",
              "      <td>114716127</td>\n",
              "      <td>snp</td>\n",
              "      <td>+</td>\n",
              "      <td>input01</td>\n",
              "      <td>NRAS</td>\n",
              "      <td>G12S</td>\n",
              "      <td>oncogenic (predicted and annotated)</td>\n",
              "      <td>driver (boostDM: non-tissue-specific model)</td>\n",
              "      <td>cgi,clinvar:177778</td>\n",
              "      <td>chr1:114716127 C&gt;T</td>\n",
              "      <td>missense_variant</td>\n",
              "      <td>ENST00000369535</td>\n",
              "      <td>+</td>\n",
              "      <td>SNV</td>\n",
              "      <td>ENST00000369535:c.34G&gt;A;p.(Gly12Ser);p.(G12S)</td>\n",
              "      <td>ENST00000369535.5:c.34G&gt;A</td>\n",
              "      <td>ENSP00000358548.4:p.Gly12Ser</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>input01_2</td>\n",
              "      <td>16</td>\n",
              "      <td>67616834</td>\n",
              "      <td>G</td>\n",
              "      <td>A</td>\n",
              "      <td>chr16</td>\n",
              "      <td>67616834</td>\n",
              "      <td>snp</td>\n",
              "      <td>+</td>\n",
              "      <td>input01</td>\n",
              "      <td>CTCF</td>\n",
              "      <td>E348K</td>\n",
              "      <td>oncogenic (predicted)</td>\n",
              "      <td>driver (oncodriveMUT)</td>\n",
              "      <td>NaN</td>\n",
              "      <td>chr16:67616834 G&gt;A</td>\n",
              "      <td>missense_variant</td>\n",
              "      <td>ENST00000646076</td>\n",
              "      <td>+</td>\n",
              "      <td>SNV</td>\n",
              "      <td>ENST00000646076:c.1042G&gt;A;p.(Glu348Lys);p.(E348K)</td>\n",
              "      <td>ENST00000646076.1:c.1042G&gt;A</td>\n",
              "      <td>ENSP00000494538.1:p.Glu348Lys</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>input01_3</td>\n",
              "      <td>17</td>\n",
              "      <td>7669662</td>\n",
              "      <td>T</td>\n",
              "      <td>G</td>\n",
              "      <td>chr17</td>\n",
              "      <td>7669662</td>\n",
              "      <td>snp</td>\n",
              "      <td>+</td>\n",
              "      <td>input01</td>\n",
              "      <td>TP53</td>\n",
              "      <td>T377P</td>\n",
              "      <td>non-oncogenic</td>\n",
              "      <td>passenger (oncodriveMUT)</td>\n",
              "      <td>NaN</td>\n",
              "      <td>chr17:7669662 T&gt;G</td>\n",
              "      <td>missense_variant</td>\n",
              "      <td>ENST00000269305</td>\n",
              "      <td>+</td>\n",
              "      <td>SNV</td>\n",
              "      <td>ENST00000269305:c.1129A&gt;C;p.(Thr377Pro);p.(T377P)</td>\n",
              "      <td>ENST00000269305.9:c.1129A&gt;C</td>\n",
              "      <td>ENSP00000269305.4:p.Thr377Pro</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>input01_4</td>\n",
              "      <td>19</td>\n",
              "      <td>12943751</td>\n",
              "      <td>GCAGAGGCTTAAGGAGGAGGAAGAAGACAAGAAACGCAAAGAGGAGGAGGAG</td>\n",
              "      <td>-</td>\n",
              "      <td>chr19</td>\n",
              "      <td>12943750</td>\n",
              "      <td>indel</td>\n",
              "      <td>+</td>\n",
              "      <td>input01</td>\n",
              "      <td>CALR</td>\n",
              "      <td>EQRLKEEEEDKKRKEEEE364-381X</td>\n",
              "      <td>oncogenic (predicted)</td>\n",
              "      <td>driver (oncodriveMUT)</td>\n",
              "      <td>NaN</td>\n",
              "      <td>chr19:12943751-12943751 GCAGAGGCTTAAGGAGGAGGAAGAAGACAAGAAACGCAAAGAGGAGGAGGAG&gt;-</td>\n",
              "      <td>frameshift_variant</td>\n",
              "      <td>ENST00000316448</td>\n",
              "      <td>+</td>\n",
              "      <td>DEL</td>\n",
              "      <td>ENST00000316448:c.1099_1150del;p.(Leu367ThrfsTer46);p.(L367Tfs*46)</td>\n",
              "      <td>ENST00000316448.10:c.1099_1150del</td>\n",
              "      <td>ENSP00000320866.4:p.Leu367ThrfsTer46</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>input01_5</td>\n",
              "      <td>20</td>\n",
              "      <td>32434638</td>\n",
              "      <td>-</td>\n",
              "      <td>G</td>\n",
              "      <td>chr20</td>\n",
              "      <td>32434638</td>\n",
              "      <td>indel</td>\n",
              "      <td>+</td>\n",
              "      <td>input01</td>\n",
              "      <td>ASXL1</td>\n",
              "      <td>-642-643X</td>\n",
              "      <td>oncogenic (predicted)</td>\n",
              "      <td>driver (oncodriveMUT)</td>\n",
              "      <td>NaN</td>\n",
              "      <td>chr20:32434638-32434639 -&gt;G</td>\n",
              "      <td>frameshift_variant</td>\n",
              "      <td>ENST00000375687</td>\n",
              "      <td>+</td>\n",
              "      <td>INS</td>\n",
              "      <td>ENST00000375687:c.1934dup;p.(Gly646TrpfsTer12);p.(G646Wfs*12)</td>\n",
              "      <td>ENST00000375687.10:c.1934dup</td>\n",
              "      <td>ENSP00000364839.4:p.Gly646TrpfsTer12</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5</th>\n",
              "      <td>input01_6</td>\n",
              "      <td>21</td>\n",
              "      <td>43094667</td>\n",
              "      <td>T</td>\n",
              "      <td>G</td>\n",
              "      <td>chr21</td>\n",
              "      <td>43094667</td>\n",
              "      <td>snp</td>\n",
              "      <td>+</td>\n",
              "      <td>input01</td>\n",
              "      <td>U2AF1</td>\n",
              "      <td>Q157P</td>\n",
              "      <td>oncogenic (predicted and annotated)</td>\n",
              "      <td>driver (boostDM: non-tissue-specific model)</td>\n",
              "      <td>cgi,oncokb,clinvar:376024</td>\n",
              "      <td>chr21:43094667 T&gt;G</td>\n",
              "      <td>missense_variant</td>\n",
              "      <td>ENST00000291552</td>\n",
              "      <td>+</td>\n",
              "      <td>SNV</td>\n",
              "      <td>ENST00000291552:c.470A&gt;C;p.(Gln157Pro);p.(Q157P)</td>\n",
              "      <td>ENST00000291552.9:c.470A&gt;C</td>\n",
              "      <td>ENSP00000291552.4:p.Gln157Pro</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>6</th>\n",
              "      <td>input01_7</td>\n",
              "      <td>8</td>\n",
              "      <td>100712766</td>\n",
              "      <td>-</td>\n",
              "      <td>A</td>\n",
              "      <td>chr8</td>\n",
              "      <td>100712766</td>\n",
              "      <td>indel</td>\n",
              "      <td>+</td>\n",
              "      <td>input01</td>\n",
              "      <td>PABPC1</td>\n",
              "      <td>K254NX</td>\n",
              "      <td>oncogenic (predicted)</td>\n",
              "      <td>driver (oncodriveMUT)</td>\n",
              "      <td>NaN</td>\n",
              "      <td>chr8:100712766-100712767 -&gt;A</td>\n",
              "      <td>frameshift_variant</td>\n",
              "      <td>ENST00000523555</td>\n",
              "      <td>+</td>\n",
              "      <td>INS</td>\n",
              "      <td>ENST00000523555:c.761_762insT;p.(Lys254AsnfsTer24);p.(K254Nfs*24)</td>\n",
              "      <td>ENST00000523555.6:c.761_762insT</td>\n",
              "      <td>ENSP00000429892.2:p.Lys254AsnfsTer24</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>7</th>\n",
              "      <td>input01_8</td>\n",
              "      <td>8</td>\n",
              "      <td>100715522</td>\n",
              "      <td>T</td>\n",
              "      <td>C</td>\n",
              "      <td>chr8</td>\n",
              "      <td>100715522</td>\n",
              "      <td>snp</td>\n",
              "      <td>+</td>\n",
              "      <td>input01</td>\n",
              "      <td>PABPC1</td>\n",
              "      <td>I195V</td>\n",
              "      <td>non-oncogenic</td>\n",
              "      <td>passenger (oncodriveMUT)</td>\n",
              "      <td>NaN</td>\n",
              "      <td>chr8:100715522 T&gt;C</td>\n",
              "      <td>missense_variant</td>\n",
              "      <td>ENST00000523555</td>\n",
              "      <td>+</td>\n",
              "      <td>SNV</td>\n",
              "      <td>ENST00000523555:c.583A&gt;G;p.(Ile195Val);p.(I195V)</td>\n",
              "      <td>ENST00000523555.6:c.583A&gt;G</td>\n",
              "      <td>ENSP00000429892.2:p.Ile195Val</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-fef10141-b74a-408f-8e47-4b4727b9a850')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-fef10141-b74a-408f-8e47-4b4727b9a850 button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-fef10141-b74a-408f-8e47-4b4727b9a850');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "<div id=\"df-928aa4a7-a7e7-47e7-9685-e6d321063e01\">\n",
              "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-928aa4a7-a7e7-47e7-9685-e6d321063e01')\"\n",
              "            title=\"Suggest charts\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "  </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "  <script>\n",
              "    async function quickchart(key) {\n",
              "      const quickchartButtonEl =\n",
              "        document.querySelector('#' + key + ' button');\n",
              "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "      try {\n",
              "        const charts = await google.colab.kernel.invokeFunction(\n",
              "            'suggestCharts', [key], {});\n",
              "      } catch (error) {\n",
              "        console.error('Error during call to suggestCharts:', error);\n",
              "      }\n",
              "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "    }\n",
              "    (() => {\n",
              "      let quickchartButtonEl =\n",
              "        document.querySelector('#df-928aa4a7-a7e7-47e7-9685-e6d321063e01 button');\n",
              "      quickchartButtonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "    })();\n",
              "  </script>\n",
              "</div>\n",
              "\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe"
            }
          },
          "metadata": {},
          "execution_count": 24
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Resultado: biomarkers.tsv"
      ],
      "metadata": {
        "id": "gQO69WqTYhp2"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "#Vai carregar o conteúdo do arquivo biomarkers.tsv no diretório /content/WP017/ e retornará um DataFrame.\n",
        "pd.read_csv('/content/WP017/biomarkers.tsv',sep='\\t',index_col=False, engine= 'python')"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 721
        },
        "id": "xIlNkJj8YiEi",
        "outputId": "877df811-86c1-480a-cb90-c1ce0eddb33d"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "   Sample ID                  Alterations                   Biomarker  \\\n",
              "0    input01                  NRAS (G12S)  NRAS (12,13,59,61,117,146)   \n",
              "1    input01                  NRAS (G12S)  NRAS (12,13,59,61,117,146)   \n",
              "2    input01                KRAS wildtype               KRAS wildtype   \n",
              "3    input01                KRAS wildtype               KRAS wildtype   \n",
              "4    input01  EGFR wildtype, ALK wildtype  EGFR wildtype;ALK wildtype   \n",
              "..       ...                          ...                         ...   \n",
              "91   input01            TP53 MUT* (T377P)                   TP53 R249   \n",
              "92   input01            TP53 MUT* (T377P)                  TP53 R273H   \n",
              "93   input01            TP53 MUT* (T377P)                  TP53 R273L   \n",
              "94   input01            TP53 MUT* (T377P)                  TP53 Y234C   \n",
              "95   input01            TP53 MUT* (T377P)                   TP53 Loss   \n",
              "\n",
              "                                                                                                                      Drugs  \\\n",
              "0                                                                                             Cetuximab(EGFR mAb inhibitor)   \n",
              "1                                                                                           Panitumumab(EGFR mAb inhibitor)   \n",
              "2                                                                                           Panitumumab(EGFR mAb inhibitor)   \n",
              "3                                                                                             Cetuximab(EGFR mAb inhibitor)   \n",
              "4   Atezolizumab(PDL1 blocking antibody);Bevacizumab(VEGF mAb inhibitor);Paclitaxel(Chemotherapy);Carboplatin(Chemotherapy)   \n",
              "..                                                                                                                      ...   \n",
              "91                                                                                                              Doxorubicin   \n",
              "92                                                                                                 Methotrexate,Doxorubicin   \n",
              "93                                                                                                    Cisplatin,Carboplatin   \n",
              "94                                                                                                    Cisplatin,Carboplatin   \n",
              "95                                                                                                                Docetaxel   \n",
              "\n",
              "                     Diseases    Response Evidence Match  \\\n",
              "0   Colorectal adenocarcinoma   Resistant        A    NO   \n",
              "1   Colorectal adenocarcinoma   Resistant        A    NO   \n",
              "2   Colorectal adenocarcinoma  Responsive        A    NO   \n",
              "3   Colorectal adenocarcinoma  Responsive        A    NO   \n",
              "4         Non-small cell lung  Responsive        A    NO   \n",
              "..                        ...         ...      ...   ...   \n",
              "91      Breast adenocarcinoma  Responsive        B    NO   \n",
              "92               Osteosarcoma   Resistant        D    NO   \n",
              "93                      Ovary   Resistant        C    NO   \n",
              "94                      Ovary   Resistant        C    NO   \n",
              "95        Non-small cell lung   Resistant        D    NO   \n",
              "\n",
              "                                                                              Source  \\\n",
              "0                                          PMID:24024839;PMID:20619739;PMID:23325582   \n",
              "1                                                                     FDA guidelines   \n",
              "2   FDA  https://www.accessdata.fda.gov/drugsatfda_docs/label/2021/125147s210lbl.pdf   \n",
              "3                                                                     PMID: 19339720   \n",
              "4    FDA https://www.accessdata.fda.gov/drugsatfda_docs/label/2022/761034s047lbl.pdf   \n",
              "..                                                                               ...   \n",
              "91                                      https://civicdb.org/links/evidence_items/399   \n",
              "92                                     https://civicdb.org/links/evidence_items/7430   \n",
              "93                                     https://civicdb.org/links/evidence_items/2286   \n",
              "94                                     https://civicdb.org/links/evidence_items/2745   \n",
              "95                                     https://civicdb.org/links/evidence_items/1146   \n",
              "\n",
              "                    BioM  Resist. Tumor type  \n",
              "0               complete      NaN     COREAD  \n",
              "1               complete      NaN     COREAD  \n",
              "2               complete     65.0     COREAD  \n",
              "3               complete     22.0     COREAD  \n",
              "4               complete      NaN      NSCLC  \n",
              "..                   ...      ...        ...  \n",
              "91  only alteration type      NaN       BRCA  \n",
              "92  only alteration type      NaN         OS  \n",
              "93  only alteration type      NaN         OV  \n",
              "94  only alteration type      NaN         OV  \n",
              "95             only gene      NaN      NSCLC  \n",
              "\n",
              "[96 rows x 12 columns]"
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-2c2a7dc5-8c85-42a6-9e59-f4a86c9c82ac\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>Sample ID</th>\n",
              "      <th>Alterations</th>\n",
              "      <th>Biomarker</th>\n",
              "      <th>Drugs</th>\n",
              "      <th>Diseases</th>\n",
              "      <th>Response</th>\n",
              "      <th>Evidence</th>\n",
              "      <th>Match</th>\n",
              "      <th>Source</th>\n",
              "      <th>BioM</th>\n",
              "      <th>Resist.</th>\n",
              "      <th>Tumor type</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>input01</td>\n",
              "      <td>NRAS (G12S)</td>\n",
              "      <td>NRAS (12,13,59,61,117,146)</td>\n",
              "      <td>Cetuximab(EGFR mAb inhibitor)</td>\n",
              "      <td>Colorectal adenocarcinoma</td>\n",
              "      <td>Resistant</td>\n",
              "      <td>A</td>\n",
              "      <td>NO</td>\n",
              "      <td>PMID:24024839;PMID:20619739;PMID:23325582</td>\n",
              "      <td>complete</td>\n",
              "      <td>NaN</td>\n",
              "      <td>COREAD</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>input01</td>\n",
              "      <td>NRAS (G12S)</td>\n",
              "      <td>NRAS (12,13,59,61,117,146)</td>\n",
              "      <td>Panitumumab(EGFR mAb inhibitor)</td>\n",
              "      <td>Colorectal adenocarcinoma</td>\n",
              "      <td>Resistant</td>\n",
              "      <td>A</td>\n",
              "      <td>NO</td>\n",
              "      <td>FDA guidelines</td>\n",
              "      <td>complete</td>\n",
              "      <td>NaN</td>\n",
              "      <td>COREAD</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>input01</td>\n",
              "      <td>KRAS wildtype</td>\n",
              "      <td>KRAS wildtype</td>\n",
              "      <td>Panitumumab(EGFR mAb inhibitor)</td>\n",
              "      <td>Colorectal adenocarcinoma</td>\n",
              "      <td>Responsive</td>\n",
              "      <td>A</td>\n",
              "      <td>NO</td>\n",
              "      <td>FDA  https://www.accessdata.fda.gov/drugsatfda_docs/label/2021/125147s210lbl.pdf</td>\n",
              "      <td>complete</td>\n",
              "      <td>65.0</td>\n",
              "      <td>COREAD</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>input01</td>\n",
              "      <td>KRAS wildtype</td>\n",
              "      <td>KRAS wildtype</td>\n",
              "      <td>Cetuximab(EGFR mAb inhibitor)</td>\n",
              "      <td>Colorectal adenocarcinoma</td>\n",
              "      <td>Responsive</td>\n",
              "      <td>A</td>\n",
              "      <td>NO</td>\n",
              "      <td>PMID: 19339720</td>\n",
              "      <td>complete</td>\n",
              "      <td>22.0</td>\n",
              "      <td>COREAD</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>input01</td>\n",
              "      <td>EGFR wildtype, ALK wildtype</td>\n",
              "      <td>EGFR wildtype;ALK wildtype</td>\n",
              "      <td>Atezolizumab(PDL1 blocking antibody);Bevacizumab(VEGF mAb inhibitor);Paclitaxel(Chemotherapy);Carboplatin(Chemotherapy)</td>\n",
              "      <td>Non-small cell lung</td>\n",
              "      <td>Responsive</td>\n",
              "      <td>A</td>\n",
              "      <td>NO</td>\n",
              "      <td>FDA https://www.accessdata.fda.gov/drugsatfda_docs/label/2022/761034s047lbl.pdf</td>\n",
              "      <td>complete</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NSCLC</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>91</th>\n",
              "      <td>input01</td>\n",
              "      <td>TP53 MUT* (T377P)</td>\n",
              "      <td>TP53 R249</td>\n",
              "      <td>Doxorubicin</td>\n",
              "      <td>Breast adenocarcinoma</td>\n",
              "      <td>Responsive</td>\n",
              "      <td>B</td>\n",
              "      <td>NO</td>\n",
              "      <td>https://civicdb.org/links/evidence_items/399</td>\n",
              "      <td>only alteration type</td>\n",
              "      <td>NaN</td>\n",
              "      <td>BRCA</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>92</th>\n",
              "      <td>input01</td>\n",
              "      <td>TP53 MUT* (T377P)</td>\n",
              "      <td>TP53 R273H</td>\n",
              "      <td>Methotrexate,Doxorubicin</td>\n",
              "      <td>Osteosarcoma</td>\n",
              "      <td>Resistant</td>\n",
              "      <td>D</td>\n",
              "      <td>NO</td>\n",
              "      <td>https://civicdb.org/links/evidence_items/7430</td>\n",
              "      <td>only alteration type</td>\n",
              "      <td>NaN</td>\n",
              "      <td>OS</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>93</th>\n",
              "      <td>input01</td>\n",
              "      <td>TP53 MUT* (T377P)</td>\n",
              "      <td>TP53 R273L</td>\n",
              "      <td>Cisplatin,Carboplatin</td>\n",
              "      <td>Ovary</td>\n",
              "      <td>Resistant</td>\n",
              "      <td>C</td>\n",
              "      <td>NO</td>\n",
              "      <td>https://civicdb.org/links/evidence_items/2286</td>\n",
              "      <td>only alteration type</td>\n",
              "      <td>NaN</td>\n",
              "      <td>OV</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>94</th>\n",
              "      <td>input01</td>\n",
              "      <td>TP53 MUT* (T377P)</td>\n",
              "      <td>TP53 Y234C</td>\n",
              "      <td>Cisplatin,Carboplatin</td>\n",
              "      <td>Ovary</td>\n",
              "      <td>Resistant</td>\n",
              "      <td>C</td>\n",
              "      <td>NO</td>\n",
              "      <td>https://civicdb.org/links/evidence_items/2745</td>\n",
              "      <td>only alteration type</td>\n",
              "      <td>NaN</td>\n",
              "      <td>OV</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>95</th>\n",
              "      <td>input01</td>\n",
              "      <td>TP53 MUT* (T377P)</td>\n",
              "      <td>TP53 Loss</td>\n",
              "      <td>Docetaxel</td>\n",
              "      <td>Non-small cell lung</td>\n",
              "      <td>Resistant</td>\n",
              "      <td>D</td>\n",
              "      <td>NO</td>\n",
              "      <td>https://civicdb.org/links/evidence_items/1146</td>\n",
              "      <td>only gene</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NSCLC</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>96 rows × 12 columns</p>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-2c2a7dc5-8c85-42a6-9e59-f4a86c9c82ac')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-2c2a7dc5-8c85-42a6-9e59-f4a86c9c82ac button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-2c2a7dc5-8c85-42a6-9e59-f4a86c9c82ac');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "<div id=\"df-eb60876b-2b20-4a9a-9cf0-250daa0fc187\">\n",
              "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-eb60876b-2b20-4a9a-9cf0-250daa0fc187')\"\n",
              "            title=\"Suggest charts\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "  </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "  <script>\n",
              "    async function quickchart(key) {\n",
              "      const quickchartButtonEl =\n",
              "        document.querySelector('#' + key + ' button');\n",
              "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "      try {\n",
              "        const charts = await google.colab.kernel.invokeFunction(\n",
              "            'suggestCharts', [key], {});\n",
              "      } catch (error) {\n",
              "        console.error('Error during call to suggestCharts:', error);\n",
              "      }\n",
              "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "    }\n",
              "    (() => {\n",
              "      let quickchartButtonEl =\n",
              "        document.querySelector('#df-eb60876b-2b20-4a9a-9cf0-250daa0fc187 button');\n",
              "      quickchartButtonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "    })();\n",
              "  </script>\n",
              "</div>\n",
              "\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe",
              "summary": "{\n  \"name\": \"pd\",\n  \"rows\": 96,\n  \"fields\": [\n    {\n      \"column\": \"Sample ID\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 1,\n        \"samples\": [\n          \"input01\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Alterations\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 11,\n        \"samples\": [\n          \"KIT wildtype\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Biomarker\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 38,\n        \"samples\": [\n          \"TP53 R249\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Drugs\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 66,\n        \"samples\": [\n          \"Selumetinib,Trametinib\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Diseases\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 25,\n        \"samples\": [\n          \"Female germ cell tumor, Male germ cell tumor\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Response\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 2,\n        \"samples\": [\n          \"Responsive\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Evidence\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 4,\n        \"samples\": [\n          \"C\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Match\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 2,\n        \"samples\": [\n          \"YES\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Source\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 80,\n        \"samples\": [\n          \"PMID:20179705\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"BioM\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 3,\n        \"samples\": [\n          \"complete\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Resist.\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 30.405591591021544,\n        \"min\": 22.0,\n        \"max\": 65.0,\n        \"num_unique_values\": 2,\n        \"samples\": [\n          22.0\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Tumor type\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 25,\n        \"samples\": [\n          \"FGCT, MGCT\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}"
            }
          },
          "metadata": {},
          "execution_count": 35
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "# **WP019**"
      ],
      "metadata": {
        "id": "-ziLcS05fRi7"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "#WP019\n",
        "#Código na célula será interpretado como comandos Bash.\n",
        "%%bash\n",
        "#Extrai as quatro primeiras colunas do arquivo, substitui o texto CHROM por CHR nas colunas extraídas, salva o resultado no arquivo e exisbe as 10 primeiras linhas do arquivo.\n",
        "cut -f1-4 /content/lmabrasil-hg38/vep_output/liftOver_WP019_hg19ToHg38.vep.filter.tsv | sed -e \"s/CHROM/CHR/g\"  > df_WP019-cgi.txt\n",
        "head df_WP019-cgi.txt"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "tuoPtr4UcTw2",
        "outputId": "6c066cf5-3566-4f41-bbf5-bf30426f567c"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "CHR\tPOS\tREF\tALT\n",
            "chr1\t179120104\tGA\tG\n",
            "chr17\t76736877\tG\tT\n",
            "chr9\t5073770\tG\tT\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "import requests\n",
        "#Contém informações de autenticação.\n",
        "headers = {'Authorization': 'silfrizzarini@gmail.com 7f1b6ddf77bc5186c964'}\n",
        "#Dicionário que contém os dados a serem enviados no corpo da requisição.\n",
        "payload = {'cancer_type': 'HEMATO', 'title': 'Somatic MF', 'reference': 'hg38'}\n",
        "#Envia a requisição.\n",
        "r = requests.post('https://www.cancergenomeinterpreter.org/api/v1',\n",
        "                headers=headers,\n",
        "                files={\n",
        "                        'mutations': open('/content/df_WP019-cgi.txt', 'rb')\n",
        "                        },\n",
        "                data=payload)\n",
        "#A resposta da requisição é armazenada na variável r. Converte a resposta em formato JSON.\n",
        "r.json()"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 35
        },
        "id": "s7fP8pMZcm1D",
        "outputId": "079e0220-99b3-48a3-9b40-d9ae2aa008d3"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "'a2d390dbddab7739127d'"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "string"
            }
          },
          "metadata": {},
          "execution_count": 27
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#Envia uma requisição para o Cancer Genome Interpreter para obter os resultados de uma análise, converte a resposta para o formato JSON e retorna informações sobre as variantes genéticas analisadas.\n",
        "import requests\n",
        "job_id =\"a2d390dbddab7739127d\"\n",
        "\n",
        "headers = {'Authorization': 'silfrizzarini@gmail.com 7f1b6ddf77bc5186c964'}\n",
        "r = requests.get('https://www.cancergenomeinterpreter.org/api/v1/%s' % job_id, headers=headers)\n",
        "r.json()"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "J7J5rqLBcxjJ",
        "outputId": "528eda85-64f8-4d76-abeb-a07d15c051b4"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "{'status': 'Done',\n",
              " 'metadata': {'id': 'a2d390dbddab7739127d',\n",
              "  'user': 'silfrizzarini@gmail.com',\n",
              "  'title': 'Somatic MF',\n",
              "  'cancertype': 'HEMATO',\n",
              "  'reference': 'hg38',\n",
              "  'dataset': 'input.tsv',\n",
              "  'date': '2024-12-21 13:40:30'}}"
            ]
          },
          "metadata": {},
          "execution_count": 28
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#Envia uma requisição para o Cancer Genome Interpreter para obter os logs da análise, retorna em formato JSON e mostra detalhes sobre o andamento ou erros da análise.\n",
        "import requests\n",
        "job_id =\"a2d390dbddab7739127d\"\n",
        "\n",
        "headers = {'Authorization': 'silfrizzarini@gmail.com 7f1b6ddf77bc5186c964'}\n",
        "payload={'action':'logs'}\n",
        "r = requests.get('https://www.cancergenomeinterpreter.org/api/v1/%s' % job_id, headers=headers, params=payload)\n",
        "r.json()"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "B4EpAmO0c-yF",
        "outputId": "516bdf6b-d87e-41f2-89b8-6fccf2510981"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "{'status': 'Done',\n",
              " 'logs': ['# cgi analyze input.tsv -c HEMATO -g hg38',\n",
              "  '2024-12-21 14:40:34,366 INFO     Parsing input01.tsv\\n',\n",
              "  '2024-12-21 14:40:37,981 INFO     Running VEP\\n',\n",
              "  '2024-12-21 14:40:38,998 INFO     Check cancer genes and consensus roles\\n',\n",
              "  '2024-12-21 14:40:39,075 INFO     Annotate BoostDM mutations\\n',\n",
              "  '2024-12-21 14:40:39,123 INFO     Annotate OncodriveMUT mutations\\n',\n",
              "  '2024-12-21 14:40:41,321 INFO     Annotate validated oncogenic mutations\\n',\n",
              "  '2024-12-21 14:40:41,459 INFO     Check oncogenic classification\\n',\n",
              "  '2024-12-21 14:40:41,519 INFO     Matching biomarkers\\n',\n",
              "  '2024-12-21 14:40:41,604 INFO     Prescription finished\\n',\n",
              "  '2024-12-21 14:40:41,614 INFO     Aggregate metrics\\n',\n",
              "  '2024-12-21 14:40:47,498 INFO     Compress output files\\n',\n",
              "  '2024-12-21 14:40:47,552 INFO     Analysis done\\n']}"
            ]
          },
          "metadata": {},
          "execution_count": 29
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#Baixa os resultados da análise em file.zip, fica disponível no diretório de trabalho.\n",
        "import requests\n",
        "job_id =\"a2d390dbddab7739127d\"\n",
        "\n",
        "headers = {'Authorization': 'silfrizzarini@gmail.com 7f1b6ddf77bc5186c964'}\n",
        "payload={'action':'download'}\n",
        "r = requests.get('https://www.cancergenomeinterpreter.org/api/v1/%s' % job_id, headers=headers, params=payload)\n",
        "with open('file.zip', 'wb') as fd:\n",
        "    fd.write(r._content)"
      ],
      "metadata": {
        "id": "sQ6Zs8yYdSTO"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "#Permite confirmar o tipo do arquivo e garantir que ele tenha sido baixado corretamente.\n",
        "!file file.zip"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "2Z6FABVbdhLw",
        "outputId": "6d23a1ac-b28c-499c-99b8-f359abaa8764"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "file.zip: Zip archive data, at least v2.0 to extract, compression method=deflate\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#Irá descompactar o conteúdo do arquivo file.zip e colocá-lo dentro de um diretório chamado WP019.\n",
        "!unzip -d WP019 file.zip"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "wMFLMMlUdnV3",
        "outputId": "15f2685c-530d-43f4-9e85-295cc6e14096"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Archive:  file.zip\n",
            "  inflating: WP019/alterations.tsv   \n",
            "  inflating: WP019/biomarkers.tsv    \n",
            "  inflating: WP019/input01.tsv       \n",
            "  inflating: WP019/summary.txt       \n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#Lê o arquivo alterations.tsv que está no diretório /content/WP019/, usando tabulação. Retorna um DataFrame contendo os dados do arquivo.\n",
        "pd.read_csv('/content/WP019/alterations.tsv',sep='\\t',index_col=False, engine= 'python')"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 285
        },
        "id": "NyV0B57udyGc",
        "outputId": "b2b9df43-8bb4-401a-cb50-0cc3251655e8"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "    Input ID  CHROMOSOME   POSITION REF ALT    CHR        POS ALT_TYPE STRAND  \\\n",
              "0  input01_1           1  179120105   A   -   chr1  179120104    indel      +   \n",
              "1  input01_2          17   76736877   G   T  chr17   76736877      snp      +   \n",
              "2  input01_3           9    5073770   G   T   chr9    5073770      snp      +   \n",
              "\n",
              "  CGI-Sample ID CGI-Gene CGI-Protein Change  \\\n",
              "0       input01     ABL2                 --   \n",
              "1       input01    SRSF2               P95H   \n",
              "2       input01     JAK2              V617F   \n",
              "\n",
              "                 CGI-Oncogenic Summary  CGI-Oncogenic Prediction  \\\n",
              "0                non-protein affecting     non-protein affecting   \n",
              "1  oncogenic (predicted and annotated)     driver (oncodriveMUT)   \n",
              "2                oncogenic (annotated)  passenger (oncodriveMUT)   \n",
              "\n",
              "  CGI-External oncogenic annotation                  CGI-Mutation  \\\n",
              "0                               NaN  chr1:179120105-179120105 A>-   \n",
              "1                        cgi,oncokb            chr17:76736877 G>T   \n",
              "2          cgi,oncokb,clinvar:14662              chr9:5073770 G>T   \n",
              "\n",
              "    CGI-Consequence   CGI-Transcript CGI-STRAND CGI-Type  \\\n",
              "0    intron_variant  ENST00000502732          +      DEL   \n",
              "1  missense_variant  ENST00000392485          +      SNV   \n",
              "2  missense_variant  ENST00000381652          +      SNV   \n",
              "\n",
              "                                            CGI-HGVS  \\\n",
              "0                  ENST00000502732:c.1045+85del;p(?)   \n",
              "1     ENST00000392485:c.284C>A;p.(Pro95His);p.(P95H)   \n",
              "2  ENST00000381652:c.1849G>T;p.(Val617Phe);p.(V617F)   \n",
              "\n",
              "                        CGI-HGVSc                      CGI-HGVSp  \n",
              "0  ENST00000502732.6:c.1045+85del                              -  \n",
              "1      ENST00000392485.2:c.284C>A   ENSP00000376276.2:p.Pro95His  \n",
              "2     ENST00000381652.4:c.1849G>T  ENSP00000371067.4:p.Val617Phe  "
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-fced02b4-f5a6-4ca7-9e98-ab77480b4a9f\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>Input ID</th>\n",
              "      <th>CHROMOSOME</th>\n",
              "      <th>POSITION</th>\n",
              "      <th>REF</th>\n",
              "      <th>ALT</th>\n",
              "      <th>CHR</th>\n",
              "      <th>POS</th>\n",
              "      <th>ALT_TYPE</th>\n",
              "      <th>STRAND</th>\n",
              "      <th>CGI-Sample ID</th>\n",
              "      <th>CGI-Gene</th>\n",
              "      <th>CGI-Protein Change</th>\n",
              "      <th>CGI-Oncogenic Summary</th>\n",
              "      <th>CGI-Oncogenic Prediction</th>\n",
              "      <th>CGI-External oncogenic annotation</th>\n",
              "      <th>CGI-Mutation</th>\n",
              "      <th>CGI-Consequence</th>\n",
              "      <th>CGI-Transcript</th>\n",
              "      <th>CGI-STRAND</th>\n",
              "      <th>CGI-Type</th>\n",
              "      <th>CGI-HGVS</th>\n",
              "      <th>CGI-HGVSc</th>\n",
              "      <th>CGI-HGVSp</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>input01_1</td>\n",
              "      <td>1</td>\n",
              "      <td>179120105</td>\n",
              "      <td>A</td>\n",
              "      <td>-</td>\n",
              "      <td>chr1</td>\n",
              "      <td>179120104</td>\n",
              "      <td>indel</td>\n",
              "      <td>+</td>\n",
              "      <td>input01</td>\n",
              "      <td>ABL2</td>\n",
              "      <td>--</td>\n",
              "      <td>non-protein affecting</td>\n",
              "      <td>non-protein affecting</td>\n",
              "      <td>NaN</td>\n",
              "      <td>chr1:179120105-179120105 A&gt;-</td>\n",
              "      <td>intron_variant</td>\n",
              "      <td>ENST00000502732</td>\n",
              "      <td>+</td>\n",
              "      <td>DEL</td>\n",
              "      <td>ENST00000502732:c.1045+85del;p(?)</td>\n",
              "      <td>ENST00000502732.6:c.1045+85del</td>\n",
              "      <td>-</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>input01_2</td>\n",
              "      <td>17</td>\n",
              "      <td>76736877</td>\n",
              "      <td>G</td>\n",
              "      <td>T</td>\n",
              "      <td>chr17</td>\n",
              "      <td>76736877</td>\n",
              "      <td>snp</td>\n",
              "      <td>+</td>\n",
              "      <td>input01</td>\n",
              "      <td>SRSF2</td>\n",
              "      <td>P95H</td>\n",
              "      <td>oncogenic (predicted and annotated)</td>\n",
              "      <td>driver (oncodriveMUT)</td>\n",
              "      <td>cgi,oncokb</td>\n",
              "      <td>chr17:76736877 G&gt;T</td>\n",
              "      <td>missense_variant</td>\n",
              "      <td>ENST00000392485</td>\n",
              "      <td>+</td>\n",
              "      <td>SNV</td>\n",
              "      <td>ENST00000392485:c.284C&gt;A;p.(Pro95His);p.(P95H)</td>\n",
              "      <td>ENST00000392485.2:c.284C&gt;A</td>\n",
              "      <td>ENSP00000376276.2:p.Pro95His</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>input01_3</td>\n",
              "      <td>9</td>\n",
              "      <td>5073770</td>\n",
              "      <td>G</td>\n",
              "      <td>T</td>\n",
              "      <td>chr9</td>\n",
              "      <td>5073770</td>\n",
              "      <td>snp</td>\n",
              "      <td>+</td>\n",
              "      <td>input01</td>\n",
              "      <td>JAK2</td>\n",
              "      <td>V617F</td>\n",
              "      <td>oncogenic (annotated)</td>\n",
              "      <td>passenger (oncodriveMUT)</td>\n",
              "      <td>cgi,oncokb,clinvar:14662</td>\n",
              "      <td>chr9:5073770 G&gt;T</td>\n",
              "      <td>missense_variant</td>\n",
              "      <td>ENST00000381652</td>\n",
              "      <td>+</td>\n",
              "      <td>SNV</td>\n",
              "      <td>ENST00000381652:c.1849G&gt;T;p.(Val617Phe);p.(V617F)</td>\n",
              "      <td>ENST00000381652.4:c.1849G&gt;T</td>\n",
              "      <td>ENSP00000371067.4:p.Val617Phe</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-fced02b4-f5a6-4ca7-9e98-ab77480b4a9f')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-fced02b4-f5a6-4ca7-9e98-ab77480b4a9f button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-fced02b4-f5a6-4ca7-9e98-ab77480b4a9f');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "<div id=\"df-403a2cc7-e7c4-41d5-b7ea-a087b9fd8f85\">\n",
              "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-403a2cc7-e7c4-41d5-b7ea-a087b9fd8f85')\"\n",
              "            title=\"Suggest charts\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "  </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "  <script>\n",
              "    async function quickchart(key) {\n",
              "      const quickchartButtonEl =\n",
              "        document.querySelector('#' + key + ' button');\n",
              "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "      try {\n",
              "        const charts = await google.colab.kernel.invokeFunction(\n",
              "            'suggestCharts', [key], {});\n",
              "      } catch (error) {\n",
              "        console.error('Error during call to suggestCharts:', error);\n",
              "      }\n",
              "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "    }\n",
              "    (() => {\n",
              "      let quickchartButtonEl =\n",
              "        document.querySelector('#df-403a2cc7-e7c4-41d5-b7ea-a087b9fd8f85 button');\n",
              "      quickchartButtonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "    })();\n",
              "  </script>\n",
              "</div>\n",
              "\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe"
            }
          },
          "metadata": {},
          "execution_count": 33
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#Vai carregar o conteúdo do arquivo biomarkers.tsv no diretório /content/WP019/ e retornará um DataFrame.\n",
        "pd.read_csv('/content/WP019/biomarkers.tsv',sep='\\t',index_col=False, engine= 'python')"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        },
        "id": "S_egXjeRd8n9",
        "outputId": "3d5c9c2d-d295-415b-eb53-9f3c059efd08"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "   Sample ID                                 Alterations  \\\n",
              "0    input01                                JAK2 (V617F)   \n",
              "1    input01                               KRAS wildtype   \n",
              "2    input01                               KRAS wildtype   \n",
              "3    input01                 EGFR wildtype, ALK wildtype   \n",
              "4    input01                 EGFR wildtype, ALK wildtype   \n",
              "5    input01                 EGFR wildtype, ALK wildtype   \n",
              "6    input01                 EGFR wildtype, ALK wildtype   \n",
              "7    input01  EGFR wildtype, ALK wildtype, ROS1 wildtype   \n",
              "8    input01                               NRAS wildtype   \n",
              "9    input01                                JAK2 (V617F)   \n",
              "10   input01                                JAK2 (V617F)   \n",
              "11   input01                                JAK2 (V617F)   \n",
              "12   input01                                JAK2 (V617F)   \n",
              "13   input01                                JAK2 (V617F)   \n",
              "14   input01                                KIT wildtype   \n",
              "15   input01                                KIT wildtype   \n",
              "16   input01                                KIT wildtype   \n",
              "17   input01                                KIT wildtype   \n",
              "18   input01                             PDGFRA wildtype   \n",
              "19   input01                             PDGFRA wildtype   \n",
              "20   input01                             PDGFRA wildtype   \n",
              "21   input01                             PDGFRA wildtype   \n",
              "22   input01                                SRSF2 (P95H)   \n",
              "23   input01                               TP53 wildtype   \n",
              "24   input01                                JAK2 (V617F)   \n",
              "25   input01                               BRAF wildtype   \n",
              "26   input01                               EGFR wildtype   \n",
              "27   input01                                JAK2 (V617F)   \n",
              "28   input01                                KIT wildtype   \n",
              "29   input01                               KRAS wildtype   \n",
              "30   input01                             PIK3CA wildtype   \n",
              "31   input01                               TP53 wildtype   \n",
              "32   input01                               TP53 wildtype   \n",
              "\n",
              "                                   Biomarker  \\\n",
              "0                               JAK2 (V617F)   \n",
              "1                              KRAS wildtype   \n",
              "2                              KRAS wildtype   \n",
              "3                 EGFR wildtype;ALK wildtype   \n",
              "4                 EGFR wildtype;ALK wildtype   \n",
              "5                 EGFR wildtype;ALK wildtype   \n",
              "6                 EGFR wildtype;ALK wildtype   \n",
              "7   EGFR wildtype;ALK wildtype;ROS1 wildtype   \n",
              "8                              NRAS wildtype   \n",
              "9                    JAK2 oncogenic mutation   \n",
              "10                              JAK2 (V617F)   \n",
              "11                        JAK2 amplification   \n",
              "12                          JAK2-BRAF fusion   \n",
              "13                              JAK2 (V617F)   \n",
              "14                              KIT wildtype   \n",
              "15                              KIT wildtype   \n",
              "16                              KIT wildtype   \n",
              "17                              KIT wildtype   \n",
              "18                           PDGFRA wildtype   \n",
              "19                           PDGFRA wildtype   \n",
              "20                           PDGFRA wildtype   \n",
              "21                           PDGFRA wildtype   \n",
              "22                  SRSF2 oncogenic mutation   \n",
              "23                             TP53 wildtype   \n",
              "24                          JAK2-PCM1 fusion   \n",
              "25                            BRAF WILD TYPE   \n",
              "26                             EGFR Wildtype   \n",
              "27            JAK2 Splice Site (c.1641+2T>G)   \n",
              "28                              KIT WILDTYPE   \n",
              "29                             KRAS Wildtype   \n",
              "30                           PIK3CA Wildtype   \n",
              "31                             TP53 Wildtype   \n",
              "32                             TP53 Wildtype   \n",
              "\n",
              "                                                                                                                      Drugs  \\\n",
              "0                                                                                                Ruxolitinib(JAK inhibitor)   \n",
              "1                                                                                           Panitumumab(EGFR mAb inhibitor)   \n",
              "2                                                                                             Cetuximab(EGFR mAb inhibitor)   \n",
              "3   Atezolizumab(PDL1 blocking antibody);Bevacizumab(VEGF mAb inhibitor);Paclitaxel(Chemotherapy);Carboplatin(Chemotherapy)   \n",
              "4                                   Atezolizumab(PDL1 blocking antibody);Paclitaxel(Chemotherapy);Carboplatin(Chemotherapy)   \n",
              "5                                     Pembrolizumab(PD-1-blocking antibody);Pemetrexed(Chemotherapy);Platinum(Chemotherapy)   \n",
              "6                     Durvalumab(PD-1-blocking antibody);Tremelimumab-actl(CTLA-4 blocking antibody);Platinum(Chemotherapy)   \n",
              "7                                                            Cemiplimab-rwlc(PD-1-blocking antibody);Platinum(Chemotherapy)   \n",
              "8                                                                                           Panitumumab(EGFR mAb inhibitor)   \n",
              "9                                                                                                            PD1 inhibitors   \n",
              "10                                                                                  JAK inhibitor(alone or in combination)s   \n",
              "11                                                                                                           JAK inhibitors   \n",
              "12                                                                                               Ruxolitinib(JAK inhibitor)   \n",
              "13                                                                                               Ruxolitinib(JAK inhibitor)   \n",
              "14                                                                                     Dasatinib(BCR-ABL inhibitor 2nd gen)   \n",
              "15                                                                                              Sorafenib(Pan-TK inhibitor)   \n",
              "16                                                                                              Sunitinib(Pan-TK inhibitor)   \n",
              "17                                                                        Imatinib(BCR-ABL inhibitor 1st gen&KIT inhibitor)   \n",
              "18                                                                        Imatinib(BCR-ABL inhibitor 1st gen&KIT inhibitor)   \n",
              "19                                                                                     Dasatinib(BCR-ABL inhibitor 2nd gen)   \n",
              "20                                                                                              Sorafenib(Pan-TK inhibitor)   \n",
              "21                                                                                              Sunitinib(Pan-TK inhibitor)   \n",
              "22                                                                                                   Spliceosome inhibitors   \n",
              "23                                                                                                          HDM2 inhibitors   \n",
              "24                                                                                                              Ruxolitinib   \n",
              "25                                                                                                  MEK Inhibitor RO4987655   \n",
              "26                                                                                                                Gefitinib   \n",
              "27                                                                                                            Pembrolizumab   \n",
              "28                                                                                                              Regorafenib   \n",
              "29                                                                                                              Regorafenib   \n",
              "30                                                                                                                  Aspirin   \n",
              "31                                                                                       Cetuximab,Oxaliplatin,Capecitabine   \n",
              "32                                                                                                    Adjuvant Chemotherapy   \n",
              "\n",
              "                         Diseases    Response Evidence Match  \\\n",
              "0                   Myelofibrosis  Responsive        A   YES   \n",
              "1       Colorectal adenocarcinoma  Responsive        A    NO   \n",
              "2       Colorectal adenocarcinoma  Responsive        A    NO   \n",
              "3             Non-small cell lung  Responsive        A    NO   \n",
              "4             Non-small cell lung  Responsive        A    NO   \n",
              "5             Non-small cell lung  Responsive        A    NO   \n",
              "6             Non-small cell lung  Responsive        A    NO   \n",
              "7             Non-small cell lung  Responsive        A    NO   \n",
              "8       Colorectal adenocarcinoma  Responsive        A    NO   \n",
              "9              Cutaneous melanoma   Resistant        C    NO   \n",
              "10         Acute myeloid leukemia  Responsive        D   YES   \n",
              "11          Breast adenocarcinoma  Responsive        D    NO   \n",
              "12   Acute lymphoblastic leukemia  Responsive        D   YES   \n",
              "13         Acute myeloid leukemia  Responsive        C   YES   \n",
              "14       Gastrointestinal stromal  Responsive        D    NO   \n",
              "15       Gastrointestinal stromal  Responsive        C    NO   \n",
              "16       Gastrointestinal stromal  Responsive        B    NO   \n",
              "17       Gastrointestinal stromal   Resistant        B    NO   \n",
              "18       Gastrointestinal stromal   Resistant        B    NO   \n",
              "19       Gastrointestinal stromal  Responsive        D    NO   \n",
              "20       Gastrointestinal stromal  Responsive        C    NO   \n",
              "21       Gastrointestinal stromal  Responsive        B    NO   \n",
              "22         Acute myeloid leukemia  Responsive        D   YES   \n",
              "23         Acute myeloid leukemia  Responsive        C   YES   \n",
              "24  Eosinophilic chronic leukemia  Responsive        C   YES   \n",
              "25             Cutaneous melanoma   Resistant        B    NO   \n",
              "26            Non-small cell lung   Resistant        B    NO   \n",
              "27             Cutaneous melanoma   Resistant        C    NO   \n",
              "28       Gastrointestinal stromal  Responsive        B    NO   \n",
              "29      Colorectal adenocarcinoma   Resistant        D    NO   \n",
              "30      Colorectal adenocarcinoma  Responsive        B    NO   \n",
              "31      Colorectal adenocarcinoma  Responsive        B    NO   \n",
              "32            Non-small cell lung  Responsive        B    NO   \n",
              "\n",
              "                                                                                                                                                                Source  \\\n",
              "0                                                                                                                                                                  FDA   \n",
              "1                                                                                     FDA  https://www.accessdata.fda.gov/drugsatfda_docs/label/2021/125147s210lbl.pdf   \n",
              "2                                                                                                                                                       PMID: 19339720   \n",
              "3                                                                                      FDA https://www.accessdata.fda.gov/drugsatfda_docs/label/2022/761034s047lbl.pdf   \n",
              "4                                                                                      FDA https://www.accessdata.fda.gov/drugsatfda_docs/label/2022/761034s047lbl.pdf   \n",
              "5                                                                                      FDA https://www.accessdata.fda.gov/drugsatfda_docs/label/2023/125514s136lbl.pdf   \n",
              "6   FDA https://www.fda.gov/drugs/resources-information-approved-drugs/fda-approves-tremelimumab-combination-durvalumab-and-platinum-based-chemotherapy-metastatic-non   \n",
              "7   FDA https://www.fda.gov/drugs/resources-information-approved-drugs/fda-approves-cemiplimab-rwlc-combination-platinum-based-chemotherapy-non-small-cell-lung-cancer   \n",
              "8                                                                                     FDA  https://www.accessdata.fda.gov/drugsatfda_docs/label/2021/125147s210lbl.pdf   \n",
              "9                                                                                                                                                        PMID:27433843   \n",
              "10                                                                                                                                                       PMID:22829971   \n",
              "11                                                                                                                                                       PMID:27075627   \n",
              "12                                                                                                                                         PMID:22875628;PMID:22899477   \n",
              "13                                                                                                                                                       PMID:22422826   \n",
              "14                                                                                                                                                       PMID:16397263   \n",
              "15                                                                                                                                             ASCO 2011 (abstr 10009)   \n",
              "16                                                                                                                                                       PMID:18955458   \n",
              "17                                                                                                                                                       PMID:18955458   \n",
              "18                                                                                                                                         PMID:14645423;PMID:18955458   \n",
              "19                                                                                                                                                       PMID:16397263   \n",
              "20                                                                                                                                             ASCO 2011 (abstr 10009)   \n",
              "21                                                                                                                                                       PMID:18955458   \n",
              "22                                                                                                                                                       PMID:27135740   \n",
              "23                                                                                                                                             AACR 2017 (abstr CT152)   \n",
              "24                                                                                                                     oncokb: oncokb=https://www.oncokb.org/gene/JAK2   \n",
              "25                                                                                                                        https://civicdb.org/links/evidence_items/995   \n",
              "26                                                                                                                       https://civicdb.org/links/evidence_items/2512   \n",
              "27                                                                                                                       https://civicdb.org/links/evidence_items/4825   \n",
              "28                                                                                                                       https://civicdb.org/links/evidence_items/4144   \n",
              "29                                                                                                                       https://civicdb.org/links/evidence_items/8840   \n",
              "30                                                                                                                       https://civicdb.org/links/evidence_items/2038   \n",
              "31                                                                                                                        https://civicdb.org/links/evidence_items/875   \n",
              "32                                                                                                                       https://civicdb.org/links/evidence_items/1149   \n",
              "\n",
              "                    BioM  Resist. Tumor type  \n",
              "0               complete      NaN         MY  \n",
              "1               complete      NaN     COREAD  \n",
              "2               complete      NaN     COREAD  \n",
              "3               complete      NaN      NSCLC  \n",
              "4               complete      NaN      NSCLC  \n",
              "5               complete      NaN      NSCLC  \n",
              "6               complete      NaN      NSCLC  \n",
              "7               complete      NaN      NSCLC  \n",
              "8               complete      NaN     COREAD  \n",
              "9               complete      NaN         CM  \n",
              "10              complete      NaN        AML  \n",
              "11             only gene      NaN       BRCA  \n",
              "12             only gene      NaN        ALL  \n",
              "13              complete      NaN        AML  \n",
              "14              complete      NaN       GIST  \n",
              "15              complete      NaN       GIST  \n",
              "16              complete      NaN       GIST  \n",
              "17              complete      NaN       GIST  \n",
              "18              complete      NaN       GIST  \n",
              "19              complete      NaN       GIST  \n",
              "20              complete      NaN       GIST  \n",
              "21              complete      NaN       GIST  \n",
              "22              complete      NaN        AML  \n",
              "23              complete      NaN        AML  \n",
              "24             only gene      NaN        ECL  \n",
              "25              complete      NaN         CM  \n",
              "26              complete      NaN      NSCLC  \n",
              "27  only alteration type      NaN         CM  \n",
              "28              complete      NaN       GIST  \n",
              "29              complete      NaN     COREAD  \n",
              "30              complete      NaN     COREAD  \n",
              "31              complete      NaN     COREAD  \n",
              "32              complete      NaN      NSCLC  "
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-64163a32-18d1-4c2a-99e9-c81b86ee6fa6\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>Sample ID</th>\n",
              "      <th>Alterations</th>\n",
              "      <th>Biomarker</th>\n",
              "      <th>Drugs</th>\n",
              "      <th>Diseases</th>\n",
              "      <th>Response</th>\n",
              "      <th>Evidence</th>\n",
              "      <th>Match</th>\n",
              "      <th>Source</th>\n",
              "      <th>BioM</th>\n",
              "      <th>Resist.</th>\n",
              "      <th>Tumor type</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>input01</td>\n",
              "      <td>JAK2 (V617F)</td>\n",
              "      <td>JAK2 (V617F)</td>\n",
              "      <td>Ruxolitinib(JAK inhibitor)</td>\n",
              "      <td>Myelofibrosis</td>\n",
              "      <td>Responsive</td>\n",
              "      <td>A</td>\n",
              "      <td>YES</td>\n",
              "      <td>FDA</td>\n",
              "      <td>complete</td>\n",
              "      <td>NaN</td>\n",
              "      <td>MY</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>input01</td>\n",
              "      <td>KRAS wildtype</td>\n",
              "      <td>KRAS wildtype</td>\n",
              "      <td>Panitumumab(EGFR mAb inhibitor)</td>\n",
              "      <td>Colorectal adenocarcinoma</td>\n",
              "      <td>Responsive</td>\n",
              "      <td>A</td>\n",
              "      <td>NO</td>\n",
              "      <td>FDA  https://www.accessdata.fda.gov/drugsatfda_docs/label/2021/125147s210lbl.pdf</td>\n",
              "      <td>complete</td>\n",
              "      <td>NaN</td>\n",
              "      <td>COREAD</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>input01</td>\n",
              "      <td>KRAS wildtype</td>\n",
              "      <td>KRAS wildtype</td>\n",
              "      <td>Cetuximab(EGFR mAb inhibitor)</td>\n",
              "      <td>Colorectal adenocarcinoma</td>\n",
              "      <td>Responsive</td>\n",
              "      <td>A</td>\n",
              "      <td>NO</td>\n",
              "      <td>PMID: 19339720</td>\n",
              "      <td>complete</td>\n",
              "      <td>NaN</td>\n",
              "      <td>COREAD</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>input01</td>\n",
              "      <td>EGFR wildtype, ALK wildtype</td>\n",
              "      <td>EGFR wildtype;ALK wildtype</td>\n",
              "      <td>Atezolizumab(PDL1 blocking antibody);Bevacizumab(VEGF mAb inhibitor);Paclitaxel(Chemotherapy);Carboplatin(Chemotherapy)</td>\n",
              "      <td>Non-small cell lung</td>\n",
              "      <td>Responsive</td>\n",
              "      <td>A</td>\n",
              "      <td>NO</td>\n",
              "      <td>FDA https://www.accessdata.fda.gov/drugsatfda_docs/label/2022/761034s047lbl.pdf</td>\n",
              "      <td>complete</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NSCLC</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>input01</td>\n",
              "      <td>EGFR wildtype, ALK wildtype</td>\n",
              "      <td>EGFR wildtype;ALK wildtype</td>\n",
              "      <td>Atezolizumab(PDL1 blocking antibody);Paclitaxel(Chemotherapy);Carboplatin(Chemotherapy)</td>\n",
              "      <td>Non-small cell lung</td>\n",
              "      <td>Responsive</td>\n",
              "      <td>A</td>\n",
              "      <td>NO</td>\n",
              "      <td>FDA https://www.accessdata.fda.gov/drugsatfda_docs/label/2022/761034s047lbl.pdf</td>\n",
              "      <td>complete</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NSCLC</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5</th>\n",
              "      <td>input01</td>\n",
              "      <td>EGFR wildtype, ALK wildtype</td>\n",
              "      <td>EGFR wildtype;ALK wildtype</td>\n",
              "      <td>Pembrolizumab(PD-1-blocking antibody);Pemetrexed(Chemotherapy);Platinum(Chemotherapy)</td>\n",
              "      <td>Non-small cell lung</td>\n",
              "      <td>Responsive</td>\n",
              "      <td>A</td>\n",
              "      <td>NO</td>\n",
              "      <td>FDA https://www.accessdata.fda.gov/drugsatfda_docs/label/2023/125514s136lbl.pdf</td>\n",
              "      <td>complete</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NSCLC</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>6</th>\n",
              "      <td>input01</td>\n",
              "      <td>EGFR wildtype, ALK wildtype</td>\n",
              "      <td>EGFR wildtype;ALK wildtype</td>\n",
              "      <td>Durvalumab(PD-1-blocking antibody);Tremelimumab-actl(CTLA-4 blocking antibody);Platinum(Chemotherapy)</td>\n",
              "      <td>Non-small cell lung</td>\n",
              "      <td>Responsive</td>\n",
              "      <td>A</td>\n",
              "      <td>NO</td>\n",
              "      <td>FDA https://www.fda.gov/drugs/resources-information-approved-drugs/fda-approves-tremelimumab-combination-durvalumab-and-platinum-based-chemotherapy-metastatic-non</td>\n",
              "      <td>complete</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NSCLC</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>7</th>\n",
              "      <td>input01</td>\n",
              "      <td>EGFR wildtype, ALK wildtype, ROS1 wildtype</td>\n",
              "      <td>EGFR wildtype;ALK wildtype;ROS1 wildtype</td>\n",
              "      <td>Cemiplimab-rwlc(PD-1-blocking antibody);Platinum(Chemotherapy)</td>\n",
              "      <td>Non-small cell lung</td>\n",
              "      <td>Responsive</td>\n",
              "      <td>A</td>\n",
              "      <td>NO</td>\n",
              "      <td>FDA https://www.fda.gov/drugs/resources-information-approved-drugs/fda-approves-cemiplimab-rwlc-combination-platinum-based-chemotherapy-non-small-cell-lung-cancer</td>\n",
              "      <td>complete</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NSCLC</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>8</th>\n",
              "      <td>input01</td>\n",
              "      <td>NRAS wildtype</td>\n",
              "      <td>NRAS wildtype</td>\n",
              "      <td>Panitumumab(EGFR mAb inhibitor)</td>\n",
              "      <td>Colorectal adenocarcinoma</td>\n",
              "      <td>Responsive</td>\n",
              "      <td>A</td>\n",
              "      <td>NO</td>\n",
              "      <td>FDA  https://www.accessdata.fda.gov/drugsatfda_docs/label/2021/125147s210lbl.pdf</td>\n",
              "      <td>complete</td>\n",
              "      <td>NaN</td>\n",
              "      <td>COREAD</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>9</th>\n",
              "      <td>input01</td>\n",
              "      <td>JAK2 (V617F)</td>\n",
              "      <td>JAK2 oncogenic mutation</td>\n",
              "      <td>PD1 inhibitors</td>\n",
              "      <td>Cutaneous melanoma</td>\n",
              "      <td>Resistant</td>\n",
              "      <td>C</td>\n",
              "      <td>NO</td>\n",
              "      <td>PMID:27433843</td>\n",
              "      <td>complete</td>\n",
              "      <td>NaN</td>\n",
              "      <td>CM</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>10</th>\n",
              "      <td>input01</td>\n",
              "      <td>JAK2 (V617F)</td>\n",
              "      <td>JAK2 (V617F)</td>\n",
              "      <td>JAK inhibitor(alone or in combination)s</td>\n",
              "      <td>Acute myeloid leukemia</td>\n",
              "      <td>Responsive</td>\n",
              "      <td>D</td>\n",
              "      <td>YES</td>\n",
              "      <td>PMID:22829971</td>\n",
              "      <td>complete</td>\n",
              "      <td>NaN</td>\n",
              "      <td>AML</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>11</th>\n",
              "      <td>input01</td>\n",
              "      <td>JAK2 (V617F)</td>\n",
              "      <td>JAK2 amplification</td>\n",
              "      <td>JAK inhibitors</td>\n",
              "      <td>Breast adenocarcinoma</td>\n",
              "      <td>Responsive</td>\n",
              "      <td>D</td>\n",
              "      <td>NO</td>\n",
              "      <td>PMID:27075627</td>\n",
              "      <td>only gene</td>\n",
              "      <td>NaN</td>\n",
              "      <td>BRCA</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>12</th>\n",
              "      <td>input01</td>\n",
              "      <td>JAK2 (V617F)</td>\n",
              "      <td>JAK2-BRAF fusion</td>\n",
              "      <td>Ruxolitinib(JAK inhibitor)</td>\n",
              "      <td>Acute lymphoblastic leukemia</td>\n",
              "      <td>Responsive</td>\n",
              "      <td>D</td>\n",
              "      <td>YES</td>\n",
              "      <td>PMID:22875628;PMID:22899477</td>\n",
              "      <td>only gene</td>\n",
              "      <td>NaN</td>\n",
              "      <td>ALL</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>13</th>\n",
              "      <td>input01</td>\n",
              "      <td>JAK2 (V617F)</td>\n",
              "      <td>JAK2 (V617F)</td>\n",
              "      <td>Ruxolitinib(JAK inhibitor)</td>\n",
              "      <td>Acute myeloid leukemia</td>\n",
              "      <td>Responsive</td>\n",
              "      <td>C</td>\n",
              "      <td>YES</td>\n",
              "      <td>PMID:22422826</td>\n",
              "      <td>complete</td>\n",
              "      <td>NaN</td>\n",
              "      <td>AML</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>14</th>\n",
              "      <td>input01</td>\n",
              "      <td>KIT wildtype</td>\n",
              "      <td>KIT wildtype</td>\n",
              "      <td>Dasatinib(BCR-ABL inhibitor 2nd gen)</td>\n",
              "      <td>Gastrointestinal stromal</td>\n",
              "      <td>Responsive</td>\n",
              "      <td>D</td>\n",
              "      <td>NO</td>\n",
              "      <td>PMID:16397263</td>\n",
              "      <td>complete</td>\n",
              "      <td>NaN</td>\n",
              "      <td>GIST</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>15</th>\n",
              "      <td>input01</td>\n",
              "      <td>KIT wildtype</td>\n",
              "      <td>KIT wildtype</td>\n",
              "      <td>Sorafenib(Pan-TK inhibitor)</td>\n",
              "      <td>Gastrointestinal stromal</td>\n",
              "      <td>Responsive</td>\n",
              "      <td>C</td>\n",
              "      <td>NO</td>\n",
              "      <td>ASCO 2011 (abstr 10009)</td>\n",
              "      <td>complete</td>\n",
              "      <td>NaN</td>\n",
              "      <td>GIST</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>16</th>\n",
              "      <td>input01</td>\n",
              "      <td>KIT wildtype</td>\n",
              "      <td>KIT wildtype</td>\n",
              "      <td>Sunitinib(Pan-TK inhibitor)</td>\n",
              "      <td>Gastrointestinal stromal</td>\n",
              "      <td>Responsive</td>\n",
              "      <td>B</td>\n",
              "      <td>NO</td>\n",
              "      <td>PMID:18955458</td>\n",
              "      <td>complete</td>\n",
              "      <td>NaN</td>\n",
              "      <td>GIST</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>17</th>\n",
              "      <td>input01</td>\n",
              "      <td>KIT wildtype</td>\n",
              "      <td>KIT wildtype</td>\n",
              "      <td>Imatinib(BCR-ABL inhibitor 1st gen&amp;KIT inhibitor)</td>\n",
              "      <td>Gastrointestinal stromal</td>\n",
              "      <td>Resistant</td>\n",
              "      <td>B</td>\n",
              "      <td>NO</td>\n",
              "      <td>PMID:18955458</td>\n",
              "      <td>complete</td>\n",
              "      <td>NaN</td>\n",
              "      <td>GIST</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>18</th>\n",
              "      <td>input01</td>\n",
              "      <td>PDGFRA wildtype</td>\n",
              "      <td>PDGFRA wildtype</td>\n",
              "      <td>Imatinib(BCR-ABL inhibitor 1st gen&amp;KIT inhibitor)</td>\n",
              "      <td>Gastrointestinal stromal</td>\n",
              "      <td>Resistant</td>\n",
              "      <td>B</td>\n",
              "      <td>NO</td>\n",
              "      <td>PMID:14645423;PMID:18955458</td>\n",
              "      <td>complete</td>\n",
              "      <td>NaN</td>\n",
              "      <td>GIST</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>19</th>\n",
              "      <td>input01</td>\n",
              "      <td>PDGFRA wildtype</td>\n",
              "      <td>PDGFRA wildtype</td>\n",
              "      <td>Dasatinib(BCR-ABL inhibitor 2nd gen)</td>\n",
              "      <td>Gastrointestinal stromal</td>\n",
              "      <td>Responsive</td>\n",
              "      <td>D</td>\n",
              "      <td>NO</td>\n",
              "      <td>PMID:16397263</td>\n",
              "      <td>complete</td>\n",
              "      <td>NaN</td>\n",
              "      <td>GIST</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>20</th>\n",
              "      <td>input01</td>\n",
              "      <td>PDGFRA wildtype</td>\n",
              "      <td>PDGFRA wildtype</td>\n",
              "      <td>Sorafenib(Pan-TK inhibitor)</td>\n",
              "      <td>Gastrointestinal stromal</td>\n",
              "      <td>Responsive</td>\n",
              "      <td>C</td>\n",
              "      <td>NO</td>\n",
              "      <td>ASCO 2011 (abstr 10009)</td>\n",
              "      <td>complete</td>\n",
              "      <td>NaN</td>\n",
              "      <td>GIST</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>21</th>\n",
              "      <td>input01</td>\n",
              "      <td>PDGFRA wildtype</td>\n",
              "      <td>PDGFRA wildtype</td>\n",
              "      <td>Sunitinib(Pan-TK inhibitor)</td>\n",
              "      <td>Gastrointestinal stromal</td>\n",
              "      <td>Responsive</td>\n",
              "      <td>B</td>\n",
              "      <td>NO</td>\n",
              "      <td>PMID:18955458</td>\n",
              "      <td>complete</td>\n",
              "      <td>NaN</td>\n",
              "      <td>GIST</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>22</th>\n",
              "      <td>input01</td>\n",
              "      <td>SRSF2 (P95H)</td>\n",
              "      <td>SRSF2 oncogenic mutation</td>\n",
              "      <td>Spliceosome inhibitors</td>\n",
              "      <td>Acute myeloid leukemia</td>\n",
              "      <td>Responsive</td>\n",
              "      <td>D</td>\n",
              "      <td>YES</td>\n",
              "      <td>PMID:27135740</td>\n",
              "      <td>complete</td>\n",
              "      <td>NaN</td>\n",
              "      <td>AML</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>23</th>\n",
              "      <td>input01</td>\n",
              "      <td>TP53 wildtype</td>\n",
              "      <td>TP53 wildtype</td>\n",
              "      <td>HDM2 inhibitors</td>\n",
              "      <td>Acute myeloid leukemia</td>\n",
              "      <td>Responsive</td>\n",
              "      <td>C</td>\n",
              "      <td>YES</td>\n",
              "      <td>AACR 2017 (abstr CT152)</td>\n",
              "      <td>complete</td>\n",
              "      <td>NaN</td>\n",
              "      <td>AML</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>24</th>\n",
              "      <td>input01</td>\n",
              "      <td>JAK2 (V617F)</td>\n",
              "      <td>JAK2-PCM1 fusion</td>\n",
              "      <td>Ruxolitinib</td>\n",
              "      <td>Eosinophilic chronic leukemia</td>\n",
              "      <td>Responsive</td>\n",
              "      <td>C</td>\n",
              "      <td>YES</td>\n",
              "      <td>oncokb: oncokb=https://www.oncokb.org/gene/JAK2</td>\n",
              "      <td>only gene</td>\n",
              "      <td>NaN</td>\n",
              "      <td>ECL</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>25</th>\n",
              "      <td>input01</td>\n",
              "      <td>BRAF wildtype</td>\n",
              "      <td>BRAF WILD TYPE</td>\n",
              "      <td>MEK Inhibitor RO4987655</td>\n",
              "      <td>Cutaneous melanoma</td>\n",
              "      <td>Resistant</td>\n",
              "      <td>B</td>\n",
              "      <td>NO</td>\n",
              "      <td>https://civicdb.org/links/evidence_items/995</td>\n",
              "      <td>complete</td>\n",
              "      <td>NaN</td>\n",
              "      <td>CM</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>26</th>\n",
              "      <td>input01</td>\n",
              "      <td>EGFR wildtype</td>\n",
              "      <td>EGFR Wildtype</td>\n",
              "      <td>Gefitinib</td>\n",
              "      <td>Non-small cell lung</td>\n",
              "      <td>Resistant</td>\n",
              "      <td>B</td>\n",
              "      <td>NO</td>\n",
              "      <td>https://civicdb.org/links/evidence_items/2512</td>\n",
              "      <td>complete</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NSCLC</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>27</th>\n",
              "      <td>input01</td>\n",
              "      <td>JAK2 (V617F)</td>\n",
              "      <td>JAK2 Splice Site (c.1641+2T&gt;G)</td>\n",
              "      <td>Pembrolizumab</td>\n",
              "      <td>Cutaneous melanoma</td>\n",
              "      <td>Resistant</td>\n",
              "      <td>C</td>\n",
              "      <td>NO</td>\n",
              "      <td>https://civicdb.org/links/evidence_items/4825</td>\n",
              "      <td>only alteration type</td>\n",
              "      <td>NaN</td>\n",
              "      <td>CM</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>28</th>\n",
              "      <td>input01</td>\n",
              "      <td>KIT wildtype</td>\n",
              "      <td>KIT WILDTYPE</td>\n",
              "      <td>Regorafenib</td>\n",
              "      <td>Gastrointestinal stromal</td>\n",
              "      <td>Responsive</td>\n",
              "      <td>B</td>\n",
              "      <td>NO</td>\n",
              "      <td>https://civicdb.org/links/evidence_items/4144</td>\n",
              "      <td>complete</td>\n",
              "      <td>NaN</td>\n",
              "      <td>GIST</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>29</th>\n",
              "      <td>input01</td>\n",
              "      <td>KRAS wildtype</td>\n",
              "      <td>KRAS Wildtype</td>\n",
              "      <td>Regorafenib</td>\n",
              "      <td>Colorectal adenocarcinoma</td>\n",
              "      <td>Resistant</td>\n",
              "      <td>D</td>\n",
              "      <td>NO</td>\n",
              "      <td>https://civicdb.org/links/evidence_items/8840</td>\n",
              "      <td>complete</td>\n",
              "      <td>NaN</td>\n",
              "      <td>COREAD</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>30</th>\n",
              "      <td>input01</td>\n",
              "      <td>PIK3CA wildtype</td>\n",
              "      <td>PIK3CA Wildtype</td>\n",
              "      <td>Aspirin</td>\n",
              "      <td>Colorectal adenocarcinoma</td>\n",
              "      <td>Responsive</td>\n",
              "      <td>B</td>\n",
              "      <td>NO</td>\n",
              "      <td>https://civicdb.org/links/evidence_items/2038</td>\n",
              "      <td>complete</td>\n",
              "      <td>NaN</td>\n",
              "      <td>COREAD</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>31</th>\n",
              "      <td>input01</td>\n",
              "      <td>TP53 wildtype</td>\n",
              "      <td>TP53 Wildtype</td>\n",
              "      <td>Cetuximab,Oxaliplatin,Capecitabine</td>\n",
              "      <td>Colorectal adenocarcinoma</td>\n",
              "      <td>Responsive</td>\n",
              "      <td>B</td>\n",
              "      <td>NO</td>\n",
              "      <td>https://civicdb.org/links/evidence_items/875</td>\n",
              "      <td>complete</td>\n",
              "      <td>NaN</td>\n",
              "      <td>COREAD</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>32</th>\n",
              "      <td>input01</td>\n",
              "      <td>TP53 wildtype</td>\n",
              "      <td>TP53 Wildtype</td>\n",
              "      <td>Adjuvant Chemotherapy</td>\n",
              "      <td>Non-small cell lung</td>\n",
              "      <td>Responsive</td>\n",
              "      <td>B</td>\n",
              "      <td>NO</td>\n",
              "      <td>https://civicdb.org/links/evidence_items/1149</td>\n",
              "      <td>complete</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NSCLC</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-64163a32-18d1-4c2a-99e9-c81b86ee6fa6')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-64163a32-18d1-4c2a-99e9-c81b86ee6fa6 button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-64163a32-18d1-4c2a-99e9-c81b86ee6fa6');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "<div id=\"df-325d0849-d5f4-4ed7-bd60-cc28060ae0f7\">\n",
              "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-325d0849-d5f4-4ed7-bd60-cc28060ae0f7')\"\n",
              "            title=\"Suggest charts\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "  </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "  <script>\n",
              "    async function quickchart(key) {\n",
              "      const quickchartButtonEl =\n",
              "        document.querySelector('#' + key + ' button');\n",
              "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "      try {\n",
              "        const charts = await google.colab.kernel.invokeFunction(\n",
              "            'suggestCharts', [key], {});\n",
              "      } catch (error) {\n",
              "        console.error('Error during call to suggestCharts:', error);\n",
              "      }\n",
              "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "    }\n",
              "    (() => {\n",
              "      let quickchartButtonEl =\n",
              "        document.querySelector('#df-325d0849-d5f4-4ed7-bd60-cc28060ae0f7 button');\n",
              "      quickchartButtonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "    })();\n",
              "  </script>\n",
              "</div>\n",
              "\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe",
              "summary": "{\n  \"name\": \"pd\",\n  \"rows\": 33,\n  \"fields\": [\n    {\n      \"column\": \"Sample ID\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 1,\n        \"samples\": [\n          \"input01\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Alterations\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 12,\n        \"samples\": [\n          \"EGFR wildtype\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Biomarker\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 20,\n        \"samples\": [\n          \"JAK2 (V617F)\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Drugs\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 25,\n        \"samples\": [\n          \"PD1 inhibitors\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Diseases\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 9,\n        \"samples\": [\n          \"Gastrointestinal stromal\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Response\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 2,\n        \"samples\": [\n          \"Resistant\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Evidence\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 4,\n        \"samples\": [\n          \"C\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Match\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 2,\n        \"samples\": [\n          \"NO\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Source\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 27,\n        \"samples\": [\n          \"PMID:22829971\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"BioM\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 3,\n        \"samples\": [\n          \"complete\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Resist.\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": null,\n        \"min\": null,\n        \"max\": null,\n        \"num_unique_values\": 0,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Tumor type\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 9,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}"
            }
          },
          "metadata": {},
          "execution_count": 36
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "# **WP058**"
      ],
      "metadata": {
        "id": "Qts_mvi5fhzL"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "#WP058\n",
        "#Código na célula será interpretado como comandos Bash.\n",
        "%%bash\n",
        "#Extrai as quatro primeiras colunas do arquivo, substitui o texto CHROM por CHR nas colunas extraídas, salva o resultado no arquivo e exisbe as 10 primeiras linhas do arquivo.\n",
        "cut -f1-4 /content/lmabrasil-hg38/vep_output/liftOver_WP058_hg19ToHg38.vep.filter.tsv | sed -e \"s/CHROM/CHR/g\"  > df_WP058-cgi.txt\n",
        "head df_WP058-cgi.txt"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Dl92TleafkKk",
        "outputId": "28ac482d-02b7-4233-82a7-5af0927346d1"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "CHR\tPOS\tREF\tALT\n",
            "chr19\t12943750\tAGCAGAGGCTTAAGGAGGAGGAAGAAGACAAGAAACGCAAAGAGGAGGAGGAG\tA\n",
            "chr20\t32434638\tA\tAG\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "import requests\n",
        "#Contém informações de autenticação.\n",
        "headers = {'Authorization': 'silfrizzarini@gmail.com 7f1b6ddf77bc5186c964'}\n",
        "#Dicionário que contém os dados a serem enviados no corpo da requisição.\n",
        "payload = {'cancer_type': 'HEMATO', 'title': 'Somatic MF', 'reference': 'hg38'}\n",
        "#Envia a requisição.\n",
        "r = requests.post('https://www.cancergenomeinterpreter.org/api/v1',\n",
        "                headers=headers,\n",
        "                files={\n",
        "                        'mutations': open('/content/df_WP058-cgi.txt', 'rb')\n",
        "                        },\n",
        "                data=payload)\n",
        "#A resposta da requisição é armazenada na variável r. Converte a resposta em formato JSON.\n",
        "r.json()"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 35
        },
        "id": "Lsfqsizcfv__",
        "outputId": "221f180b-f714-45ca-ff84-3d1f2f93e421"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "'304408f760307939cc95'"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "string"
            }
          },
          "metadata": {},
          "execution_count": 38
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#Envia uma requisição para o Cancer Genome Interpreter para obter os resultados de uma análise, converte a resposta para o formato JSON e retorna informações sobre as variantes genéticas analisadas.\n",
        "import requests\n",
        "job_id =\"304408f760307939cc95\"\n",
        "\n",
        "headers = {'Authorization': 'silfrizzarini@gmail.com 7f1b6ddf77bc5186c964'}\n",
        "r = requests.get('https://www.cancergenomeinterpreter.org/api/v1/%s' % job_id, headers=headers)\n",
        "r.json()"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "AlhNyNEvf2qN",
        "outputId": "0b0fe12b-973e-4ee9-f391-3220487a4d16"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "{'status': 'Done',\n",
              " 'metadata': {'id': '304408f760307939cc95',\n",
              "  'user': 'silfrizzarini@gmail.com',\n",
              "  'title': 'Somatic MF',\n",
              "  'cancertype': 'HEMATO',\n",
              "  'reference': 'hg38',\n",
              "  'dataset': 'input.tsv',\n",
              "  'date': '2024-12-21 13:56:41'}}"
            ]
          },
          "metadata": {},
          "execution_count": 40
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#Envia uma requisição para o Cancer Genome Interpreter para obter os logs da análise, retorna em formato JSON e mostra detalhes sobre o andamento ou erros da análise.\n",
        "import requests\n",
        "job_id =\"304408f760307939cc95\"\n",
        "\n",
        "headers = {'Authorization': 'silfrizzarini@gmail.com 7f1b6ddf77bc5186c964'}\n",
        "payload={'action':'logs'}\n",
        "r = requests.get('https://www.cancergenomeinterpreter.org/api/v1/%s' % job_id, headers=headers, params=payload)\n",
        "r.json()"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "JkWnwhUtf7Ra",
        "outputId": "4ab15120-fb6f-424f-f8b0-8af624e6da56"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "{'status': 'Done',\n",
              " 'logs': ['# cgi analyze input.tsv -c HEMATO -g hg38',\n",
              "  '2024-12-21 14:56:46,173 INFO     Parsing input01.tsv\\n',\n",
              "  '2024-12-21 14:56:49,738 INFO     Running VEP\\n',\n",
              "  '2024-12-21 14:56:51,215 INFO     Check cancer genes and consensus roles\\n',\n",
              "  '2024-12-21 14:56:51,302 INFO     Annotate BoostDM mutations\\n',\n",
              "  '2024-12-21 14:56:51,359 INFO     Annotate OncodriveMUT mutations\\n',\n",
              "  '2024-12-21 14:56:58,113 INFO     Annotate validated oncogenic mutations\\n',\n",
              "  '2024-12-21 14:56:58,254 INFO     Check oncogenic classification\\n',\n",
              "  '2024-12-21 14:56:58,313 INFO     Matching biomarkers\\n',\n",
              "  '2024-12-21 14:56:58,401 INFO     Prescription finished\\n',\n",
              "  '2024-12-21 14:56:58,412 INFO     Aggregate metrics\\n',\n",
              "  '2024-12-21 14:57:03,266 INFO     Compress output files\\n',\n",
              "  '2024-12-21 14:57:03,335 INFO     Analysis done\\n']}"
            ]
          },
          "metadata": {},
          "execution_count": 41
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#Baixa os resultados da análise em file.zip, fica disponível no diretório de trabalho.\n",
        "import requests\n",
        "job_id =\"304408f760307939cc95\"\n",
        "\n",
        "headers = {'Authorization': 'silfrizzarini@gmail.com 7f1b6ddf77bc5186c964'}\n",
        "payload={'action':'download'}\n",
        "r = requests.get('https://www.cancergenomeinterpreter.org/api/v1/%s' % job_id, headers=headers, params=payload)\n",
        "with open('file.zip', 'wb') as fd:\n",
        "    fd.write(r._content)"
      ],
      "metadata": {
        "id": "fKpDSBqGgB9K"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "#Permite confirmar o tipo do arquivo e garantir que ele tenha sido baixado corretamente.\n",
        "!file file.zip"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "ExdJv64AgFT_",
        "outputId": "997c1d90-4d6a-48f6-fc9b-2a83f013a21f"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "file.zip: Zip archive data, at least v2.0 to extract, compression method=deflate\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#Irá descompactar o conteúdo do arquivo file.zip e colocá-lo dentro de um diretório chamado WP019.\n",
        "!unzip -d WP058 file.zip"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "FWRWBn_2gI_E",
        "outputId": "37d63c39-9b85-46e7-dcc2-02046445cb6e"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Archive:  file.zip\n",
            "  inflating: WP058/alterations.tsv   \n",
            "  inflating: WP058/biomarkers.tsv    \n",
            "  inflating: WP058/input01.tsv       \n",
            "  inflating: WP058/summary.txt       \n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#Lê o arquivo alterations.tsv que está no diretório /content/WP058/, usando tabulação. Retorna um DataFrame contendo os dados do arquivo.\n",
        "pd.read_csv('/content/WP058/alterations.tsv',sep='\\t',index_col=False, engine= 'python')"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 219
        },
        "id": "skcbzXjDgJcN",
        "outputId": "27ea11a3-614e-4cbf-974e-71127c90afe8"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "    Input ID  CHROMOSOME  POSITION  \\\n",
              "0  input01_1          19  12943751   \n",
              "1  input01_2          20  32434638   \n",
              "\n",
              "                                                    REF ALT    CHR       POS  \\\n",
              "0  GCAGAGGCTTAAGGAGGAGGAAGAAGACAAGAAACGCAAAGAGGAGGAGGAG   -  chr19  12943750   \n",
              "1                                                     -   G  chr20  32434638   \n",
              "\n",
              "  ALT_TYPE STRAND CGI-Sample ID CGI-Gene          CGI-Protein Change  \\\n",
              "0    indel      +       input01     CALR  EQRLKEEEEDKKRKEEEE364-381X   \n",
              "1    indel      +       input01    ASXL1                   -642-643X   \n",
              "\n",
              "   CGI-Oncogenic Summary CGI-Oncogenic Prediction  \\\n",
              "0  oncogenic (predicted)    driver (oncodriveMUT)   \n",
              "1  oncogenic (predicted)    driver (oncodriveMUT)   \n",
              "\n",
              "   CGI-External oncogenic annotation  \\\n",
              "0                                NaN   \n",
              "1                                NaN   \n",
              "\n",
              "                                                                     CGI-Mutation  \\\n",
              "0  chr19:12943751-12943751 GCAGAGGCTTAAGGAGGAGGAAGAAGACAAGAAACGCAAAGAGGAGGAGGAG>-   \n",
              "1                                                     chr20:32434638-32434639 ->G   \n",
              "\n",
              "      CGI-Consequence   CGI-Transcript CGI-STRAND CGI-Type  \\\n",
              "0  frameshift_variant  ENST00000316448          +      DEL   \n",
              "1  frameshift_variant  ENST00000375687          +      INS   \n",
              "\n",
              "                                                             CGI-HGVS  \\\n",
              "0  ENST00000316448:c.1099_1150del;p.(Leu367ThrfsTer46);p.(L367Tfs*46)   \n",
              "1       ENST00000375687:c.1934dup;p.(Gly646TrpfsTer12);p.(G646Wfs*12)   \n",
              "\n",
              "                           CGI-HGVSc                             CGI-HGVSp  \n",
              "0  ENST00000316448.10:c.1099_1150del  ENSP00000320866.4:p.Leu367ThrfsTer46  \n",
              "1       ENST00000375687.10:c.1934dup  ENSP00000364839.4:p.Gly646TrpfsTer12  "
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-bd7455d7-32bb-49bc-9c41-0a6b17ae44cd\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>Input ID</th>\n",
              "      <th>CHROMOSOME</th>\n",
              "      <th>POSITION</th>\n",
              "      <th>REF</th>\n",
              "      <th>ALT</th>\n",
              "      <th>CHR</th>\n",
              "      <th>POS</th>\n",
              "      <th>ALT_TYPE</th>\n",
              "      <th>STRAND</th>\n",
              "      <th>CGI-Sample ID</th>\n",
              "      <th>CGI-Gene</th>\n",
              "      <th>CGI-Protein Change</th>\n",
              "      <th>CGI-Oncogenic Summary</th>\n",
              "      <th>CGI-Oncogenic Prediction</th>\n",
              "      <th>CGI-External oncogenic annotation</th>\n",
              "      <th>CGI-Mutation</th>\n",
              "      <th>CGI-Consequence</th>\n",
              "      <th>CGI-Transcript</th>\n",
              "      <th>CGI-STRAND</th>\n",
              "      <th>CGI-Type</th>\n",
              "      <th>CGI-HGVS</th>\n",
              "      <th>CGI-HGVSc</th>\n",
              "      <th>CGI-HGVSp</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>input01_1</td>\n",
              "      <td>19</td>\n",
              "      <td>12943751</td>\n",
              "      <td>GCAGAGGCTTAAGGAGGAGGAAGAAGACAAGAAACGCAAAGAGGAGGAGGAG</td>\n",
              "      <td>-</td>\n",
              "      <td>chr19</td>\n",
              "      <td>12943750</td>\n",
              "      <td>indel</td>\n",
              "      <td>+</td>\n",
              "      <td>input01</td>\n",
              "      <td>CALR</td>\n",
              "      <td>EQRLKEEEEDKKRKEEEE364-381X</td>\n",
              "      <td>oncogenic (predicted)</td>\n",
              "      <td>driver (oncodriveMUT)</td>\n",
              "      <td>NaN</td>\n",
              "      <td>chr19:12943751-12943751 GCAGAGGCTTAAGGAGGAGGAAGAAGACAAGAAACGCAAAGAGGAGGAGGAG&gt;-</td>\n",
              "      <td>frameshift_variant</td>\n",
              "      <td>ENST00000316448</td>\n",
              "      <td>+</td>\n",
              "      <td>DEL</td>\n",
              "      <td>ENST00000316448:c.1099_1150del;p.(Leu367ThrfsTer46);p.(L367Tfs*46)</td>\n",
              "      <td>ENST00000316448.10:c.1099_1150del</td>\n",
              "      <td>ENSP00000320866.4:p.Leu367ThrfsTer46</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>input01_2</td>\n",
              "      <td>20</td>\n",
              "      <td>32434638</td>\n",
              "      <td>-</td>\n",
              "      <td>G</td>\n",
              "      <td>chr20</td>\n",
              "      <td>32434638</td>\n",
              "      <td>indel</td>\n",
              "      <td>+</td>\n",
              "      <td>input01</td>\n",
              "      <td>ASXL1</td>\n",
              "      <td>-642-643X</td>\n",
              "      <td>oncogenic (predicted)</td>\n",
              "      <td>driver (oncodriveMUT)</td>\n",
              "      <td>NaN</td>\n",
              "      <td>chr20:32434638-32434639 -&gt;G</td>\n",
              "      <td>frameshift_variant</td>\n",
              "      <td>ENST00000375687</td>\n",
              "      <td>+</td>\n",
              "      <td>INS</td>\n",
              "      <td>ENST00000375687:c.1934dup;p.(Gly646TrpfsTer12);p.(G646Wfs*12)</td>\n",
              "      <td>ENST00000375687.10:c.1934dup</td>\n",
              "      <td>ENSP00000364839.4:p.Gly646TrpfsTer12</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-bd7455d7-32bb-49bc-9c41-0a6b17ae44cd')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-bd7455d7-32bb-49bc-9c41-0a6b17ae44cd button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-bd7455d7-32bb-49bc-9c41-0a6b17ae44cd');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "<div id=\"df-7676f5b9-ef00-4fe0-8221-d1546aaa0339\">\n",
              "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-7676f5b9-ef00-4fe0-8221-d1546aaa0339')\"\n",
              "            title=\"Suggest charts\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "  </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "  <script>\n",
              "    async function quickchart(key) {\n",
              "      const quickchartButtonEl =\n",
              "        document.querySelector('#' + key + ' button');\n",
              "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "      try {\n",
              "        const charts = await google.colab.kernel.invokeFunction(\n",
              "            'suggestCharts', [key], {});\n",
              "      } catch (error) {\n",
              "        console.error('Error during call to suggestCharts:', error);\n",
              "      }\n",
              "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "    }\n",
              "    (() => {\n",
              "      let quickchartButtonEl =\n",
              "        document.querySelector('#df-7676f5b9-ef00-4fe0-8221-d1546aaa0339 button');\n",
              "      quickchartButtonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "    })();\n",
              "  </script>\n",
              "</div>\n",
              "\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe"
            }
          },
          "metadata": {},
          "execution_count": 45
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#Vai carregar o conteúdo do arquivo biomarkers.tsv no diretório /content/WP058/ e retornará um DataFrame.\n",
        "pd.read_csv('/content/WP058/biomarkers.tsv',sep='\\t',index_col=False, engine= 'python')"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        },
        "id": "57UN_A1mgMgO",
        "outputId": "1186c3e0-3bb4-4956-e2ed-678d5d6ffa85"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "   Sample ID                                 Alterations  \\\n",
              "0    input01                               KRAS wildtype   \n",
              "1    input01                               KRAS wildtype   \n",
              "2    input01                 EGFR wildtype, ALK wildtype   \n",
              "3    input01                 EGFR wildtype, ALK wildtype   \n",
              "4    input01                 EGFR wildtype, ALK wildtype   \n",
              "5    input01                 EGFR wildtype, ALK wildtype   \n",
              "6    input01  EGFR wildtype, ALK wildtype, ROS1 wildtype   \n",
              "7    input01                               NRAS wildtype   \n",
              "8    input01                                KIT wildtype   \n",
              "9    input01                                KIT wildtype   \n",
              "10   input01                                KIT wildtype   \n",
              "11   input01                                KIT wildtype   \n",
              "12   input01                             PDGFRA wildtype   \n",
              "13   input01                             PDGFRA wildtype   \n",
              "14   input01                             PDGFRA wildtype   \n",
              "15   input01                             PDGFRA wildtype   \n",
              "16   input01                               TP53 wildtype   \n",
              "17   input01                               BRAF wildtype   \n",
              "18   input01                               EGFR wildtype   \n",
              "19   input01                                KIT wildtype   \n",
              "20   input01                               KRAS wildtype   \n",
              "21   input01                             PIK3CA wildtype   \n",
              "22   input01                               TP53 wildtype   \n",
              "23   input01                               TP53 wildtype   \n",
              "\n",
              "                                   Biomarker  \\\n",
              "0                              KRAS wildtype   \n",
              "1                              KRAS wildtype   \n",
              "2                 EGFR wildtype;ALK wildtype   \n",
              "3                 EGFR wildtype;ALK wildtype   \n",
              "4                 EGFR wildtype;ALK wildtype   \n",
              "5                 EGFR wildtype;ALK wildtype   \n",
              "6   EGFR wildtype;ALK wildtype;ROS1 wildtype   \n",
              "7                              NRAS wildtype   \n",
              "8                               KIT wildtype   \n",
              "9                               KIT wildtype   \n",
              "10                              KIT wildtype   \n",
              "11                              KIT wildtype   \n",
              "12                           PDGFRA wildtype   \n",
              "13                           PDGFRA wildtype   \n",
              "14                           PDGFRA wildtype   \n",
              "15                           PDGFRA wildtype   \n",
              "16                             TP53 wildtype   \n",
              "17                            BRAF WILD TYPE   \n",
              "18                             EGFR Wildtype   \n",
              "19                              KIT WILDTYPE   \n",
              "20                             KRAS Wildtype   \n",
              "21                           PIK3CA Wildtype   \n",
              "22                             TP53 Wildtype   \n",
              "23                             TP53 Wildtype   \n",
              "\n",
              "                                                                                                                      Drugs  \\\n",
              "0                                                                                           Panitumumab(EGFR mAb inhibitor)   \n",
              "1                                                                                             Cetuximab(EGFR mAb inhibitor)   \n",
              "2   Atezolizumab(PDL1 blocking antibody);Bevacizumab(VEGF mAb inhibitor);Paclitaxel(Chemotherapy);Carboplatin(Chemotherapy)   \n",
              "3                                   Atezolizumab(PDL1 blocking antibody);Paclitaxel(Chemotherapy);Carboplatin(Chemotherapy)   \n",
              "4                                     Pembrolizumab(PD-1-blocking antibody);Pemetrexed(Chemotherapy);Platinum(Chemotherapy)   \n",
              "5                     Durvalumab(PD-1-blocking antibody);Tremelimumab-actl(CTLA-4 blocking antibody);Platinum(Chemotherapy)   \n",
              "6                                                            Cemiplimab-rwlc(PD-1-blocking antibody);Platinum(Chemotherapy)   \n",
              "7                                                                                           Panitumumab(EGFR mAb inhibitor)   \n",
              "8                                                                                      Dasatinib(BCR-ABL inhibitor 2nd gen)   \n",
              "9                                                                                               Sorafenib(Pan-TK inhibitor)   \n",
              "10                                                                                              Sunitinib(Pan-TK inhibitor)   \n",
              "11                                                                        Imatinib(BCR-ABL inhibitor 1st gen&KIT inhibitor)   \n",
              "12                                                                        Imatinib(BCR-ABL inhibitor 1st gen&KIT inhibitor)   \n",
              "13                                                                                     Dasatinib(BCR-ABL inhibitor 2nd gen)   \n",
              "14                                                                                              Sorafenib(Pan-TK inhibitor)   \n",
              "15                                                                                              Sunitinib(Pan-TK inhibitor)   \n",
              "16                                                                                                          HDM2 inhibitors   \n",
              "17                                                                                                  MEK Inhibitor RO4987655   \n",
              "18                                                                                                                Gefitinib   \n",
              "19                                                                                                              Regorafenib   \n",
              "20                                                                                                              Regorafenib   \n",
              "21                                                                                                                  Aspirin   \n",
              "22                                                                                       Cetuximab,Oxaliplatin,Capecitabine   \n",
              "23                                                                                                    Adjuvant Chemotherapy   \n",
              "\n",
              "                     Diseases    Response Evidence Match  \\\n",
              "0   Colorectal adenocarcinoma  Responsive        A    NO   \n",
              "1   Colorectal adenocarcinoma  Responsive        A    NO   \n",
              "2         Non-small cell lung  Responsive        A    NO   \n",
              "3         Non-small cell lung  Responsive        A    NO   \n",
              "4         Non-small cell lung  Responsive        A    NO   \n",
              "5         Non-small cell lung  Responsive        A    NO   \n",
              "6         Non-small cell lung  Responsive        A    NO   \n",
              "7   Colorectal adenocarcinoma  Responsive        A    NO   \n",
              "8    Gastrointestinal stromal  Responsive        D    NO   \n",
              "9    Gastrointestinal stromal  Responsive        C    NO   \n",
              "10   Gastrointestinal stromal  Responsive        B    NO   \n",
              "11   Gastrointestinal stromal   Resistant        B    NO   \n",
              "12   Gastrointestinal stromal   Resistant        B    NO   \n",
              "13   Gastrointestinal stromal  Responsive        D    NO   \n",
              "14   Gastrointestinal stromal  Responsive        C    NO   \n",
              "15   Gastrointestinal stromal  Responsive        B    NO   \n",
              "16     Acute myeloid leukemia  Responsive        C   YES   \n",
              "17         Cutaneous melanoma   Resistant        B    NO   \n",
              "18        Non-small cell lung   Resistant        B    NO   \n",
              "19   Gastrointestinal stromal  Responsive        B    NO   \n",
              "20  Colorectal adenocarcinoma   Resistant        D    NO   \n",
              "21  Colorectal adenocarcinoma  Responsive        B    NO   \n",
              "22  Colorectal adenocarcinoma  Responsive        B    NO   \n",
              "23        Non-small cell lung  Responsive        B    NO   \n",
              "\n",
              "                                                                                                                                                                Source  \\\n",
              "0                                                                                     FDA  https://www.accessdata.fda.gov/drugsatfda_docs/label/2021/125147s210lbl.pdf   \n",
              "1                                                                                                                                                       PMID: 19339720   \n",
              "2                                                                                      FDA https://www.accessdata.fda.gov/drugsatfda_docs/label/2022/761034s047lbl.pdf   \n",
              "3                                                                                      FDA https://www.accessdata.fda.gov/drugsatfda_docs/label/2022/761034s047lbl.pdf   \n",
              "4                                                                                      FDA https://www.accessdata.fda.gov/drugsatfda_docs/label/2023/125514s136lbl.pdf   \n",
              "5   FDA https://www.fda.gov/drugs/resources-information-approved-drugs/fda-approves-tremelimumab-combination-durvalumab-and-platinum-based-chemotherapy-metastatic-non   \n",
              "6   FDA https://www.fda.gov/drugs/resources-information-approved-drugs/fda-approves-cemiplimab-rwlc-combination-platinum-based-chemotherapy-non-small-cell-lung-cancer   \n",
              "7                                                                                     FDA  https://www.accessdata.fda.gov/drugsatfda_docs/label/2021/125147s210lbl.pdf   \n",
              "8                                                                                                                                                        PMID:16397263   \n",
              "9                                                                                                                                              ASCO 2011 (abstr 10009)   \n",
              "10                                                                                                                                                       PMID:18955458   \n",
              "11                                                                                                                                                       PMID:18955458   \n",
              "12                                                                                                                                         PMID:14645423;PMID:18955458   \n",
              "13                                                                                                                                                       PMID:16397263   \n",
              "14                                                                                                                                             ASCO 2011 (abstr 10009)   \n",
              "15                                                                                                                                                       PMID:18955458   \n",
              "16                                                                                                                                             AACR 2017 (abstr CT152)   \n",
              "17                                                                                                                        https://civicdb.org/links/evidence_items/995   \n",
              "18                                                                                                                       https://civicdb.org/links/evidence_items/2512   \n",
              "19                                                                                                                       https://civicdb.org/links/evidence_items/4144   \n",
              "20                                                                                                                       https://civicdb.org/links/evidence_items/8840   \n",
              "21                                                                                                                       https://civicdb.org/links/evidence_items/2038   \n",
              "22                                                                                                                        https://civicdb.org/links/evidence_items/875   \n",
              "23                                                                                                                       https://civicdb.org/links/evidence_items/1149   \n",
              "\n",
              "        BioM  Resist. Tumor type  \n",
              "0   complete      NaN     COREAD  \n",
              "1   complete      NaN     COREAD  \n",
              "2   complete      NaN      NSCLC  \n",
              "3   complete      NaN      NSCLC  \n",
              "4   complete      NaN      NSCLC  \n",
              "5   complete      NaN      NSCLC  \n",
              "6   complete      NaN      NSCLC  \n",
              "7   complete      NaN     COREAD  \n",
              "8   complete      NaN       GIST  \n",
              "9   complete      NaN       GIST  \n",
              "10  complete      NaN       GIST  \n",
              "11  complete      NaN       GIST  \n",
              "12  complete      NaN       GIST  \n",
              "13  complete      NaN       GIST  \n",
              "14  complete      NaN       GIST  \n",
              "15  complete      NaN       GIST  \n",
              "16  complete      NaN        AML  \n",
              "17  complete      NaN         CM  \n",
              "18  complete      NaN      NSCLC  \n",
              "19  complete      NaN       GIST  \n",
              "20  complete      NaN     COREAD  \n",
              "21  complete      NaN     COREAD  \n",
              "22  complete      NaN     COREAD  \n",
              "23  complete      NaN      NSCLC  "
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-d334f09d-2d6f-4677-840d-90291d75866a\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>Sample ID</th>\n",
              "      <th>Alterations</th>\n",
              "      <th>Biomarker</th>\n",
              "      <th>Drugs</th>\n",
              "      <th>Diseases</th>\n",
              "      <th>Response</th>\n",
              "      <th>Evidence</th>\n",
              "      <th>Match</th>\n",
              "      <th>Source</th>\n",
              "      <th>BioM</th>\n",
              "      <th>Resist.</th>\n",
              "      <th>Tumor type</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>input01</td>\n",
              "      <td>KRAS wildtype</td>\n",
              "      <td>KRAS wildtype</td>\n",
              "      <td>Panitumumab(EGFR mAb inhibitor)</td>\n",
              "      <td>Colorectal adenocarcinoma</td>\n",
              "      <td>Responsive</td>\n",
              "      <td>A</td>\n",
              "      <td>NO</td>\n",
              "      <td>FDA  https://www.accessdata.fda.gov/drugsatfda_docs/label/2021/125147s210lbl.pdf</td>\n",
              "      <td>complete</td>\n",
              "      <td>NaN</td>\n",
              "      <td>COREAD</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>input01</td>\n",
              "      <td>KRAS wildtype</td>\n",
              "      <td>KRAS wildtype</td>\n",
              "      <td>Cetuximab(EGFR mAb inhibitor)</td>\n",
              "      <td>Colorectal adenocarcinoma</td>\n",
              "      <td>Responsive</td>\n",
              "      <td>A</td>\n",
              "      <td>NO</td>\n",
              "      <td>PMID: 19339720</td>\n",
              "      <td>complete</td>\n",
              "      <td>NaN</td>\n",
              "      <td>COREAD</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>input01</td>\n",
              "      <td>EGFR wildtype, ALK wildtype</td>\n",
              "      <td>EGFR wildtype;ALK wildtype</td>\n",
              "      <td>Atezolizumab(PDL1 blocking antibody);Bevacizumab(VEGF mAb inhibitor);Paclitaxel(Chemotherapy);Carboplatin(Chemotherapy)</td>\n",
              "      <td>Non-small cell lung</td>\n",
              "      <td>Responsive</td>\n",
              "      <td>A</td>\n",
              "      <td>NO</td>\n",
              "      <td>FDA https://www.accessdata.fda.gov/drugsatfda_docs/label/2022/761034s047lbl.pdf</td>\n",
              "      <td>complete</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NSCLC</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>input01</td>\n",
              "      <td>EGFR wildtype, ALK wildtype</td>\n",
              "      <td>EGFR wildtype;ALK wildtype</td>\n",
              "      <td>Atezolizumab(PDL1 blocking antibody);Paclitaxel(Chemotherapy);Carboplatin(Chemotherapy)</td>\n",
              "      <td>Non-small cell lung</td>\n",
              "      <td>Responsive</td>\n",
              "      <td>A</td>\n",
              "      <td>NO</td>\n",
              "      <td>FDA https://www.accessdata.fda.gov/drugsatfda_docs/label/2022/761034s047lbl.pdf</td>\n",
              "      <td>complete</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NSCLC</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>input01</td>\n",
              "      <td>EGFR wildtype, ALK wildtype</td>\n",
              "      <td>EGFR wildtype;ALK wildtype</td>\n",
              "      <td>Pembrolizumab(PD-1-blocking antibody);Pemetrexed(Chemotherapy);Platinum(Chemotherapy)</td>\n",
              "      <td>Non-small cell lung</td>\n",
              "      <td>Responsive</td>\n",
              "      <td>A</td>\n",
              "      <td>NO</td>\n",
              "      <td>FDA https://www.accessdata.fda.gov/drugsatfda_docs/label/2023/125514s136lbl.pdf</td>\n",
              "      <td>complete</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NSCLC</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5</th>\n",
              "      <td>input01</td>\n",
              "      <td>EGFR wildtype, ALK wildtype</td>\n",
              "      <td>EGFR wildtype;ALK wildtype</td>\n",
              "      <td>Durvalumab(PD-1-blocking antibody);Tremelimumab-actl(CTLA-4 blocking antibody);Platinum(Chemotherapy)</td>\n",
              "      <td>Non-small cell lung</td>\n",
              "      <td>Responsive</td>\n",
              "      <td>A</td>\n",
              "      <td>NO</td>\n",
              "      <td>FDA https://www.fda.gov/drugs/resources-information-approved-drugs/fda-approves-tremelimumab-combination-durvalumab-and-platinum-based-chemotherapy-metastatic-non</td>\n",
              "      <td>complete</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NSCLC</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>6</th>\n",
              "      <td>input01</td>\n",
              "      <td>EGFR wildtype, ALK wildtype, ROS1 wildtype</td>\n",
              "      <td>EGFR wildtype;ALK wildtype;ROS1 wildtype</td>\n",
              "      <td>Cemiplimab-rwlc(PD-1-blocking antibody);Platinum(Chemotherapy)</td>\n",
              "      <td>Non-small cell lung</td>\n",
              "      <td>Responsive</td>\n",
              "      <td>A</td>\n",
              "      <td>NO</td>\n",
              "      <td>FDA https://www.fda.gov/drugs/resources-information-approved-drugs/fda-approves-cemiplimab-rwlc-combination-platinum-based-chemotherapy-non-small-cell-lung-cancer</td>\n",
              "      <td>complete</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NSCLC</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>7</th>\n",
              "      <td>input01</td>\n",
              "      <td>NRAS wildtype</td>\n",
              "      <td>NRAS wildtype</td>\n",
              "      <td>Panitumumab(EGFR mAb inhibitor)</td>\n",
              "      <td>Colorectal adenocarcinoma</td>\n",
              "      <td>Responsive</td>\n",
              "      <td>A</td>\n",
              "      <td>NO</td>\n",
              "      <td>FDA  https://www.accessdata.fda.gov/drugsatfda_docs/label/2021/125147s210lbl.pdf</td>\n",
              "      <td>complete</td>\n",
              "      <td>NaN</td>\n",
              "      <td>COREAD</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>8</th>\n",
              "      <td>input01</td>\n",
              "      <td>KIT wildtype</td>\n",
              "      <td>KIT wildtype</td>\n",
              "      <td>Dasatinib(BCR-ABL inhibitor 2nd gen)</td>\n",
              "      <td>Gastrointestinal stromal</td>\n",
              "      <td>Responsive</td>\n",
              "      <td>D</td>\n",
              "      <td>NO</td>\n",
              "      <td>PMID:16397263</td>\n",
              "      <td>complete</td>\n",
              "      <td>NaN</td>\n",
              "      <td>GIST</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>9</th>\n",
              "      <td>input01</td>\n",
              "      <td>KIT wildtype</td>\n",
              "      <td>KIT wildtype</td>\n",
              "      <td>Sorafenib(Pan-TK inhibitor)</td>\n",
              "      <td>Gastrointestinal stromal</td>\n",
              "      <td>Responsive</td>\n",
              "      <td>C</td>\n",
              "      <td>NO</td>\n",
              "      <td>ASCO 2011 (abstr 10009)</td>\n",
              "      <td>complete</td>\n",
              "      <td>NaN</td>\n",
              "      <td>GIST</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>10</th>\n",
              "      <td>input01</td>\n",
              "      <td>KIT wildtype</td>\n",
              "      <td>KIT wildtype</td>\n",
              "      <td>Sunitinib(Pan-TK inhibitor)</td>\n",
              "      <td>Gastrointestinal stromal</td>\n",
              "      <td>Responsive</td>\n",
              "      <td>B</td>\n",
              "      <td>NO</td>\n",
              "      <td>PMID:18955458</td>\n",
              "      <td>complete</td>\n",
              "      <td>NaN</td>\n",
              "      <td>GIST</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>11</th>\n",
              "      <td>input01</td>\n",
              "      <td>KIT wildtype</td>\n",
              "      <td>KIT wildtype</td>\n",
              "      <td>Imatinib(BCR-ABL inhibitor 1st gen&amp;KIT inhibitor)</td>\n",
              "      <td>Gastrointestinal stromal</td>\n",
              "      <td>Resistant</td>\n",
              "      <td>B</td>\n",
              "      <td>NO</td>\n",
              "      <td>PMID:18955458</td>\n",
              "      <td>complete</td>\n",
              "      <td>NaN</td>\n",
              "      <td>GIST</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>12</th>\n",
              "      <td>input01</td>\n",
              "      <td>PDGFRA wildtype</td>\n",
              "      <td>PDGFRA wildtype</td>\n",
              "      <td>Imatinib(BCR-ABL inhibitor 1st gen&amp;KIT inhibitor)</td>\n",
              "      <td>Gastrointestinal stromal</td>\n",
              "      <td>Resistant</td>\n",
              "      <td>B</td>\n",
              "      <td>NO</td>\n",
              "      <td>PMID:14645423;PMID:18955458</td>\n",
              "      <td>complete</td>\n",
              "      <td>NaN</td>\n",
              "      <td>GIST</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>13</th>\n",
              "      <td>input01</td>\n",
              "      <td>PDGFRA wildtype</td>\n",
              "      <td>PDGFRA wildtype</td>\n",
              "      <td>Dasatinib(BCR-ABL inhibitor 2nd gen)</td>\n",
              "      <td>Gastrointestinal stromal</td>\n",
              "      <td>Responsive</td>\n",
              "      <td>D</td>\n",
              "      <td>NO</td>\n",
              "      <td>PMID:16397263</td>\n",
              "      <td>complete</td>\n",
              "      <td>NaN</td>\n",
              "      <td>GIST</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>14</th>\n",
              "      <td>input01</td>\n",
              "      <td>PDGFRA wildtype</td>\n",
              "      <td>PDGFRA wildtype</td>\n",
              "      <td>Sorafenib(Pan-TK inhibitor)</td>\n",
              "      <td>Gastrointestinal stromal</td>\n",
              "      <td>Responsive</td>\n",
              "      <td>C</td>\n",
              "      <td>NO</td>\n",
              "      <td>ASCO 2011 (abstr 10009)</td>\n",
              "      <td>complete</td>\n",
              "      <td>NaN</td>\n",
              "      <td>GIST</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>15</th>\n",
              "      <td>input01</td>\n",
              "      <td>PDGFRA wildtype</td>\n",
              "      <td>PDGFRA wildtype</td>\n",
              "      <td>Sunitinib(Pan-TK inhibitor)</td>\n",
              "      <td>Gastrointestinal stromal</td>\n",
              "      <td>Responsive</td>\n",
              "      <td>B</td>\n",
              "      <td>NO</td>\n",
              "      <td>PMID:18955458</td>\n",
              "      <td>complete</td>\n",
              "      <td>NaN</td>\n",
              "      <td>GIST</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>16</th>\n",
              "      <td>input01</td>\n",
              "      <td>TP53 wildtype</td>\n",
              "      <td>TP53 wildtype</td>\n",
              "      <td>HDM2 inhibitors</td>\n",
              "      <td>Acute myeloid leukemia</td>\n",
              "      <td>Responsive</td>\n",
              "      <td>C</td>\n",
              "      <td>YES</td>\n",
              "      <td>AACR 2017 (abstr CT152)</td>\n",
              "      <td>complete</td>\n",
              "      <td>NaN</td>\n",
              "      <td>AML</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>17</th>\n",
              "      <td>input01</td>\n",
              "      <td>BRAF wildtype</td>\n",
              "      <td>BRAF WILD TYPE</td>\n",
              "      <td>MEK Inhibitor RO4987655</td>\n",
              "      <td>Cutaneous melanoma</td>\n",
              "      <td>Resistant</td>\n",
              "      <td>B</td>\n",
              "      <td>NO</td>\n",
              "      <td>https://civicdb.org/links/evidence_items/995</td>\n",
              "      <td>complete</td>\n",
              "      <td>NaN</td>\n",
              "      <td>CM</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>18</th>\n",
              "      <td>input01</td>\n",
              "      <td>EGFR wildtype</td>\n",
              "      <td>EGFR Wildtype</td>\n",
              "      <td>Gefitinib</td>\n",
              "      <td>Non-small cell lung</td>\n",
              "      <td>Resistant</td>\n",
              "      <td>B</td>\n",
              "      <td>NO</td>\n",
              "      <td>https://civicdb.org/links/evidence_items/2512</td>\n",
              "      <td>complete</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NSCLC</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>19</th>\n",
              "      <td>input01</td>\n",
              "      <td>KIT wildtype</td>\n",
              "      <td>KIT WILDTYPE</td>\n",
              "      <td>Regorafenib</td>\n",
              "      <td>Gastrointestinal stromal</td>\n",
              "      <td>Responsive</td>\n",
              "      <td>B</td>\n",
              "      <td>NO</td>\n",
              "      <td>https://civicdb.org/links/evidence_items/4144</td>\n",
              "      <td>complete</td>\n",
              "      <td>NaN</td>\n",
              "      <td>GIST</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>20</th>\n",
              "      <td>input01</td>\n",
              "      <td>KRAS wildtype</td>\n",
              "      <td>KRAS Wildtype</td>\n",
              "      <td>Regorafenib</td>\n",
              "      <td>Colorectal adenocarcinoma</td>\n",
              "      <td>Resistant</td>\n",
              "      <td>D</td>\n",
              "      <td>NO</td>\n",
              "      <td>https://civicdb.org/links/evidence_items/8840</td>\n",
              "      <td>complete</td>\n",
              "      <td>NaN</td>\n",
              "      <td>COREAD</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>21</th>\n",
              "      <td>input01</td>\n",
              "      <td>PIK3CA wildtype</td>\n",
              "      <td>PIK3CA Wildtype</td>\n",
              "      <td>Aspirin</td>\n",
              "      <td>Colorectal adenocarcinoma</td>\n",
              "      <td>Responsive</td>\n",
              "      <td>B</td>\n",
              "      <td>NO</td>\n",
              "      <td>https://civicdb.org/links/evidence_items/2038</td>\n",
              "      <td>complete</td>\n",
              "      <td>NaN</td>\n",
              "      <td>COREAD</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>22</th>\n",
              "      <td>input01</td>\n",
              "      <td>TP53 wildtype</td>\n",
              "      <td>TP53 Wildtype</td>\n",
              "      <td>Cetuximab,Oxaliplatin,Capecitabine</td>\n",
              "      <td>Colorectal adenocarcinoma</td>\n",
              "      <td>Responsive</td>\n",
              "      <td>B</td>\n",
              "      <td>NO</td>\n",
              "      <td>https://civicdb.org/links/evidence_items/875</td>\n",
              "      <td>complete</td>\n",
              "      <td>NaN</td>\n",
              "      <td>COREAD</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>23</th>\n",
              "      <td>input01</td>\n",
              "      <td>TP53 wildtype</td>\n",
              "      <td>TP53 Wildtype</td>\n",
              "      <td>Adjuvant Chemotherapy</td>\n",
              "      <td>Non-small cell lung</td>\n",
              "      <td>Responsive</td>\n",
              "      <td>B</td>\n",
              "      <td>NO</td>\n",
              "      <td>https://civicdb.org/links/evidence_items/1149</td>\n",
              "      <td>complete</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NSCLC</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-d334f09d-2d6f-4677-840d-90291d75866a')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-d334f09d-2d6f-4677-840d-90291d75866a button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-d334f09d-2d6f-4677-840d-90291d75866a');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "<div id=\"df-1b343b4c-a9ff-41c1-9aa7-3fa87f8a997c\">\n",
              "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-1b343b4c-a9ff-41c1-9aa7-3fa87f8a997c')\"\n",
              "            title=\"Suggest charts\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "  </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "  <script>\n",
              "    async function quickchart(key) {\n",
              "      const quickchartButtonEl =\n",
              "        document.querySelector('#' + key + ' button');\n",
              "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "      try {\n",
              "        const charts = await google.colab.kernel.invokeFunction(\n",
              "            'suggestCharts', [key], {});\n",
              "      } catch (error) {\n",
              "        console.error('Error during call to suggestCharts:', error);\n",
              "      }\n",
              "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "    }\n",
              "    (() => {\n",
              "      let quickchartButtonEl =\n",
              "        document.querySelector('#df-1b343b4c-a9ff-41c1-9aa7-3fa87f8a997c button');\n",
              "      quickchartButtonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "    })();\n",
              "  </script>\n",
              "</div>\n",
              "\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe",
              "summary": "{\n  \"name\": \"pd\",\n  \"rows\": 24,\n  \"fields\": [\n    {\n      \"column\": \"Sample ID\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 1,\n        \"samples\": [\n          \"input01\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Alterations\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 10,\n        \"samples\": [\n          \"EGFR wildtype\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Biomarker\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 13,\n        \"samples\": [\n          \"PIK3CA Wildtype\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Drugs\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 18,\n        \"samples\": [\n          \"Panitumumab(EGFR mAb inhibitor)\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Diseases\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 5,\n        \"samples\": [\n          \"Non-small cell lung\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Response\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 2,\n        \"samples\": [\n          \"Resistant\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Evidence\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 4,\n        \"samples\": [\n          \"D\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Match\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 2,\n        \"samples\": [\n          \"YES\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Source\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 18,\n        \"samples\": [\n          \"FDA  https://www.accessdata.fda.gov/drugsatfda_docs/label/2021/125147s210lbl.pdf\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"BioM\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 1,\n        \"samples\": [\n          \"complete\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Resist.\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": null,\n        \"min\": null,\n        \"max\": null,\n        \"num_unique_values\": 0,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Tumor type\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 5,\n        \"samples\": [],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}"
            }
          },
          "metadata": {},
          "execution_count": 46
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "# **WP068**"
      ],
      "metadata": {
        "id": "j_qV0Wtbhmdr"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "#WP068\n",
        "#Código na célula será interpretado como comandos Bash.\n",
        "%%bash\n",
        "#Extrai as quatro primeiras colunas do arquivo, substitui o texto CHROM por CHR nas colunas extraídas, salva o resultado no arquivo e exisbe as 10 primeiras linhas do arquivo.\n",
        "cut -f1-4 /content/lmabrasil-hg38/vep_output/liftOver_WP068_hg19ToHg38.vep.filter.tsv | sed -e \"s/CHROM/CHR/g\"  > df_WP068-cgi.txt\n",
        "head df_WP068-cgi.txt"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "KJ0fP_TVhqSh",
        "outputId": "30cf07dc-2489-4f19-f6a1-d2573034763f"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "CHR\tPOS\tREF\tALT\n",
            "chr4\t54733155\tA\tT\n",
            "chr4\t105276128\tT\tC\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "import requests\n",
        "#Contém informações de autenticação.\n",
        "headers = {'Authorization': 'silfrizzarini@gmail.com 7f1b6ddf77bc5186c964'}\n",
        "#Dicionário que contém os dados a serem enviados no corpo da requisição.\n",
        "payload = {'cancer_type': 'HEMATO', 'title': 'Somatic MF', 'reference': 'hg38'}\n",
        "#Envia a requisição.\n",
        "r = requests.post('https://www.cancergenomeinterpreter.org/api/v1',\n",
        "                headers=headers,\n",
        "                files={\n",
        "                        'mutations': open('/content/df_WP068-cgi.txt', 'rb')\n",
        "                        },\n",
        "                data=payload)\n",
        "#A resposta da requisição é armazenada na variável r. Converte a resposta em formato JSON.\n",
        "r.json()"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 35
        },
        "id": "uZccM-zehubk",
        "outputId": "60ab4d95-175a-4bb1-80ab-a01c4c92d527"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "'c8cdb4eff2a1247837d4'"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "string"
            }
          },
          "metadata": {},
          "execution_count": 48
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#Envia uma requisição para o Cancer Genome Interpreter para obter os resultados de uma análise, converte a resposta para o formato JSON e retorna informações sobre as variantes genéticas analisadas.\n",
        "import requests\n",
        "job_id =\"c8cdb4eff2a1247837d4\"\n",
        "\n",
        "headers = {'Authorization': 'silfrizzarini@gmail.com 7f1b6ddf77bc5186c964'}\n",
        "r = requests.get('https://www.cancergenomeinterpreter.org/api/v1/%s' % job_id, headers=headers)\n",
        "r.json()"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "dx_N7YUlhyGa",
        "outputId": "0a718a51-d3eb-4280-eb26-b2593451742a"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "{'status': 'Done',\n",
              " 'metadata': {'id': 'c8cdb4eff2a1247837d4',\n",
              "  'user': 'silfrizzarini@gmail.com',\n",
              "  'title': 'Somatic MF',\n",
              "  'cancertype': 'HEMATO',\n",
              "  'reference': 'hg38',\n",
              "  'dataset': 'input.tsv',\n",
              "  'date': '2024-12-21 14:05:43'}}"
            ]
          },
          "metadata": {},
          "execution_count": 50
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#Envia uma requisição para o Cancer Genome Interpreter para obter os logs da análise, retorna em formato JSON e mostra detalhes sobre o andamento ou erros da análise.\n",
        "import requests\n",
        "job_id =\"c8cdb4eff2a1247837d4\"\n",
        "\n",
        "headers = {'Authorization': 'silfrizzarini@gmail.com 7f1b6ddf77bc5186c964'}\n",
        "payload={'action':'logs'}\n",
        "r = requests.get('https://www.cancergenomeinterpreter.org/api/v1/%s' % job_id, headers=headers, params=payload)\n",
        "r.json()"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Z9fwZWbOh19f",
        "outputId": "4ef35942-468c-4d3a-c75f-6307fa703365"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "{'status': 'Done',\n",
              " 'logs': ['# cgi analyze input.tsv -c HEMATO -g hg38',\n",
              "  '2024-12-21 15:05:47,542 INFO     Parsing input01.tsv\\n',\n",
              "  '2024-12-21 15:05:51,120 INFO     Running VEP\\n',\n",
              "  '2024-12-21 15:05:52,121 INFO     Check cancer genes and consensus roles\\n',\n",
              "  '2024-12-21 15:05:52,199 INFO     Annotate BoostDM mutations\\n',\n",
              "  '2024-12-21 15:05:53,937 INFO     Annotate OncodriveMUT mutations\\n',\n",
              "  '2024-12-21 15:05:56,195 INFO     Annotate validated oncogenic mutations\\n',\n",
              "  '2024-12-21 15:05:56,337 INFO     Check oncogenic classification\\n',\n",
              "  '2024-12-21 15:05:56,396 INFO     Matching biomarkers\\n',\n",
              "  '2024-12-21 15:05:56,484 INFO     Prescription finished\\n',\n",
              "  '2024-12-21 15:05:56,496 INFO     Aggregate metrics\\n',\n",
              "  '2024-12-21 15:06:02,486 INFO     Compress output files\\n',\n",
              "  '2024-12-21 15:06:02,543 INFO     Analysis done\\n']}"
            ]
          },
          "metadata": {},
          "execution_count": 51
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#Baixa os resultados da análise em file.zip, fica disponível no diretório de trabalho.\n",
        "import requests\n",
        "job_id =\"c8cdb4eff2a1247837d4\"\n",
        "\n",
        "headers = {'Authorization': 'silfrizzarini@gmail.com 7f1b6ddf77bc5186c964'}\n",
        "payload={'action':'download'}\n",
        "r = requests.get('https://www.cancergenomeinterpreter.org/api/v1/%s' % job_id, headers=headers, params=payload)\n",
        "with open('file.zip', 'wb') as fd:\n",
        "    fd.write(r._content)"
      ],
      "metadata": {
        "id": "e-Vuu60th69y"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "#Permite confirmar o tipo do arquivo e garantir que ele tenha sido baixado corretamente.\n",
        "!file file.zip"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "NQbKlAVzh-nX",
        "outputId": "387a4d22-4d12-4f98-e962-801f023b71b4"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "file.zip: Zip archive data, at least v2.0 to extract, compression method=deflate\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#Irá descompactar o conteúdo do arquivo file.zip e colocá-lo dentro de um diretório chamado WP068.\n",
        "!unzip -d WP068 file.zip"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "0bLflNkeiCYh",
        "outputId": "82c18ea8-4789-4324-f33e-d71e09333e8c"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Archive:  file.zip\n",
            "  inflating: WP068/alterations.tsv   \n",
            "  inflating: WP068/biomarkers.tsv    \n",
            "  inflating: WP068/input01.tsv       \n",
            "  inflating: WP068/summary.txt       \n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#Lê o arquivo alterations.tsv que está no diretório /content/WP068/, usando tabulação. Retorna um DataFrame contendo os dados do arquivo.\n",
        "pd.read_csv('/content/WP068/alterations.tsv',sep='\\t',index_col=False, engine= 'python')"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 271
        },
        "id": "RqnUPsgCiFko",
        "outputId": "96e36134-a3b8-47d5-b1be-057738df604f"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "    Input ID  CHROMOSOME   POSITION REF ALT   CHR        POS ALT_TYPE STRAND  \\\n",
              "0  input01_1           4   54733155   A   T  chr4   54733155      snp      +   \n",
              "1  input01_2           4  105276128   T   C  chr4  105276128      snp      +   \n",
              "\n",
              "  CGI-Sample ID CGI-Gene CGI-Protein Change  \\\n",
              "0       input01      KIT              D816V   \n",
              "1       input01     TET2             I1894T   \n",
              "\n",
              "                 CGI-Oncogenic Summary  \\\n",
              "0  oncogenic (predicted and annotated)   \n",
              "1  oncogenic (predicted and annotated)   \n",
              "\n",
              "                      CGI-Oncogenic Prediction  \\\n",
              "0  driver (boostDM: non-tissue-specific model)   \n",
              "1                        driver (oncodriveMUT)   \n",
              "\n",
              "  CGI-External oncogenic annotation        CGI-Mutation   CGI-Consequence  \\\n",
              "0          cgi,oncokb,clinvar:13852   chr4:54733155 A>T  missense_variant   \n",
              "1                   clinvar:2416868  chr4:105276128 T>C  missense_variant   \n",
              "\n",
              "    CGI-Transcript CGI-STRAND CGI-Type  \\\n",
              "0  ENST00000288135          +      SNV   \n",
              "1  ENST00000513237          +      SNV   \n",
              "\n",
              "                                              CGI-HGVS  \\\n",
              "0    ENST00000288135:c.2447A>T;p.(Asp816Val);p.(D816V)   \n",
              "1  ENST00000513237:c.5681T>C;p.(Ile1894Thr);p.(I1894T)   \n",
              "\n",
              "                     CGI-HGVSc                       CGI-HGVSp  \n",
              "0  ENST00000288135.6:c.2447A>T   ENSP00000288135.6:p.Asp816Val  \n",
              "1  ENST00000513237.5:c.5681T>C  ENSP00000425443.1:p.Ile1894Thr  "
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-e667b719-89a1-45fe-a6ff-49c5e20b8003\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>Input ID</th>\n",
              "      <th>CHROMOSOME</th>\n",
              "      <th>POSITION</th>\n",
              "      <th>REF</th>\n",
              "      <th>ALT</th>\n",
              "      <th>CHR</th>\n",
              "      <th>POS</th>\n",
              "      <th>ALT_TYPE</th>\n",
              "      <th>STRAND</th>\n",
              "      <th>CGI-Sample ID</th>\n",
              "      <th>CGI-Gene</th>\n",
              "      <th>CGI-Protein Change</th>\n",
              "      <th>CGI-Oncogenic Summary</th>\n",
              "      <th>CGI-Oncogenic Prediction</th>\n",
              "      <th>CGI-External oncogenic annotation</th>\n",
              "      <th>CGI-Mutation</th>\n",
              "      <th>CGI-Consequence</th>\n",
              "      <th>CGI-Transcript</th>\n",
              "      <th>CGI-STRAND</th>\n",
              "      <th>CGI-Type</th>\n",
              "      <th>CGI-HGVS</th>\n",
              "      <th>CGI-HGVSc</th>\n",
              "      <th>CGI-HGVSp</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>input01_1</td>\n",
              "      <td>4</td>\n",
              "      <td>54733155</td>\n",
              "      <td>A</td>\n",
              "      <td>T</td>\n",
              "      <td>chr4</td>\n",
              "      <td>54733155</td>\n",
              "      <td>snp</td>\n",
              "      <td>+</td>\n",
              "      <td>input01</td>\n",
              "      <td>KIT</td>\n",
              "      <td>D816V</td>\n",
              "      <td>oncogenic (predicted and annotated)</td>\n",
              "      <td>driver (boostDM: non-tissue-specific model)</td>\n",
              "      <td>cgi,oncokb,clinvar:13852</td>\n",
              "      <td>chr4:54733155 A&gt;T</td>\n",
              "      <td>missense_variant</td>\n",
              "      <td>ENST00000288135</td>\n",
              "      <td>+</td>\n",
              "      <td>SNV</td>\n",
              "      <td>ENST00000288135:c.2447A&gt;T;p.(Asp816Val);p.(D816V)</td>\n",
              "      <td>ENST00000288135.6:c.2447A&gt;T</td>\n",
              "      <td>ENSP00000288135.6:p.Asp816Val</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>input01_2</td>\n",
              "      <td>4</td>\n",
              "      <td>105276128</td>\n",
              "      <td>T</td>\n",
              "      <td>C</td>\n",
              "      <td>chr4</td>\n",
              "      <td>105276128</td>\n",
              "      <td>snp</td>\n",
              "      <td>+</td>\n",
              "      <td>input01</td>\n",
              "      <td>TET2</td>\n",
              "      <td>I1894T</td>\n",
              "      <td>oncogenic (predicted and annotated)</td>\n",
              "      <td>driver (oncodriveMUT)</td>\n",
              "      <td>clinvar:2416868</td>\n",
              "      <td>chr4:105276128 T&gt;C</td>\n",
              "      <td>missense_variant</td>\n",
              "      <td>ENST00000513237</td>\n",
              "      <td>+</td>\n",
              "      <td>SNV</td>\n",
              "      <td>ENST00000513237:c.5681T&gt;C;p.(Ile1894Thr);p.(I1894T)</td>\n",
              "      <td>ENST00000513237.5:c.5681T&gt;C</td>\n",
              "      <td>ENSP00000425443.1:p.Ile1894Thr</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-e667b719-89a1-45fe-a6ff-49c5e20b8003')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-e667b719-89a1-45fe-a6ff-49c5e20b8003 button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-e667b719-89a1-45fe-a6ff-49c5e20b8003');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "<div id=\"df-4973be0c-a02f-4b2c-aee9-68fc2c6ce130\">\n",
              "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-4973be0c-a02f-4b2c-aee9-68fc2c6ce130')\"\n",
              "            title=\"Suggest charts\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "  </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "  <script>\n",
              "    async function quickchart(key) {\n",
              "      const quickchartButtonEl =\n",
              "        document.querySelector('#' + key + ' button');\n",
              "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "      try {\n",
              "        const charts = await google.colab.kernel.invokeFunction(\n",
              "            'suggestCharts', [key], {});\n",
              "      } catch (error) {\n",
              "        console.error('Error during call to suggestCharts:', error);\n",
              "      }\n",
              "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "    }\n",
              "    (() => {\n",
              "      let quickchartButtonEl =\n",
              "        document.querySelector('#df-4973be0c-a02f-4b2c-aee9-68fc2c6ce130 button');\n",
              "      quickchartButtonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "    })();\n",
              "  </script>\n",
              "</div>\n",
              "\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe"
            }
          },
          "metadata": {},
          "execution_count": 56
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#Vai carregar o conteúdo do arquivo biomarkers.tsv no diretório /content/WP068/ e retornará um DataFrame.\n",
        "pd.read_csv('/content/WP068/biomarkers.tsv',sep='\\t',index_col=False, engine= 'python')"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        },
        "id": "lkI0ZHbIiJSo",
        "outputId": "255cbbd5-464f-4af8-c09e-caf2fd86a189"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "    Sample ID      Alterations                              Biomarker  \\\n",
              "0     input01      KIT (D816V)  KIT mutation in exon 9,11,13,14 or 17   \n",
              "1     input01      KIT (D816V)  KIT mutation in exon 9,11,13,14 or 17   \n",
              "2     input01      KIT (D816V)  KIT mutation in exon 9,11,13,14 or 17   \n",
              "3     input01      KIT (D816V)  KIT mutation in exon 9,11,13,14 or 17   \n",
              "4     input01      KIT (D816V)                             KIT (D816)   \n",
              "..        ...              ...                                    ...   \n",
              "110   input01      KIT (D816V)                              KIT Y823D   \n",
              "111   input01    KRAS wildtype                          KRAS Wildtype   \n",
              "112   input01  PIK3CA wildtype                        PIK3CA Wildtype   \n",
              "113   input01    TP53 wildtype                          TP53 Wildtype   \n",
              "114   input01    TP53 wildtype                          TP53 Wildtype   \n",
              "\n",
              "                                                 Drugs  \\\n",
              "0    Imatinib(BCR-ABL inhibitor 1st gen&KIT inhibitor)   \n",
              "1    Imatinib(BCR-ABL inhibitor 1st gen&KIT inhibitor)   \n",
              "2                    Regorafenib(Pan-kinase inhibitor)   \n",
              "3                          Sunitinib(Pan-TK inhibitor)   \n",
              "4    Imatinib(BCR-ABL inhibitor 1st gen&KIT inhibitor)   \n",
              "..                                                 ...   \n",
              "110                                           Imatinib   \n",
              "111                                        Regorafenib   \n",
              "112                                            Aspirin   \n",
              "113                 Cetuximab,Oxaliplatin,Capecitabine   \n",
              "114                              Adjuvant Chemotherapy   \n",
              "\n",
              "                                                                                                                                                                                                                                         Diseases  \\\n",
              "0                                                                                                                                                                                                                        Gastrointestinal stromal   \n",
              "1                                                                                                                                                                                                                              Cutaneous melanoma   \n",
              "2                                                                                                                                                                                                                        Gastrointestinal stromal   \n",
              "3                                                                                                                                                                                                                        Gastrointestinal stromal   \n",
              "4    Gastrointestinal stromal, Myelodisplasic syndrome, Myelodisplasic proliferative syndrome, Hyper eosinophilic advanced snydrome, Eosinophilic chronic leukemia, Chronic myeloid leukemia, Acute lymphoblastic leukemia, Systemic mastocytosis   \n",
              "..                                                                                                                                                                                                                                            ...   \n",
              "110                                                                                                                                                                                                                      Gastrointestinal stromal   \n",
              "111                                                                                                                                                                                                                     Colorectal adenocarcinoma   \n",
              "112                                                                                                                                                                                                                     Colorectal adenocarcinoma   \n",
              "113                                                                                                                                                                                                                     Colorectal adenocarcinoma   \n",
              "114                                                                                                                                                                                                                           Non-small cell lung   \n",
              "\n",
              "       Response Evidence Match                                         Source  \\\n",
              "0    Responsive        A    NO                                            FDA   \n",
              "1    Responsive        A    NO                                           NCCN   \n",
              "2    Responsive        A    NO                                            FDA   \n",
              "3    Responsive        A    NO                                            FDA   \n",
              "4     Resistant        A   YES                                            FDA   \n",
              "..          ...      ...   ...                                            ...   \n",
              "110   Resistant        C    NO  https://civicdb.org/links/evidence_items/2919   \n",
              "111   Resistant        D    NO  https://civicdb.org/links/evidence_items/8840   \n",
              "112  Responsive        B    NO  https://civicdb.org/links/evidence_items/2038   \n",
              "113  Responsive        B    NO   https://civicdb.org/links/evidence_items/875   \n",
              "114  Responsive        B    NO  https://civicdb.org/links/evidence_items/1149   \n",
              "\n",
              "                     BioM  Resist.                               Tumor type  \n",
              "0                complete     77.0                                     GIST  \n",
              "1    only alteration type     77.0                                       CM  \n",
              "2                complete      NaN                                     GIST  \n",
              "3                complete   1096.0                                     GIST  \n",
              "4                complete      NaN  GIST, MDS, MDPS, HES, ECL, CML, ALL, SM  \n",
              "..                    ...      ...                                      ...  \n",
              "110  only alteration type      NaN                                     GIST  \n",
              "111              complete      NaN                                   COREAD  \n",
              "112              complete      NaN                                   COREAD  \n",
              "113              complete      NaN                                   COREAD  \n",
              "114              complete      NaN                                    NSCLC  \n",
              "\n",
              "[115 rows x 12 columns]"
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-64608987-3435-485a-a8d8-2cd46f0c1098\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>Sample ID</th>\n",
              "      <th>Alterations</th>\n",
              "      <th>Biomarker</th>\n",
              "      <th>Drugs</th>\n",
              "      <th>Diseases</th>\n",
              "      <th>Response</th>\n",
              "      <th>Evidence</th>\n",
              "      <th>Match</th>\n",
              "      <th>Source</th>\n",
              "      <th>BioM</th>\n",
              "      <th>Resist.</th>\n",
              "      <th>Tumor type</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>input01</td>\n",
              "      <td>KIT (D816V)</td>\n",
              "      <td>KIT mutation in exon 9,11,13,14 or 17</td>\n",
              "      <td>Imatinib(BCR-ABL inhibitor 1st gen&amp;KIT inhibitor)</td>\n",
              "      <td>Gastrointestinal stromal</td>\n",
              "      <td>Responsive</td>\n",
              "      <td>A</td>\n",
              "      <td>NO</td>\n",
              "      <td>FDA</td>\n",
              "      <td>complete</td>\n",
              "      <td>77.0</td>\n",
              "      <td>GIST</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>input01</td>\n",
              "      <td>KIT (D816V)</td>\n",
              "      <td>KIT mutation in exon 9,11,13,14 or 17</td>\n",
              "      <td>Imatinib(BCR-ABL inhibitor 1st gen&amp;KIT inhibitor)</td>\n",
              "      <td>Cutaneous melanoma</td>\n",
              "      <td>Responsive</td>\n",
              "      <td>A</td>\n",
              "      <td>NO</td>\n",
              "      <td>NCCN</td>\n",
              "      <td>only alteration type</td>\n",
              "      <td>77.0</td>\n",
              "      <td>CM</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>input01</td>\n",
              "      <td>KIT (D816V)</td>\n",
              "      <td>KIT mutation in exon 9,11,13,14 or 17</td>\n",
              "      <td>Regorafenib(Pan-kinase inhibitor)</td>\n",
              "      <td>Gastrointestinal stromal</td>\n",
              "      <td>Responsive</td>\n",
              "      <td>A</td>\n",
              "      <td>NO</td>\n",
              "      <td>FDA</td>\n",
              "      <td>complete</td>\n",
              "      <td>NaN</td>\n",
              "      <td>GIST</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>input01</td>\n",
              "      <td>KIT (D816V)</td>\n",
              "      <td>KIT mutation in exon 9,11,13,14 or 17</td>\n",
              "      <td>Sunitinib(Pan-TK inhibitor)</td>\n",
              "      <td>Gastrointestinal stromal</td>\n",
              "      <td>Responsive</td>\n",
              "      <td>A</td>\n",
              "      <td>NO</td>\n",
              "      <td>FDA</td>\n",
              "      <td>complete</td>\n",
              "      <td>1096.0</td>\n",
              "      <td>GIST</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>input01</td>\n",
              "      <td>KIT (D816V)</td>\n",
              "      <td>KIT (D816)</td>\n",
              "      <td>Imatinib(BCR-ABL inhibitor 1st gen&amp;KIT inhibitor)</td>\n",
              "      <td>Gastrointestinal stromal, Myelodisplasic syndrome, Myelodisplasic proliferative syndrome, Hyper eosinophilic advanced snydrome, Eosinophilic chronic leukemia, Chronic myeloid leukemia, Acute lymphoblastic leukemia, Systemic mastocytosis</td>\n",
              "      <td>Resistant</td>\n",
              "      <td>A</td>\n",
              "      <td>YES</td>\n",
              "      <td>FDA</td>\n",
              "      <td>complete</td>\n",
              "      <td>NaN</td>\n",
              "      <td>GIST, MDS, MDPS, HES, ECL, CML, ALL, SM</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>110</th>\n",
              "      <td>input01</td>\n",
              "      <td>KIT (D816V)</td>\n",
              "      <td>KIT Y823D</td>\n",
              "      <td>Imatinib</td>\n",
              "      <td>Gastrointestinal stromal</td>\n",
              "      <td>Resistant</td>\n",
              "      <td>C</td>\n",
              "      <td>NO</td>\n",
              "      <td>https://civicdb.org/links/evidence_items/2919</td>\n",
              "      <td>only alteration type</td>\n",
              "      <td>NaN</td>\n",
              "      <td>GIST</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>111</th>\n",
              "      <td>input01</td>\n",
              "      <td>KRAS wildtype</td>\n",
              "      <td>KRAS Wildtype</td>\n",
              "      <td>Regorafenib</td>\n",
              "      <td>Colorectal adenocarcinoma</td>\n",
              "      <td>Resistant</td>\n",
              "      <td>D</td>\n",
              "      <td>NO</td>\n",
              "      <td>https://civicdb.org/links/evidence_items/8840</td>\n",
              "      <td>complete</td>\n",
              "      <td>NaN</td>\n",
              "      <td>COREAD</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>112</th>\n",
              "      <td>input01</td>\n",
              "      <td>PIK3CA wildtype</td>\n",
              "      <td>PIK3CA Wildtype</td>\n",
              "      <td>Aspirin</td>\n",
              "      <td>Colorectal adenocarcinoma</td>\n",
              "      <td>Responsive</td>\n",
              "      <td>B</td>\n",
              "      <td>NO</td>\n",
              "      <td>https://civicdb.org/links/evidence_items/2038</td>\n",
              "      <td>complete</td>\n",
              "      <td>NaN</td>\n",
              "      <td>COREAD</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>113</th>\n",
              "      <td>input01</td>\n",
              "      <td>TP53 wildtype</td>\n",
              "      <td>TP53 Wildtype</td>\n",
              "      <td>Cetuximab,Oxaliplatin,Capecitabine</td>\n",
              "      <td>Colorectal adenocarcinoma</td>\n",
              "      <td>Responsive</td>\n",
              "      <td>B</td>\n",
              "      <td>NO</td>\n",
              "      <td>https://civicdb.org/links/evidence_items/875</td>\n",
              "      <td>complete</td>\n",
              "      <td>NaN</td>\n",
              "      <td>COREAD</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>114</th>\n",
              "      <td>input01</td>\n",
              "      <td>TP53 wildtype</td>\n",
              "      <td>TP53 Wildtype</td>\n",
              "      <td>Adjuvant Chemotherapy</td>\n",
              "      <td>Non-small cell lung</td>\n",
              "      <td>Responsive</td>\n",
              "      <td>B</td>\n",
              "      <td>NO</td>\n",
              "      <td>https://civicdb.org/links/evidence_items/1149</td>\n",
              "      <td>complete</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NSCLC</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>115 rows × 12 columns</p>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-64608987-3435-485a-a8d8-2cd46f0c1098')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-64608987-3435-485a-a8d8-2cd46f0c1098 button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-64608987-3435-485a-a8d8-2cd46f0c1098');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "<div id=\"df-75da9184-136a-4314-8d91-442016ce7210\">\n",
              "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-75da9184-136a-4314-8d91-442016ce7210')\"\n",
              "            title=\"Suggest charts\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "  </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "  <script>\n",
              "    async function quickchart(key) {\n",
              "      const quickchartButtonEl =\n",
              "        document.querySelector('#' + key + ' button');\n",
              "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "      try {\n",
              "        const charts = await google.colab.kernel.invokeFunction(\n",
              "            'suggestCharts', [key], {});\n",
              "      } catch (error) {\n",
              "        console.error('Error during call to suggestCharts:', error);\n",
              "      }\n",
              "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "    }\n",
              "    (() => {\n",
              "      let quickchartButtonEl =\n",
              "        document.querySelector('#df-75da9184-136a-4314-8d91-442016ce7210 button');\n",
              "      quickchartButtonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "    })();\n",
              "  </script>\n",
              "</div>\n",
              "\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe",
              "summary": "{\n  \"name\": \"pd\",\n  \"rows\": 115,\n  \"fields\": [\n    {\n      \"column\": \"Sample ID\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 1,\n        \"samples\": [\n          \"input01\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Alterations\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 10,\n        \"samples\": [\n          \"EGFR wildtype\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Biomarker\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 64,\n        \"samples\": [\n          \"KIT N822K\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Drugs\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 44,\n        \"samples\": [\n          \"Regorafenib,Ponatinib,Imatinib,Sunitinib\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Diseases\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 10,\n        \"samples\": [\n          \"Thymic\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Response\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 3,\n        \"samples\": [\n          \"Responsive\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Evidence\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 4,\n        \"samples\": [\n          \"C\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Match\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 2,\n        \"samples\": [\n          \"YES\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Source\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 99,\n        \"samples\": [\n          \"https://civicdb.org/links/evidence_items/4583\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"BioM\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 3,\n        \"samples\": [\n          \"complete\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Resist.\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 449.33676432913626,\n        \"min\": 77.0,\n        \"max\": 1096.0,\n        \"num_unique_values\": 2,\n        \"samples\": [\n          1096.0\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Tumor type\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 10,\n        \"samples\": [\n          \"THYM\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}"
            }
          },
          "metadata": {},
          "execution_count": 57
        }
      ]
    }
  ]
}